{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Differential RNA expression analysis\n",
        "## (baking bread)"
      ],
      "metadata": {
        "id": "zuGhogSAGolf"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 0. PREPARING PROGRAMMS"
      ],
      "metadata": {
        "id": "o_6blWbXLP1b"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# aligner\n",
        "!apt install hisat2\n",
        "\n",
        "# sort, view, etc.\n",
        "!apt install samtools\n",
        "\n",
        "# conda\n",
        "!pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()\n",
        "!conda --version\n",
        "# gffread (in conda env)\n",
        "!conda install -c bioconda gffread\n",
        "\n",
        "# featureCounts\n",
        "!apt install subread\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZiP-7lSGLQe5",
        "outputId": "e0b170e2-4db3-4431-ebdc-b5d9001de1a0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "hisat2 is already the newest version (2.1.0-1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 39 not upgraded.\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  python-cryptography-doc python-cryptography-vectors python-enum34-doc\n",
            "  python-genshi python-lxml python-lockfile-doc python-openssl-doc\n",
            "  python-openssl-dbg python-setuptools python-pyparsing-doc python-rdflib-doc\n",
            "  python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 35 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 8,568 kB of archives.\n",
            "After this operation, 41.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 8,568 kB in 1s (15.4 MB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 156232 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../33-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../34-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "### 1.PREPARING DATA"
      ],
      "metadata": {
        "id": "0SSFEf2sLP9f"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Downloading yeast reads:"
      ],
      "metadata": {
        "id": "tqCNz5CnD0ea"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# SRR941816: fermentation 0 minutes replicate 1 \n",
        "!wget ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941816/SRR941816.fastq.gz\n",
        "!gunzip SRR941816.fastq.gz\n",
        "!mv SRR941816.fastq min0repl1.fastq\n",
        "\n",
        "# SRR941817: fermentation 0 minutes replicate 2 \n",
        "!wget ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941817/SRR941817.fastq.gz \n",
        "!gunzip SRR941817.fastq.gz\n",
        "!mv SRR941817.fastq min0repl2.fastq\n",
        "\n",
        "# SRR941818: fermentation 30 minutes replicate 1\n",
        "!wget ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941818/SRR941818.fastq.gz\n",
        "!gunzip SRR941818.fastq.gz\n",
        "!mv SRR941818.fastq min30repl1.fastq\n",
        "\n",
        "# SRR941819: fermentation 30 minutes replicate 2\n",
        "!wget ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941819/SRR941819.fastq.gz\n",
        "!gunzip SRR941819.fastq.gz\n",
        "!mv SRR941819.fastq min30repl2.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "O9KY0UotDJ-u",
        "outputId": "13cf9eab-623a-4872-e301-c016d177bb6c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-31 17:39:16--  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941816/SRR941816.fastq.gz\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 433539958 (413M) [application/octet-stream]\n",
            "Saving to: ‘SRR941816.fastq.gz’\n",
            "\n",
            "SRR941816.fastq.gz  100%[===================>] 413.46M  1.36MB/s    in 5m 53s  \n",
            "\n",
            "2022-03-31 17:45:10 (1.17 MB/s) - ‘SRR941816.fastq.gz’ saved [433539958/433539958]\n",
            "\n",
            "--2022-03-31 17:45:24--  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941817/SRR941817.fastq.gz\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 477099436 (455M) [application/octet-stream]\n",
            "Saving to: ‘SRR941817.fastq.gz’\n",
            "\n",
            "SRR941817.fastq.gz  100%[===================>] 455.00M  1.36MB/s    in 6m 16s  \n",
            "\n",
            "2022-03-31 17:51:41 (1.21 MB/s) - ‘SRR941817.fastq.gz’ saved [477099436/477099436]\n",
            "\n",
            "--2022-03-31 17:51:57--  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941818/SRR941818.fastq.gz\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 83133773 (79M) [application/octet-stream]\n",
            "Saving to: ‘SRR941818.fastq.gz’\n",
            "\n",
            "SRR941818.fastq.gz   32%[=====>              ]  26.00M   184KB/s    in 10m 56s \n",
            "\n",
            "2022-03-31 18:02:54 (40.6 KB/s) - Connection closed at byte 27262976. Retrying.\n",
            "\n",
            "--2022-03-31 18:02:55--  (try: 2)  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941818/SRR941818.fastq.gz\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:80... connected.\n",
            "HTTP request sent, awaiting response... 206 Partial Content\n",
            "Length: 83133773 (79M), 55870797 (53M) remaining [application/octet-stream]\n",
            "Saving to: ‘SRR941818.fastq.gz’\n",
            "\n",
            "SRR941818.fastq.gz  100%[++++++=============>]  79.28M   724KB/s    in 83s     \n",
            "\n",
            "2022-03-31 18:04:18 (659 KB/s) - ‘SRR941818.fastq.gz’ saved [83133773/83133773]\n",
            "\n",
            "--2022-03-31 18:04:21--  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941819/SRR941819.fastq.gz\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 295598163 (282M) [application/octet-stream]\n",
            "Saving to: ‘SRR941819.fastq.gz’\n",
            "\n",
            "SRR941819.fastq.gz   36%[======>             ] 103.97M   249KB/s    in 12m 16s \n",
            "\n",
            "2022-03-31 18:16:37 (145 KB/s) - Connection closed at byte 109018916. Retrying.\n",
            "\n",
            "--2022-03-31 18:16:38--  (try: 2)  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941819/SRR941819.fastq.gz\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:80... connected.\n",
            "HTTP request sent, awaiting response... 206 Partial Content\n",
            "Length: 295598163 (282M), 186579247 (178M) remaining [application/octet-stream]\n",
            "Saving to: ‘SRR941819.fastq.gz’\n",
            "\n",
            "SRR941819.fastq.gz   38%[+++++++             ] 108.47M   282KB/s    in 2m 51s  \n",
            "\n",
            "2022-03-31 18:19:29 (27.0 KB/s) - Connection closed at byte 113737728. Retrying.\n",
            "\n",
            "--2022-03-31 18:19:31--  (try: 3)  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941819/SRR941819.fastq.gz\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:80... connected.\n",
            "HTTP request sent, awaiting response... 206 Partial Content\n",
            "Length: 295598163 (282M), 181860435 (173M) remaining [application/octet-stream]\n",
            "Saving to: ‘SRR941819.fastq.gz’\n",
            "\n",
            "SRR941819.fastq.gz   60%[+++++++====>        ] 171.66M  48.2KB/s    in 13m 9s  \n",
            "\n",
            "2022-03-31 18:32:40 (82.1 KB/s) - Connection closed at byte 179994896. Retrying.\n",
            "\n",
            "--2022-03-31 18:32:43--  (try: 4)  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941819/SRR941819.fastq.gz\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:80... connected.\n",
            "HTTP request sent, awaiting response... 206 Partial Content\n",
            "Length: 295598163 (282M), 115603267 (110M) remaining [application/octet-stream]\n",
            "Saving to: ‘SRR941819.fastq.gz’\n",
            "\n",
            "SRR941819.fastq.gz   97%[++++++++++++======> ] 275.16M  --.-KB/s    in 47m 6s  \n",
            "\n",
            "2022-03-31 19:19:50 (37.5 KB/s) - Read error at byte 288530680/295598163 (Connection reset by peer). Retrying.\n",
            "\n",
            "--2022-03-31 19:19:54--  (try: 5)  http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR941/SRR941819/SRR941819.fastq.gz\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:80... connected.\n",
            "HTTP request sent, awaiting response... 206 Partial Content\n",
            "Length: 295598163 (282M), 7067483 (6.7M) remaining [application/octet-stream]\n",
            "Saving to: ‘SRR941819.fastq.gz’\n",
            "\n",
            "SRR941819.fastq.gz  100%[+++++++++++++++++++>] 281.90M   112KB/s    in 7m 5s   \n",
            "\n",
            "2022-03-31 19:27:00 (16.2 KB/s) - ‘SRR941819.fastq.gz’ saved [295598163/295598163]\n",
            "\n",
            "gzip: SRR941816.fastq.gz: No such file or directory\n",
            "mv: cannot stat 'SRR941816.fastq': No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "As a reference genome we will use *Saccharomyces cerevisiae* (strain S288c, assembly R64), in the genome database at [NCBI](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/):"
      ],
      "metadata": {
        "id": "XFo4wfvRFgb8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# reference genome file\n",
        "!wget ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.fna.gz\n",
        "!gunzip GCF_000146045.2_R64_genomic.fna.gz\n",
        "!mv GCF_000146045.2_R64_genomic.fna reference_R64.fna\n",
        "\n",
        "# annotation file\n",
        "!wget ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz\n",
        "!gunzip GCF_000146045.2_R64_genomic.gff.gz\n",
        "!mv GCF_000146045.2_R64_genomic.gff annotation_R64.gff\n",
        "\n",
        "# reference transcriptome (RNA)\n",
        "!wget ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_rna.fna.gz \n",
        "!gunzip GCF_000146045.2_R64_rna.fna.gz\n",
        "!mv GCF_000146045.2_R64_rna.fna referenceRNA_R64_rna.fna\n",
        "   "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "i3cIDwpcDKBm",
        "outputId": "ff2a1e46-a241-45f9-85ff-a7255a1f7269"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "URL transformed to HTTPS due to an HSTS policy\n",
            "--2022-03-31 21:55:04--  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.11, 2607:f220:41f:250::228, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2180038 (2.1M) [application/x-gzip]\n",
            "Saving to: ‘GCF_000146045.2_R64_genomic.gff.gz’\n",
            "\n",
            "\r          GCF_00014   0%[                    ]       0  --.-KB/s               \rGCF_000146045.2_R64 100%[===================>]   2.08M  --.-KB/s    in 0.1s    \n",
            "\n",
            "2022-03-31 21:55:05 (17.6 MB/s) - ‘GCF_000146045.2_R64_genomic.gff.gz’ saved [2180038/2180038]\n",
            "\n",
            "URL transformed to HTTPS due to an HSTS policy\n",
            "--2022-03-31 21:55:05--  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_rna.fna.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.11, 2607:f220:41f:250::228, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2784470 (2.7M) [application/x-gzip]\n",
            "Saving to: ‘GCF_000146045.2_R64_rna.fna.gz’\n",
            "\n",
            "GCF_000146045.2_R64 100%[===================>]   2.66M  8.53MB/s    in 0.3s    \n",
            "\n",
            "2022-03-31 21:55:05 (8.53 MB/s) - ‘GCF_000146045.2_R64_rna.fna.gz’ saved [2784470/2784470]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "### 2. ANALYSIS"
      ],
      "metadata": {
        "id": "qB8gSGVsL9t_"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "#### 2.1 Aligning with HISAT2"
      ],
      "metadata": {
        "id": "1A9mAmO5MHVA"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "First, we have to index our genome with `hisat2-build <reference.fasta> <genome index>` <br>\n",
        "`<reference.fasta>` $-$ comma-separated list of files with ref sequences<br>\n",
        "`<genome index>` $-$ write ht2 data to files with this dir/basename\n"
      ],
      "metadata": {
        "id": "FEqfSe_hMoqp"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!hisat2-build reference_R64.fna ref_index_R64"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z7sWO2GgD4zC",
        "outputId": "9e3001b1-62c3-4216-80ab-286ad1ff0724"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"ref_index_R64.*.ht2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Local offset rate: 3 (one in 8)\n",
            "  Local fTable chars: 6\n",
            "  Local sequence length: 57344\n",
            "  Local sequence overlap between two consecutive indexes: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  reference_R64.fna\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:00\n",
            "  Time to read SNPs and splice sites: 00:00:00\n",
            "Using parameters --bmax 2279457 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 2279457 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:00\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:00\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 1.73673e+06 (target: 2279456)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering GFM loop\n",
            "Getting block 1 of 7\n",
            "  Reserving size (2279457) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 1917483 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:00\n",
            "Returning block of 1917484 for bucket 1\n",
            "Getting block 2 of 7\n",
            "  Reserving size (2279457) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 1565107 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:00\n",
            "Returning block of 1565108 for bucket 2\n",
            "Getting block 3 of 7\n",
            "  Reserving size (2279457) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 2077071 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:01\n",
            "Returning block of 2077072 for bucket 3\n",
            "Getting block 4 of 7\n",
            "  Reserving size (2279457) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 1959963 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:01\n",
            "Returning block of 1959964 for bucket 4\n",
            "Getting block 5 of 7\n",
            "  Reserving size (2279457) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 1858740 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:01\n",
            "Returning block of 1858741 for bucket 5\n",
            "Getting block 6 of 7\n",
            "  Reserving size (2279457) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 630502 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:01\n",
            "Returning block of 630503 for bucket 6\n",
            "Getting block 7 of 7\n",
            "  Reserving size (2279457) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 2148233 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:01\n",
            "Returning block of 2148234 for bucket 7\n",
            "Exited GFM loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 3766349\n",
            "fchr[G]: 6086925\n",
            "fchr[T]: 8404025\n",
            "fchr[$]: 12157105\n",
            "Exiting GFM::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 8248454 bytes to primary GFM file: ref_index_R64.1.ht2\n",
            "Wrote 3039284 bytes to secondary GFM file: ref_index_R64.2.ht2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from GFM constructor\n",
            "Returning from initFromVector\n",
            "Wrote 5399069 bytes to primary GFM file: ref_index_R64.5.ht2\n",
            "Wrote 3092708 bytes to secondary GFM file: ref_index_R64.6.ht2\n",
            "Re-opening _in5 and _in5 as input streams\n",
            "Returning from HierEbwt constructor\n",
            "Headers:\n",
            "    len: 12157105\n",
            "    gbwtLen: 12157106\n",
            "    nodes: 12157106\n",
            "    sz: 3039277\n",
            "    gbwtSz: 3039277\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 0\n",
            "    eftabSz: 0\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 759820\n",
            "    offsSz: 3039280\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideGbwtSz: 48\n",
            "    sideGbwtLen: 192\n",
            "    numSides: 63319\n",
            "    numLines: 63319\n",
            "    gbwtTotLen: 4052416\n",
            "    gbwtTotSz: 4052416\n",
            "    reverse: 0\n",
            "    linearFM: Yes\n",
            "Total time for call to driver() for forward index: 00:00:11\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "Next, we will run hisat2 in single-end mode with the following command <br> `hisat2 -p [number of threads] -x [path to HISAT2 index] -U [FASTQ file]`<br> and then sort alignments with `samtools sort`."
      ],
      "metadata": {
        "id": "WCZRgsFvfwyn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!hisat2 -p 2 -x ref_index_R64 -U /content/min0repl1.fastq | samtools sort > out_0_1.bam\n",
        "!hisat2 -p 2 -x ref_index_R64 -U /content/min0repl2.fastq | samtools sort > out_0_2.bam\n",
        "!hisat2 -p 2 -x ref_index_R64 -U /content/min30repl1.fastq | samtools sort > out_30_1.bam\n",
        "!hisat2 -p 2 -x ref_index_R64 -U /content/min30repl2.fastq | samtools sort > out_30_2.bam"
      ],
      "metadata": {
        "id": "ZnYgPVWgj7L-",
        "outputId": "8be71ca7-08c5-43dd-9533-a396a67e1254",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "9929568 reads; of these:\n",
            "  9929568 (100.00%) were unpaired; of these:\n",
            "    511726 (5.15%) aligned 0 times\n",
            "    8644592 (87.06%) aligned exactly 1 time\n",
            "    773250 (7.79%) aligned >1 times\n",
            "94.85% overall alignment rate\n",
            "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
            "1721675 reads; of these:\n",
            "  1721675 (100.00%) were unpaired; of these:\n",
            "    66368 (3.85%) aligned 0 times\n",
            "    1507910 (87.58%) aligned exactly 1 time\n",
            "    147397 (8.56%) aligned >1 times\n",
            "96.15% overall alignment rate\n",
            "6172452 reads; of these:\n",
            "  6172452 (100.00%) were unpaired; of these:\n",
            "    234528 (3.80%) aligned 0 times\n",
            "    5367724 (86.96%) aligned exactly 1 time\n",
            "    570200 (9.24%) aligned >1 times\n",
            "96.20% overall alignment rate\n",
            "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# to see what's inside\n",
        "\n",
        "!samtools view /content/out_0_1.bam | head"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zB1-QhuRnN8A",
        "outputId": "c5106955-7b83-4f6b-f86c-363d7f861dcc"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "SRR941816.1244588\t0\tNC_001133.9\t1780\t60\t1S49M\t*\t0\t0\tCACCCGTTCGTAAAATTGGCGTTTGTCTCTAGTTTGCGATAGTGTAGATA\t@CCFFFDEHHHHHJJJJJJIJHIJJJJIIJIIHIIJJJGJJJGGEIIJJI\tAS:i:-1\tZS:i:-2\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:49\tYT:Z:UU\tNH:i:1\n",
            "SRR941816.5959616\t256\tNC_001133.9\t1787\t1\t50M\t*\t0\t0\tCGTAAAATTGGCGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTT\tCC@FFFFFHHHHHDHJIGGJIJIJIIIJJJHJJJFIGIIGIJJJJHIJJI\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:3\n",
            "SRR941816.735151\t256\tNC_001133.9\t1795\t1\t50M\t*\t0\t0\tTGGCGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAG\t@@@DD8@DCACFHBHECE@HGDG=GEHG?C?FGIGI>BFF?DHG3BB>GB\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:4\n",
            "SRR941816.2433849\t256\tNC_001133.9\t1796\t1\t50M\t*\t0\t0\tGGCGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAGC\tCCCFFFFFHHHHHJJIJJIIJJJJJHIGHIIJJJJJHIJJJJJIJIJJJI\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:5\n",
            "SRR941816.5944219\t0\tNC_001133.9\t1796\t1\t50M\t*\t0\t0\tGGCGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAGC\t@C@FDDDFHFHDF>FEGIEDHBHIBHG<FEHDHDHHJJJIG>GEHGIIFG\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:5\n",
            "SRR941816.149624\t272\tNC_001133.9\t1796\t1\t50M\t*\t0\t0\tGGCGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAGC\tHEFFJJIFJJIIIIJIJIJIIHJIJJJIJGIHGJJJJHHHHHFFFFFC@@\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:5\n",
            "SRR941816.26753\t256\tNC_001133.9\t1797\t1\t50M\t*\t0\t0\tGCGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAGCA\t@@@FDFFDFDHHHI<FHHJIGEHGIGFDGGIJGEEBGGIIIJIGHGBGEH\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:5\n",
            "SRR941816.473448\t0\tNC_001133.9\t1799\t1\t50M\t*\t0\t0\tGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAGCACT\tCBCFFFFFHHHHHJJJJJHJJJHIHIJJJJJJJHJJJJJJIJJJJJIJJJ\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:5\n",
            "SRR941816.5176110\t256\tNC_001133.9\t1799\t1\t50M\t*\t0\t0\tGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAGCACT\tCCCFFFFFHHHHHJJJJJIJJJHIHHIJIJJJJIJJJJJJGIIIIJJJJJ\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:5\n",
            "SRR941816.8085739\t256\tNC_001133.9\t1799\t1\t50M\t*\t0\t0\tGTTTGTCTCTAGTTTGCGATAGTGTAGATACCGTCCTTGGATAGAGCACT\tCCCFFFDFHHGHHIIJJJIJIIHIEIGIIEHIJGIJJJJJCHGGHIIIIG\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:50\tYT:Z:UU\tNH:i:5\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "#### 2.2 Quantifying with featureCounts"
      ],
      "metadata": {
        "id": "X3n04JXlMKDe"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "FeatureCounts can not work with GFF files (actually, seems like it can). We need to convert the GFF file to GTF format. For this purpose we will use gffread. <br>\n",
        "`gffread <input GFF> -T -o <output GTF>`"
      ],
      "metadata": {
        "id": "JNwT3KSi52i5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# Converting from GFF to GTF:\n",
        "# -T    main output will be GTF instead of GFF3\n",
        "# -o    write the output records into <outfile> instead of stdout\n",
        "\n",
        "!gffread annotation_R64.gff -T -o annotation_R64.gtf"
      ],
      "metadata": {
        "id": "1ax1fX9K5ulZ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "Now we are ready to run featureCounts.<br>\n",
        "`featureCounts -g gene_id -a <annotation file> -o <output file> <input file(s) - SAM/BAM>`"
      ],
      "metadata": {
        "id": "T833iPpm7zB5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "## Mandatory arguments:\n",
        "#   -a <string>         Name of an annotation file. GTF/GFF format by default.\n",
        "#                       See -F option for more format information. Inbuilt\n",
        "#                       annotations (SAF format) is available in 'annotation'\n",
        "#                       directory of the package.\n",
        "\n",
        "#   -o <string>         Name of the output file including read counts. A separate\n",
        "#                       file including summary statistics of counting results is\n",
        "#                       also included in the output ('<string>.summary')\n",
        "\n",
        "#   input_file1 [input_file2] ...   A list of SAM or BAM format files. They can be\n",
        "#                       either name or location sorted. If no files provided,\n",
        "#                       <stdin> input is expected. Location-sorted paired-end reads\n",
        "#                       are automatically sorted by read names.\n",
        "\n",
        "## Optional arguments:\n",
        "#   -g <string>         Specify attribute type in GTF annotation. 'gene_id' by \n",
        "#                       default. Meta-features used for read counting will be \n",
        "#                       extracted from annotation using the provided value.\n",
        "\n",
        "\n",
        "!featureCounts -g gene_id -a annotation_R64.gtf -o feature_R64 out*.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uSsPC32Goq0i",
        "outputId": "da9c4cc1-393b-48f4-9ef5-6965ecd1353a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
            "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
            "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
            "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
            "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
            "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
            "\t  v1.6.0\n",
            "\n",
            "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
            "||  \u001b[0m                                                                          ||\n",
            "||             Input files : \u001b[36m4 BAM files  \u001b[0m \u001b[0m                                   ||\n",
            "||                           \u001b[32mS\u001b[36m out_0_1.bam\u001b[0m \u001b[0m                                   ||\n",
            "||                           \u001b[32mS\u001b[36m out_0_2.bam\u001b[0m \u001b[0m                                   ||\n",
            "||                           \u001b[32mS\u001b[36m out_30_1.bam\u001b[0m \u001b[0m                                  ||\n",
            "||                           \u001b[32mS\u001b[36m out_30_2.bam\u001b[0m \u001b[0m                                  ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "||             Output file : \u001b[36mfeature_R64\u001b[0m \u001b[0m                                     ||\n",
            "||                 Summary : \u001b[36mfeature_R64.summary\u001b[0m \u001b[0m                             ||\n",
            "||              Annotation : \u001b[36mannotation_R64.gtf (GTF)\u001b[0m \u001b[0m                        ||\n",
            "||      Dir for temp files : \u001b[36m./\u001b[0m \u001b[0m                                              ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
            "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
            "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
            "||         Strand specific : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
            "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
            "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
            "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "\\\\=====================\u001b[36m http://subread.sourceforge.net/ \u001b[0m======================//\n",
            "\n",
            "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Load annotation file annotation_R64.gtf \u001b[0m... \u001b[0m                               ||\n",
            "||    Features : \u001b[36m6830\u001b[0m \u001b[0m                                                        ||\n",
            "||    Meta-features : \u001b[36m6458\u001b[0m \u001b[0m                                                   ||\n",
            "||    Chromosomes/contigs : \u001b[36m17\u001b[0m \u001b[0m                                               ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file out_0_1.bam... \u001b[0m                                           ||\n",
            "||    Single-end reads are included. \u001b[0m                                         ||\n",
            "||    Assign reads to features... \u001b[0m                                            ||\n",
            "||    Total reads : \u001b[36m9749623\u001b[0m \u001b[0m                                                  ||\n",
            "||    Successfully assigned reads : \u001b[36m7283047 (74.7%)\u001b[0m \u001b[0m                          ||\n",
            "||    Running time : \u001b[36m0.20 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file out_0_2.bam... \u001b[0m                                           ||\n",
            "||    Single-end reads are included. \u001b[0m                                         ||\n",
            "||    Assign reads to features... \u001b[0m                                            ||\n",
            "||    Total reads : \u001b[36m10810484\u001b[0m \u001b[0m                                                 ||\n",
            "||    Successfully assigned reads : \u001b[36m7982567 (73.8%)\u001b[0m \u001b[0m                          ||\n",
            "||    Running time : \u001b[36m0.21 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file out_30_1.bam... \u001b[0m                                          ||\n",
            "||    Single-end reads are included. \u001b[0m                                         ||\n",
            "||    Assign reads to features... \u001b[0m                                            ||\n",
            "||    Total reads : \u001b[36m1880816\u001b[0m \u001b[0m                                                  ||\n",
            "||    Successfully assigned reads : \u001b[36m1401055 (74.5%)\u001b[0m \u001b[0m                          ||\n",
            "||    Running time : \u001b[36m0.04 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file out_30_2.bam... \u001b[0m                                          ||\n",
            "||    Single-end reads are included. \u001b[0m                                         ||\n",
            "||    Assign reads to features... \u001b[0m                                            ||\n",
            "||    Total reads : \u001b[36m6782903\u001b[0m \u001b[0m                                                  ||\n",
            "||    Successfully assigned reads : \u001b[36m4972100 (73.3%)\u001b[0m \u001b[0m                          ||\n",
            "||    Running time : \u001b[36m0.14 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "||                         \u001b[36mRead assignment finished.\u001b[0m \u001b[0m                         ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Summary of counting results can be found in file \"feature_R64.summary\" \u001b[0m    ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "\\\\=====================\u001b[36m http://subread.sourceforge.net/ \u001b[0m======================//\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "Summary:"
      ],
      "metadata": {
        "id": "oUje_rLQDP3Y"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cat feature_R64.summary"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CaRBBGoWoq3e",
        "outputId": "4fb58cdc-c2af-4683-f9b0-e22f71ae35b5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Status\tout_0_1.bam\tout_0_2.bam\tout_30_1.bam\tout_30_2.bam\n",
            "Assigned\t7283047\t7982567\t1401055\t4972100\n",
            "Unassigned_Unmapped\t520137\t511726\t66368\t234528\n",
            "Unassigned_MappingQuality\t0\t0\t0\t0\n",
            "Unassigned_Chimera\t0\t0\t0\t0\n",
            "Unassigned_FragmentLength\t0\t0\t0\t0\n",
            "Unassigned_Duplicate\t0\t0\t0\t0\n",
            "Unassigned_MultiMapping\t1299782\t1654166\t306538\t1180651\n",
            "Unassigned_Secondary\t0\t0\t0\t0\n",
            "Unassigned_Nonjunction\t0\t0\t0\t0\n",
            "Unassigned_NoFeatures\t610914\t626058\t102674\t380347\n",
            "Unassigned_Overlapping_Length\t0\t0\t0\t0\n",
            "Unassigned_Ambiguity\t35743\t35967\t4181\t15277\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "We don’t need all columns from featureCounts output file for further analysis, so we'll simplify it:"
      ],
      "metadata": {
        "id": "FaigofveDagg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cat feature_R64 | cut -f 1,7-10 > simple_counts.txt"
      ],
      "metadata": {
        "id": "Q5tNikXUDaw0"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!head simple_counts.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vhq3ONpdpVyl",
        "outputId": "94df6041-6a86-4fd9-c676-d437ee6ad8e6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "# Program:featureCounts v1.6.0; Command:\"featureCounts\" \"-g\" \"gene_id\" \"-a\" \"annotation_R64.gtf\" \"-o\" \"feature_R64\" \"out_0_1.bam\" \"out_0_2.bam\" \"out_30_1.bam\" \"out_30_2.bam\" \n",
            "Geneid\tout_0_1.bam\tout_0_2.bam\tout_30_1.bam\tout_30_2.bam\n",
            "gene-YAL068C\t14\t16\t2\t6\n",
            "gene-YAL067W-A\t0\t0\t0\t0\n",
            "gene-YAL067C\t116\t69\t5\t11\n",
            "gene-YAL065C\t11\t3\t2\t3\n",
            "gene-YAL064W-B\t1\t3\t0\t0\n",
            "gene-YAL064C-A\t0\t1\t1\t0\n",
            "gene-YAL064W\t0\t0\t0\t0\n",
            "gene-YAL063C-A\t0\t0\t0\t0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# total genes (-2 for header)\n",
        "\n",
        "!cat simple_counts.txt | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Cw6-VvKA_1TB",
        "outputId": "415d29a9-15d5-4385-bbc6-dce2130a3654"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "6460\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# total expressed genes (approximately)\n",
        "\n",
        "!cat simple_counts.txt | cut -f2,3,4,5 | awk '$1+$2 > $4+$5 {print $1,\"\\t\",$2}' | wc -l\n",
        "!cat simple_counts.txt | cut -f2,3,4,5 | awk '$1+$2 <= $4+$5 {print $1,\"\\t\",$2}' | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IrLNRH5CcJdR",
        "outputId": "f8cd86ad-d7fd-4c1c-e30a-2c01b631acd2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "4701\n",
            "1759\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "#### 2.3 Find differentially expressed genes with Deseq2"
      ],
      "metadata": {
        "id": "aXEzEHwcMYZO"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "First, let's download some R scripts:\n",
        "* `deseq2.r` calculates some metrics for our genes and generates following files:\n",
        "    * result.txt will contain calculated metrics for our genes\n",
        "    * norm-matrix-deseq2.txt will contain normalised counts that we will use in visualisation\n",
        "\n",
        "* `draw-heatmap.r` draws heatmap"
      ],
      "metadata": {
        "id": "iRAstwj1D6W0"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://figshare.com/ndownloader/files/26882912\n",
        "!mv 26882912 deseq2.r\n",
        "!wget https://figshare.com/ndownloader/files/26882924\n",
        "!mv 26882924 draw-heatmap.r"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JOpgbjGxC496",
        "outputId": "2abc9359-50e6-4e5d-bbe0-97244bf61540"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-31 22:20:48--  https://figshare.com/ndownloader/files/26882912\n",
            "Resolving figshare.com (figshare.com)... 63.33.124.90, 54.229.21.211, 2a05:d018:1f4:d003:3c75:e77:d705:4610, ...\n",
            "Connecting to figshare.com (figshare.com)|63.33.124.90|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://s3-eu-west-1.amazonaws.com/pfigshare-u-files/26882912/deseq2.r?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIYCQYOYV5JSSROOA/20220331/eu-west-1/s3/aws4_request&X-Amz-Date=20220331T222048Z&X-Amz-Expires=10&X-Amz-SignedHeaders=host&X-Amz-Signature=4b20dcef59bff0a146b1a73a494fd2eadc2161125e519ce10bb36769e67cbf0f [following]\n",
            "--2022-03-31 22:20:48--  https://s3-eu-west-1.amazonaws.com/pfigshare-u-files/26882912/deseq2.r?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIYCQYOYV5JSSROOA/20220331/eu-west-1/s3/aws4_request&X-Amz-Date=20220331T222048Z&X-Amz-Expires=10&X-Amz-SignedHeaders=host&X-Amz-Signature=4b20dcef59bff0a146b1a73a494fd2eadc2161125e519ce10bb36769e67cbf0f\n",
            "Resolving s3-eu-west-1.amazonaws.com (s3-eu-west-1.amazonaws.com)... 52.218.97.18\n",
            "Connecting to s3-eu-west-1.amazonaws.com (s3-eu-west-1.amazonaws.com)|52.218.97.18|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1343 (1.3K) [application/octet-stream]\n",
            "Saving to: ‘26882912.1’\n",
            "\n",
            "26882912.1          100%[===================>]   1.31K  --.-KB/s    in 0s      \n",
            "\n",
            "2022-03-31 22:20:49 (20.8 MB/s) - ‘26882912.1’ saved [1343/1343]\n",
            "\n",
            "--2022-03-31 22:20:49--  https://figshare.com/ndownloader/files/26882924\n",
            "Resolving figshare.com (figshare.com)... 63.33.124.90, 54.229.21.211, 2a05:d018:1f4:d003:3c75:e77:d705:4610, ...\n",
            "Connecting to figshare.com (figshare.com)|63.33.124.90|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://s3-eu-west-1.amazonaws.com/pfigshare-u-files/26882924/drawheatmap.r?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIYCQYOYV5JSSROOA/20220331/eu-west-1/s3/aws4_request&X-Amz-Date=20220331T222049Z&X-Amz-Expires=10&X-Amz-SignedHeaders=host&X-Amz-Signature=9ad8d8b9ec79bba60ef5cd61a0daf8892a828876084bd30e7c502bdb0b4b06ae [following]\n",
            "--2022-03-31 22:20:49--  https://s3-eu-west-1.amazonaws.com/pfigshare-u-files/26882924/drawheatmap.r?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIYCQYOYV5JSSROOA/20220331/eu-west-1/s3/aws4_request&X-Amz-Date=20220331T222049Z&X-Amz-Expires=10&X-Amz-SignedHeaders=host&X-Amz-Signature=9ad8d8b9ec79bba60ef5cd61a0daf8892a828876084bd30e7c502bdb0b4b06ae\n",
            "Resolving s3-eu-west-1.amazonaws.com (s3-eu-west-1.amazonaws.com)... 52.218.97.18\n",
            "Connecting to s3-eu-west-1.amazonaws.com (s3-eu-west-1.amazonaws.com)|52.218.97.18|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 770 [application/octet-stream]\n",
            "Saving to: ‘26882924’\n",
            "\n",
            "26882924            100%[===================>]     770  --.-KB/s    in 0s      \n",
            "\n",
            "2022-03-31 22:20:50 (34.1 MB/s) - ‘26882924’ saved [770/770]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "Following {r} code was added to the first script<br>\n",
        "`if (!requireNamespace(\"BiocManager\", quietly = TRUE))`\n",
        "<br>....`install.packages(\"BiocManager\")`\n",
        "<br>`BiocManager::install(\"DESeq2\")`\n",
        "\n",
        "$~$"
      ],
      "metadata": {
        "id": "mnhXF7PqJY7P"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "Calculate metrics using modified first script:<br>"
      ],
      "metadata": {
        "id": "WZ17L2tAEtpF"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cat simple_counts.txt | R -f deseq2.r"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RrZBvpqJ-7p8",
        "outputId": "c86cdb1d-9376-4ef7-e073-e653cdbf3c88"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "R version 4.1.3 (2022-03-10) -- \"One Push-Up\"\n",
            "Copyright (C) 2022 The R Foundation for Statistical Computing\n",
            "Platform: x86_64-pc-linux-gnu (64-bit)\n",
            "\n",
            "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
            "You are welcome to redistribute it under certain conditions.\n",
            "Type 'license()' or 'licence()' for distribution details.\n",
            "\n",
            "  Natural language support but running in an English locale\n",
            "\n",
            "R is a collaborative project with many contributors.\n",
            "Type 'contributors()' for more information and\n",
            "'citation()' on how to cite R or R packages in publications.\n",
            "\n",
            "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
            "'help.start()' for an HTML browser interface to help.\n",
            "Type 'q()' to quit R.\n",
            "\n",
            "> # Load the library.\n",
            "> if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
            "+     install.packages(\"BiocManager\")\n",
            "> BiocManager::install(\"DESeq2\")\n",
            "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
            "'?repositories' for details\n",
            "\n",
            "replacement repositories:\n",
            "    CRAN: https://cran.rstudio.com\n",
            "\n",
            "Bioconductor version 3.14 (BiocManager 1.30.16), R 4.1.3 (2022-03-10)\n",
            "Installing package(s) 'DESeq2'\n",
            "also installing the dependencies ‘bitops’, ‘formatR’, ‘plogr’, ‘png’, ‘Biostrings’, ‘RCurl’, ‘GenomeInfoDbData’, ‘zlibbioc’, ‘matrixStats’, ‘lambda.r’, ‘futile.options’, ‘RSQLite’, ‘KEGGREST’, ‘XML’, ‘xtable’, ‘GenomeInfoDb’, ‘XVector’, ‘MatrixGenerics’, ‘DelayedArray’, ‘futile.logger’, ‘snow’, ‘BH’, ‘AnnotationDbi’, ‘annotate’, ‘S4Vectors’, ‘IRanges’, ‘GenomicRanges’, ‘SummarizedExperiment’, ‘BiocGenerics’, ‘Biobase’, ‘BiocParallel’, ‘genefilter’, ‘locfit’, ‘geneplotter’, ‘RcppArmadillo’\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/bitops_1.0-7.tar.gz'\n",
            "Content type 'application/x-gzip' length 10809 bytes (10 KB)\n",
            "==================================================\n",
            "downloaded 10 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/formatR_1.12.tar.gz'\n",
            "Content type 'application/x-gzip' length 95944 bytes (93 KB)\n",
            "==================================================\n",
            "downloaded 93 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/plogr_0.2.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 7795 bytes\n",
            "==================================================\n",
            "downloaded 7795 bytes\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/png_0.1-7.tar.gz'\n",
            "Content type 'application/x-gzip' length 24990 bytes (24 KB)\n",
            "==================================================\n",
            "downloaded 24 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/Biostrings_2.62.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 12645827 bytes (12.1 MB)\n",
            "==================================================\n",
            "downloaded 12.1 MB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/RCurl_1.98-1.6.tar.gz'\n",
            "Content type 'application/x-gzip' length 731035 bytes (713 KB)\n",
            "==================================================\n",
            "downloaded 713 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/data/annotation/src/contrib/GenomeInfoDbData_1.2.7.tar.gz'\n",
            "Content type 'application/x-gzip' length 11193674 bytes (10.7 MB)\n",
            "==================================================\n",
            "downloaded 10.7 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/zlibbioc_1.40.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 257409 bytes (251 KB)\n",
            "==================================================\n",
            "downloaded 251 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/matrixStats_0.61.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 201077 bytes (196 KB)\n",
            "==================================================\n",
            "downloaded 196 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/lambda.r_1.2.4.tar.gz'\n",
            "Content type 'application/x-gzip' length 25666 bytes (25 KB)\n",
            "==================================================\n",
            "downloaded 25 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/futile.options_1.0.1.tar.gz'\n",
            "Content type 'application/x-gzip' length 3919 bytes\n",
            "==================================================\n",
            "downloaded 3919 bytes\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/RSQLite_2.2.11.tar.gz'\n",
            "Content type 'application/x-gzip' length 3693890 bytes (3.5 MB)\n",
            "==================================================\n",
            "downloaded 3.5 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/KEGGREST_1.34.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 25778 bytes (25 KB)\n",
            "==================================================\n",
            "downloaded 25 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/XML_3.99-0.9.tar.gz'\n",
            "Content type 'application/x-gzip' length 970514 bytes (947 KB)\n",
            "==================================================\n",
            "downloaded 947 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/xtable_1.8-4.tar.gz'\n",
            "Content type 'application/x-gzip' length 564589 bytes (551 KB)\n",
            "==================================================\n",
            "downloaded 551 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/GenomeInfoDb_1.30.1.tar.gz'\n",
            "Content type 'application/x-gzip' length 3457147 bytes (3.3 MB)\n",
            "==================================================\n",
            "downloaded 3.3 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/XVector_0.34.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 67783 bytes (66 KB)\n",
            "==================================================\n",
            "downloaded 66 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/MatrixGenerics_1.6.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 28270 bytes (27 KB)\n",
            "==================================================\n",
            "downloaded 27 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/DelayedArray_0.20.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 676428 bytes (660 KB)\n",
            "==================================================\n",
            "downloaded 660 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/futile.logger_1.4.3.tar.gz'\n",
            "Content type 'application/x-gzip' length 17456 bytes (17 KB)\n",
            "==================================================\n",
            "downloaded 17 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/snow_0.4-4.tar.gz'\n",
            "Content type 'application/x-gzip' length 20464 bytes (19 KB)\n",
            "==================================================\n",
            "downloaded 19 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/BH_1.78.0-0.tar.gz'\n",
            "Content type 'application/x-gzip' length 13269768 bytes (12.7 MB)\n",
            "==================================================\n",
            "downloaded 12.7 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/AnnotationDbi_1.56.2.tar.gz'\n",
            "Content type 'application/x-gzip' length 4330396 bytes (4.1 MB)\n",
            "==================================================\n",
            "downloaded 4.1 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/annotate_1.72.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 1723060 bytes (1.6 MB)\n",
            "==================================================\n",
            "downloaded 1.6 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/S4Vectors_0.32.4.tar.gz'\n",
            "Content type 'application/x-gzip' length 668558 bytes (652 KB)\n",
            "==================================================\n",
            "downloaded 652 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/IRanges_2.28.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 455647 bytes (444 KB)\n",
            "==================================================\n",
            "downloaded 444 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/GenomicRanges_1.46.1.tar.gz'\n",
            "Content type 'application/x-gzip' length 1171939 bytes (1.1 MB)\n",
            "==================================================\n",
            "downloaded 1.1 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/SummarizedExperiment_1.24.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 1524667 bytes (1.5 MB)\n",
            "==================================================\n",
            "downloaded 1.5 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/BiocGenerics_0.40.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 45969 bytes (44 KB)\n",
            "==================================================\n",
            "downloaded 44 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/Biobase_2.54.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 1641111 bytes (1.6 MB)\n",
            "==================================================\n",
            "downloaded 1.6 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/BiocParallel_1.28.3.tar.gz'\n",
            "Content type 'application/x-gzip' length 883999 bytes (863 KB)\n",
            "==================================================\n",
            "downloaded 863 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/genefilter_1.76.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 807639 bytes (788 KB)\n",
            "==================================================\n",
            "downloaded 788 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/locfit_1.5-9.5.tar.gz'\n",
            "Content type 'application/x-gzip' length 201031 bytes (196 KB)\n",
            "==================================================\n",
            "downloaded 196 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/geneplotter_1.72.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 1437803 bytes (1.4 MB)\n",
            "==================================================\n",
            "downloaded 1.4 MB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/RcppArmadillo_0.10.8.1.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 1364404 bytes (1.3 MB)\n",
            "==================================================\n",
            "downloaded 1.3 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.14/bioc/src/contrib/DESeq2_1.34.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 2088431 bytes (2.0 MB)\n",
            "==================================================\n",
            "downloaded 2.0 MB\n",
            "\n",
            "* installing *source* package ‘bitops’ ...\n",
            "** package ‘bitops’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c bit-ops.c -o bit-ops.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c cksum.c -o cksum.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.c -o init.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o bitops.so bit-ops.o cksum.o init.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-bitops/00new/bitops/libs\n",
            "** R\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (bitops)\n",
            "* installing *source* package ‘formatR’ ...\n",
            "** package ‘formatR’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (formatR)\n",
            "* installing *source* package ‘plogr’ ...\n",
            "** package ‘plogr’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (plogr)\n",
            "* installing *source* package ‘png’ ...\n",
            "** package ‘png’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      `libpng-config --cflags` -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c read.c -o read.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      `libpng-config --cflags` -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c write.c -o write.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o png.so read.o write.o -lpng16 -lm -lz -lm -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-png/00new/png/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (png)\n",
            "* installing *source* package ‘GenomeInfoDbData’ ...\n",
            "** using staged installation\n",
            "** data\n",
            "** inst\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (GenomeInfoDbData)\n",
            "* installing *source* package ‘zlibbioc’ ...\n",
            "** using staged installation\n",
            "configure: creating ./config.status\n",
            "config.status: creating src/Makevars\n",
            "** libs\n",
            "mkdir -p \"/usr/local/lib/R/site-library/00LOCK-zlibbioc/00new/zlibbioc/include\"\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c R_init_zlibbioc.c -o R_init_zlibbioc.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o zlibbioc.so R_init_zlibbioc.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-zlibbioc/00new/zlibbioc/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (zlibbioc)\n",
            "* installing *source* package ‘matrixStats’ ...\n",
            "** package ‘matrixStats’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c 000.init.c -o 000.init.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c allocMatrix2.c -o allocMatrix2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c anyMissing.c -o anyMissing.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c binCounts.c -o binCounts.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c binMeans.c -o binMeans.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c colCounts.c -o colCounts.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c colOrderStats.c -o colOrderStats.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c colRanges.c -o colRanges.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c diff2.c -o diff2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c indexByRow.c -o indexByRow.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c logSumExp.c -o logSumExp.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c mean2.c -o mean2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c naming.c -o naming.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c productExpSumLog.c -o productExpSumLog.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c psortKM.c -o psortKM.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowCounts.c -o rowCounts.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowCummaxs.c -o rowCummaxs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowCummins.c -o rowCummins.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowCumprods.c -o rowCumprods.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowCumsums.c -o rowCumsums.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowDiffs.c -o rowDiffs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowLogSumExp.c -o rowLogSumExp.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowMads.c -o rowMads.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowMeans2.c -o rowMeans2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowMedians.c -o rowMedians.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowOrderStats.c -o rowOrderStats.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowRanges.c -o rowRanges.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowRanksWithTies.c -o rowRanksWithTies.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowSums2.c -o rowSums2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowVars.c -o rowVars.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c signTabulate.c -o signTabulate.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sum2.c -o sum2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c validateIndices.c -o validateIndices.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c weightedMean.c -o weightedMean.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c weightedMedian.c -o weightedMedian.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c x_OP_y.c -o x_OP_y.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o matrixStats.so 000.init.o allocMatrix2.o anyMissing.o binCounts.o binMeans.o colCounts.o colOrderStats.o colRanges.o diff2.o indexByRow.o logSumExp.o mean2.o naming.o productExpSumLog.o psortKM.o rowCounts.o rowCummaxs.o rowCummins.o rowCumprods.o rowCumsums.o rowDiffs.o rowLogSumExp.o rowMads.o rowMeans2.o rowMedians.o rowOrderStats.o rowRanges.o rowRanksWithTies.o rowSums2.o rowVars.o signTabulate.o sum2.o validateIndices.o weightedMean.o weightedMedian.o x_OP_y.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-matrixStats/00new/matrixStats/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (matrixStats)\n",
            "* installing *source* package ‘futile.options’ ...\n",
            "** package ‘futile.options’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (futile.options)\n",
            "* installing *source* package ‘XML’ ...\n",
            "** package ‘XML’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "checking for gcc... gcc\n",
            "checking whether the C compiler works... yes\n",
            "checking for C compiler default output file name... a.out\n",
            "checking for suffix of executables... \n",
            "checking whether we are cross compiling... no\n",
            "checking for suffix of object files... o\n",
            "checking whether the compiler supports GNU C... yes\n",
            "checking whether gcc accepts -g... yes\n",
            "checking for gcc option to enable C11 features... none needed\n",
            "checking how to run the C preprocessor... gcc -E\n",
            "checking for sed... /bin/sed\n",
            "checking for pkg-config... /usr/bin/pkg-config\n",
            "checking for xml2-config... /usr/local/bin/xml2-config\n",
            "USE_XML2 = yes\n",
            "SED_EXTENDED_ARG: -E\n",
            "Minor 9, Patch 10 for 2.9.10\n",
            "Located parser file -I/usr/local/include/libxml2 -I/usr/local/include/parser.h\n",
            "Checking for 1.8:  -I/usr/local/include/libxml2 -I/usr/local/include\n",
            "Using libxml2.*\n",
            "checking for gzopen in -lz... yes\n",
            "checking for xmlParseFile in -lxml2... yes\n",
            "checking for xmlHashSize in -lxml2... yes\n",
            "Using built-in xmlHashSize\n",
            "Checking DTD parsing (presence of externalSubset)...\n",
            "checking for xmlHashSize in -lxml2... (cached) yes\n",
            "Found xmlHashSize\n",
            "checking for xmlOutputBufferCreateBuffer in -lxml2... yes\n",
            "have xmlOutputBufferCreateBuffer()\n",
            "checking for xmlDocDumpFormatMemoryEnc in -lxml2... yes\n",
            "checking for libxml/xmlversion.h... yes\n",
            "Expat:  FALSE\n",
            "Checking for return type of xmlHashScan element routine.\n",
            "No return value for xmlHashScan\n",
            "xmlNs has a context field\n",
            "Checking for cetype_t enumeration\n",
            "No cetype_t enumeration defined in R headers.\n",
            "checking for xmlsec1-config... no\n",
            "nodegc default \n",
            "xml-debug default \n",
            "Version has XML_WITH_ZLIB\n",
            "Version has xmlHasFeature()\n",
            "\n",
            "****************************************\n",
            "Configuration information:\n",
            "\n",
            "Libxml settings\n",
            "\n",
            "libxml include directory: -I/usr/local/include/libxml2 -I/usr/local/include\n",
            "libxml library directory: -L/usr/local/lib -lxml2 -L/usr/local/lib -lz -L/usr/local/lib -llzma -lpthread -L/usr/local/lib -liconv -L/usr/local/lib -licui18n -licuuc -licudata -lm -ldl -lz  -lxml2\n",
            "libxml 2:                 -DLIBXML2=1\n",
            "\n",
            "Compilation flags:         -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1 \n",
            "Link flags:               -L/usr/local/lib -lxml2 -L/usr/local/lib -lz -L/usr/local/lib -llzma -lpthread -L/usr/local/lib -liconv -L/usr/local/lib -licui18n -licuuc -licudata -lm -ldl -lz  -lxml2\n",
            "\n",
            "****************************************\n",
            "configure: creating ./config.status\n",
            "config.status: creating src/Makevars\n",
            "config.status: creating R/supports.R\n",
            "config.status: creating inst/scripts/RSXML.csh\n",
            "config.status: creating inst/scripts/RSXML.bsh\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DocParse.c -o DocParse.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c EventParse.c -o EventParse.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ExpatParse.c -o ExpatParse.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c HTMLParse.c -o HTMLParse.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c NodeGC.c -o NodeGC.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RSDTD.c -o RSDTD.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RUtils.c -o RUtils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Rcatalog.c -o Rcatalog.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Utils.c -o Utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XMLEventParse.c -o XMLEventParse.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XMLHashTree.c -o XMLHashTree.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XMLTree.c -o XMLTree.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c fixNS.c -o fixNS.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c libxmlFeatures.c -o libxmlFeatures.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c schema.c -o schema.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c xmlsecurity.c -o xmlsecurity.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -DLIBXML -I/usr/local/include/libxml2 -I/usr/local/include -DUSE_EXTERNAL_SUBSET=1 -DROOT_HAS_DTD_NODE=1 -DDUMP_WITH_ENCODING=1 -DUSE_XML_VERSION_H=1 -DXML_ELEMENT_ETYPE=1 -DXML_ATTRIBUTE_ATYPE=1 -DNO_XML_HASH_SCANNER_RETURN=1 -DLIBXML_NAMESPACE_HAS_CONTEXT=1 -DHAVE_XML_WITH_ZLIB=1 -DHAVE_XML_HAS_FEATURE=1 -DUSE_R=1 -D_R_=1  -DHAVE_VALIDITY=1 -DXML_REF_COUNT_NODES=1  -I. -DLIBXML2=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c xpath.c -o xpath.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o XML.so DocParse.o EventParse.o ExpatParse.o HTMLParse.o NodeGC.o RSDTD.o RUtils.o Rcatalog.o Utils.o XMLEventParse.o XMLHashTree.o XMLTree.o fixNS.o libxmlFeatures.o schema.o xmlsecurity.o xpath.o -L/usr/local/lib -lxml2 -L/usr/local/lib -lz -L/usr/local/lib -llzma -lpthread -L/usr/local/lib -liconv -L/usr/local/lib -licui18n -licuuc -licudata -lm -ldl -lz -lxml2 -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-XML/00new/XML/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Creating a generic function for ‘source’ from package ‘base’ in package ‘XML’\n",
            "in method for ‘xmlAttrsToDataFrame’ with signature ‘\"AsIs\"’: no definition for class “AsIs”\n",
            "in method for ‘readKeyValueDB’ with signature ‘\"AsIs\"’: no definition for class “AsIs”\n",
            "in method for ‘readSolrDoc’ with signature ‘\"AsIs\"’: no definition for class “AsIs”\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (XML)\n",
            "* installing *source* package ‘xtable’ ...\n",
            "** package ‘xtable’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (xtable)\n",
            "* installing *source* package ‘snow’ ...\n",
            "** package ‘snow’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (snow)\n",
            "* installing *source* package ‘BH’ ...\n",
            "** package ‘BH’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** inst\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (BH)\n",
            "* installing *source* package ‘BiocGenerics’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Creating a new generic function for ‘append’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘as.data.frame’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘cbind’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘rbind’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘do.call’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘duplicated’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘anyDuplicated’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘eval’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘pmax’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘pmin’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘pmax.int’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘pmin.int’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘Reduce’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘Filter’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘Find’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘Map’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘Position’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘get’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘mget’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘grep’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘grepl’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘is.unsorted’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘lapply’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘sapply’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘mapply’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘match’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘order’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘paste’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘rank’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘rownames’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘colnames’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘union’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘intersect’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘setdiff’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘sort’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘table’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘tapply’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘unique’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘unsplit’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘var’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘sd’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘which.min’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘which.max’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘IQR’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘mad’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘xtabs’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘basename’ in package ‘BiocGenerics’\n",
            "Creating a new generic function for ‘dirname’ in package ‘BiocGenerics’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (BiocGenerics)\n",
            "* installing *source* package ‘locfit’ ...\n",
            "** package ‘locfit’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c S_enter.c -o S_enter.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c band.c -o band.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c dbinom.c -o dbinom.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c dens_haz.c -o dens_haz.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c dens_int.c -o dens_int.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c dens_odi.c -o dens_odi.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c density.c -o density.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ev_atree.c -o ev_atree.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ev_interp.c -o ev_interp.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ev_kdtre.c -o ev_kdtre.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ev_main.c -o ev_main.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ev_sphere.c -o ev_sphere.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ev_trian.c -o ev_trian.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c family.c -o family.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c fitted.c -o fitted.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c frend.c -o frend.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lf_adap.c -o lf_adap.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lf_dercor.c -o lf_dercor.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lf_fitfun.c -o lf_fitfun.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lf_nbhd.c -o lf_nbhd.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lf_robust.c -o lf_robust.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lf_vari.c -o lf_vari.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lf_wdiag.c -o lf_wdiag.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lfstr.c -o lfstr.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c locfit.c -o locfit.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_chol.c -o m_chol.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_eigen.c -o m_eigen.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_icirc.c -o m_icirc.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_imont.c -o m_imont.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_isimp.c -o m_isimp.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_isphr.c -o m_isphr.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_jacob.c -o m_jacob.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_max.c -o m_max.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_qr.c -o m_qr.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_solve.c -o m_solve.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_svd.c -o m_svd.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c m_vector.c -o m_vector.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c math.c -o math.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c minmax.c -o minmax.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c pcomp.c -o pcomp.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c preplot.c -o preplot.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c prob.c -o prob.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c procv.c -o procv.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c scb.c -o scb.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c scb_cons.c -o scb_cons.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c scb_crit.c -o scb_crit.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c scb_iface.c -o scb_iface.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c simul.c -o simul.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c smisc.c -o smisc.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c startlf.c -o startlf.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c weight.c -o weight.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o locfit.so S_enter.o band.o dbinom.o dens_haz.o dens_int.o dens_odi.o density.o ev_atree.o ev_interp.o ev_kdtre.o ev_main.o ev_sphere.o ev_trian.o family.o fitted.o frend.o lf_adap.o lf_dercor.o lf_fitfun.o lf_nbhd.o lf_robust.o lf_vari.o lf_wdiag.o lfstr.o locfit.o m_chol.o m_eigen.o m_icirc.o m_imont.o m_isimp.o m_isphr.o m_jacob.o m_max.o m_qr.o m_solve.o m_svd.o m_vector.o math.o minmax.o pcomp.o preplot.o prob.o procv.o scb.o scb_cons.o scb_crit.o scb_iface.o simul.o smisc.o startlf.o weight.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-locfit/00new/locfit/libs\n",
            "** R\n",
            "** data\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (locfit)\n",
            "* installing *source* package ‘RcppArmadillo’ ...\n",
            "** package ‘RcppArmadillo’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "checking whether the C++ compiler works... yes\n",
            "checking for C++ compiler default output file name... a.out\n",
            "checking for suffix of executables... \n",
            "checking whether we are cross compiling... no\n",
            "checking for suffix of object files... o\n",
            "checking whether we are using the GNU C++ compiler... yes\n",
            "checking whether g++ -std=gnu++14 accepts -g... yes\n",
            "checking how to run the C++ preprocessor... g++ -std=gnu++14 -E\n",
            "checking whether we are using the GNU C++ compiler... (cached) yes\n",
            "checking whether g++ -std=gnu++14 accepts -g... (cached) yes\n",
            "checking whether we have a suitable tempdir... /tmp\n",
            "checking whether R CMD SHLIB can already compile programs using OpenMP... yes\n",
            "checking LAPACK_LIBS... system LAPACK found\n",
            "configure: creating ./config.status\n",
            "config.status: creating inst/include/RcppArmadilloConfigGenerated.h\n",
            "config.status: creating src/Makevars\n",
            "** libs\n",
            "g++ -std=gnu++11 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/Rcpp/include'   -I../inst/include  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RcppArmadillo.cpp -o RcppArmadillo.o\n",
            "g++ -std=gnu++11 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/Rcpp/include'   -I../inst/include  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RcppExports.cpp -o RcppExports.o\n",
            "g++ -std=gnu++11 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/Rcpp/include'   -I../inst/include  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c fastLm.cpp -o fastLm.o\n",
            "g++ -std=gnu++11 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o RcppArmadillo.so RcppArmadillo.o RcppExports.o fastLm.o -llapack -lblas -lgfortran -lm -lquadmath -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-RcppArmadillo/00new/RcppArmadillo/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (RcppArmadillo)\n",
            "* installing *source* package ‘RCurl’ ...\n",
            "** package ‘RCurl’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "checking for curl-config... /usr/local/bin/curl-config\n",
            "checking for gcc... gcc\n",
            "checking whether the C compiler works... yes\n",
            "checking for C compiler default output file name... a.out\n",
            "checking for suffix of executables... \n",
            "checking whether we are cross compiling... no\n",
            "checking for suffix of object files... o\n",
            "checking whether the compiler supports GNU C... yes\n",
            "checking whether gcc accepts -g... yes\n",
            "checking for gcc option to enable C11 features... none needed\n",
            "checking how to run the C preprocessor... gcc -E\n",
            "Version has a libidn field\n",
            "Version has CURLOPT_URL\n",
            "Version has CURLINFO_EFFECTIVE_URL\n",
            "Version has CURLINFO_RESPONSE_CODE\n",
            "Version has CURLINFO_TOTAL_TIME\n",
            "Version has CURLINFO_NAMELOOKUP_TIME\n",
            "Version has CURLINFO_CONNECT_TIME\n",
            "Version has CURLINFO_PRETRANSFER_TIME\n",
            "Version has CURLINFO_SIZE_UPLOAD\n",
            "Version has CURLINFO_SIZE_DOWNLOAD\n",
            "Version has CURLINFO_SPEED_DOWNLOAD\n",
            "Version has CURLINFO_SPEED_UPLOAD\n",
            "Version has CURLINFO_HEADER_SIZE\n",
            "Version has CURLINFO_REQUEST_SIZE\n",
            "Version has CURLINFO_SSL_VERIFYRESULT\n",
            "Version has CURLINFO_FILETIME\n",
            "Version has CURLINFO_CONTENT_LENGTH_DOWNLOAD\n",
            "Version has CURLINFO_CONTENT_LENGTH_UPLOAD\n",
            "Version has CURLINFO_STARTTRANSFER_TIME\n",
            "Version has CURLINFO_CONTENT_TYPE\n",
            "Version has CURLINFO_REDIRECT_TIME\n",
            "Version has CURLINFO_REDIRECT_COUNT\n",
            "Version has CURLINFO_PRIVATE\n",
            "Version has CURLINFO_HTTP_CONNECTCODE\n",
            "Version has CURLINFO_HTTPAUTH_AVAIL\n",
            "Version has CURLINFO_PROXYAUTH_AVAIL\n",
            "Version has CURLINFO_OS_ERRNO\n",
            "Version has CURLINFO_NUM_CONNECTS\n",
            "Version has CURLINFO_SSL_ENGINES\n",
            "Version has CURLINFO_COOKIELIST\n",
            "Version has CURLINFO_LASTSOCKET\n",
            "Version has CURLINFO_FTP_ENTRY_PATH\n",
            "Version has CURLINFO_REDIRECT_URL\n",
            "Version has CURLINFO_PRIMARY_IP\n",
            "Version has CURLINFO_APPCONNECT_TIME\n",
            "Version has CURLINFO_CERTINFO\n",
            "Version has CURLINFO_CONDITION_UNMET\n",
            "Version has CURLOPT_WRITEDATA\n",
            "Version has CURLOPT_KEYPASSWD\n",
            "Version has CURLOPT_DIRLISTONLY\n",
            "Version has CURLOPT_APPEND\n",
            "Version has CURLOPT_KRBLEVEL\n",
            "Version has CURLOPT_USE_SSL\n",
            "Version has CURLOPT_TIMEOUT_MS\n",
            "Version has CURLOPT_CONNECTTIMEOUT_MS\n",
            "Version has CURLOPT_HTTP_TRANSFER_DECODING\n",
            "Version has CURLOPT_HTTP_CONTENT_DECODING\n",
            "Version has CURLOPT_NEW_FILE_PERMS\n",
            "Version has CURLOPT_NEW_DIRECTORY_PERMS\n",
            "Version has CURLOPT_POSTREDIR\n",
            "Version has CURLOPT_OPENSOCKETFUNCTION\n",
            "Version has CURLOPT_OPENSOCKETDATA\n",
            "Version has CURLOPT_COPYPOSTFIELDS\n",
            "Version has CURLOPT_PROXY_TRANSFER_MODE\n",
            "Version has CURLOPT_SEEKFUNCTION\n",
            "Version has CURLOPT_SEEKDATA\n",
            "Version has CURLOPT_CRLFILE\n",
            "Version has CURLOPT_ISSUERCERT\n",
            "Version has CURLOPT_ADDRESS_SCOPE\n",
            "Version has CURLOPT_CERTINFO\n",
            "Version has CURLOPT_USERNAME\n",
            "Version has CURLOPT_PASSWORD\n",
            "Version has CURLOPT_PROXYUSERNAME\n",
            "Version has CURLOPT_PROXYPASSWORD\n",
            "Version has CURLOPT_SSH_HOST_PUBLIC_KEY_MD5\n",
            "Version has CURLOPT_NOPROXY\n",
            "Version has CURLOPT_TFTP_BLKSIZE\n",
            "Version has CURLOPT_SOCKS5_GSSAPI_SERVICE\n",
            "Version has CURLOPT_SOCKS5_GSSAPI_NEC\n",
            "Version has CURLOPT_PROTOCOLS\n",
            "Version has CURLOPT_REDIR_PROTOCOLS\n",
            "Version has CURLOPT_SSH_AUTH_TYPES\n",
            "Version has CURLOPT_SSH_PUBLIC_KEYFILE\n",
            "Version has CURLOPT_SSH_PRIVATE_KEYFILE\n",
            "Version has CURLOPT_FTP_SSL_CCC\n",
            "Version has CURLOPT_COOKIELIST\n",
            "Version has CURLOPT_IGNORE_CONTENT_LENGTH\n",
            "Version has CURLOPT_FTP_SKIP_PASV_IP\n",
            "Version has CURLOPT_FTP_FILEMETHOD\n",
            "Version has CURLOPT_LOCALPORT\n",
            "Version has CURLOPT_LOCALPORTRANGE\n",
            "Version has CURLOPT_CONNECT_ONLY\n",
            "Version has CURLOPT_CONV_FROM_NETWORK_FUNCTION\n",
            "Version has CURLOPT_CONV_TO_NETWORK_FUNCTION\n",
            "Version has CURLOPT_CONV_FROM_UTF8_FUNCTION\n",
            "Version has CURLOPT_MAX_SEND_SPEED_LARGE\n",
            "Version has CURLOPT_MAX_RECV_SPEED_LARGE\n",
            "Version has CURLOPT_FTP_ALTERNATIVE_TO_USER\n",
            "Version has CURLOPT_SOCKOPTFUNCTION\n",
            "Version has CURLOPT_SOCKOPTDATA\n",
            "Version has CURLOPT_SSL_SESSIONID_CACHE\n",
            "libcurl version: libcurl 7.75.0\n",
            "configure: creating ./config.status\n",
            "config.status: creating src/Makevars\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c base64.c -o base64.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c curl.c -o curl.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c curlInit.c -o curlInit.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c curl_base64.c -o curl_base64.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c enums.c -o enums.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c json.c -o json.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c memoryManagement.c -o memoryManagement.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I/usr/local/include -DHAVE_LIBIDN_FIELD=1 -DHAVE_CURLOPT_URL=1 -DHAVE_CURLINFO_EFFECTIVE_URL=1 -DHAVE_CURLINFO_RESPONSE_CODE=1 -DHAVE_CURLINFO_TOTAL_TIME=1 -DHAVE_CURLINFO_NAMELOOKUP_TIME=1 -DHAVE_CURLINFO_CONNECT_TIME=1 -DHAVE_CURLINFO_PRETRANSFER_TIME=1 -DHAVE_CURLINFO_SIZE_UPLOAD=1 -DHAVE_CURLINFO_SIZE_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_DOWNLOAD=1 -DHAVE_CURLINFO_SPEED_UPLOAD=1 -DHAVE_CURLINFO_HEADER_SIZE=1 -DHAVE_CURLINFO_REQUEST_SIZE=1 -DHAVE_CURLINFO_SSL_VERIFYRESULT=1 -DHAVE_CURLINFO_FILETIME=1 -DHAVE_CURLINFO_CONTENT_LENGTH_DOWNLOAD=1 -DHAVE_CURLINFO_CONTENT_LENGTH_UPLOAD=1 -DHAVE_CURLINFO_STARTTRANSFER_TIME=1 -DHAVE_CURLINFO_CONTENT_TYPE=1 -DHAVE_CURLINFO_REDIRECT_TIME=1 -DHAVE_CURLINFO_REDIRECT_COUNT=1 -DHAVE_CURLINFO_PRIVATE=1 -DHAVE_CURLINFO_HTTP_CONNECTCODE=1 -DHAVE_CURLINFO_HTTPAUTH_AVAIL=1 -DHAVE_CURLINFO_PROXYAUTH_AVAIL=1 -DHAVE_CURLINFO_OS_ERRNO=1 -DHAVE_CURLINFO_NUM_CONNECTS=1 -DHAVE_CURLINFO_SSL_ENGINES=1 -DHAVE_CURLINFO_COOKIELIST=1 -DHAVE_CURLINFO_LASTSOCKET=1 -DHAVE_CURLINFO_FTP_ENTRY_PATH=1 -DHAVE_CURLINFO_REDIRECT_URL=1 -DHAVE_CURLINFO_PRIMARY_IP=1 -DHAVE_CURLINFO_APPCONNECT_TIME=1 -DHAVE_CURLINFO_CERTINFO=1 -DHAVE_CURLINFO_CONDITION_UNMET=1 -DHAVE_CURLOPT_WRITEDATA=1 -DHAVE_CURLOPT_KEYPASSWD=1 -DHAVE_CURLOPT_DIRLISTONLY=1 -DHAVE_CURLOPT_APPEND=1 -DHAVE_CURLOPT_KRBLEVEL=1 -DHAVE_CURLOPT_USE_SSL=1 -DHAVE_CURLOPT_TIMEOUT_MS=1 -DHAVE_CURLOPT_CONNECTTIMEOUT_MS=1 -DHAVE_CURLOPT_HTTP_TRANSFER_DECODING=1 -DHAVE_CURLOPT_HTTP_CONTENT_DECODING=1 -DHAVE_CURLOPT_NEW_FILE_PERMS=1 -DHAVE_CURLOPT_NEW_DIRECTORY_PERMS=1 -DHAVE_CURLOPT_POSTREDIR=1 -DHAVE_CURLOPT_OPENSOCKETFUNCTION=1 -DHAVE_CURLOPT_OPENSOCKETDATA=1 -DHAVE_CURLOPT_COPYPOSTFIELDS=1 -DHAVE_CURLOPT_PROXY_TRANSFER_MODE=1 -DHAVE_CURLOPT_SEEKFUNCTION=1 -DHAVE_CURLOPT_SEEKDATA=1 -DHAVE_CURLOPT_CRLFILE=1 -DHAVE_CURLOPT_ISSUERCERT=1 -DHAVE_CURLOPT_ADDRESS_SCOPE=1 -DHAVE_CURLOPT_CERTINFO=1 -DHAVE_CURLOPT_USERNAME=1 -DHAVE_CURLOPT_PASSWORD=1 -DHAVE_CURLOPT_PROXYUSERNAME=1 -DHAVE_CURLOPT_PROXYPASSWORD=1 -DHAVE_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5=1 -DHAVE_CURLOPT_NOPROXY=1 -DHAVE_CURLOPT_TFTP_BLKSIZE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_SERVICE=1 -DHAVE_CURLOPT_SOCKS5_GSSAPI_NEC=1 -DHAVE_CURLOPT_PROTOCOLS=1 -DHAVE_CURLOPT_REDIR_PROTOCOLS=1 -DHAVE_CURLOPT_SSH_AUTH_TYPES=1 -DHAVE_CURLOPT_SSH_PUBLIC_KEYFILE=1 -DHAVE_CURLOPT_SSH_PRIVATE_KEYFILE=1 -DHAVE_CURLOPT_FTP_SSL_CCC=1 -DHAVE_CURLOPT_COOKIELIST=1 -DHAVE_CURLOPT_IGNORE_CONTENT_LENGTH=1 -DHAVE_CURLOPT_FTP_SKIP_PASV_IP=1 -DHAVE_CURLOPT_FTP_FILEMETHOD=1 -DHAVE_CURLOPT_LOCALPORT=1 -DHAVE_CURLOPT_LOCALPORTRANGE=1 -DHAVE_CURLOPT_CONNECT_ONLY=1 -DHAVE_CURLOPT_CONV_FROM_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_TO_NETWORK_FUNCTION=1 -DHAVE_CURLOPT_CONV_FROM_UTF8_FUNCTION=1 -DHAVE_CURLOPT_MAX_SEND_SPEED_LARGE=1 -DHAVE_CURLOPT_MAX_RECV_SPEED_LARGE=1 -DHAVE_CURLOPT_FTP_ALTERNATIVE_TO_USER=1 -DHAVE_CURLOPT_SOCKOPTFUNCTION=1 -DHAVE_CURLOPT_SOCKOPTDATA=1 -DHAVE_CURLOPT_SSL_SESSIONID_CACHE=1 -I/usr/local/include/libxml2 -I/usr/local/include     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c myUTF8.c -o myUTF8.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o RCurl.so base64.o curl.o curlInit.o curl_base64.o enums.o json.o memoryManagement.o myUTF8.o -L/usr/local/lib -lcurl -L/usr/local/lib -lxml2 -L/usr/local/lib -lz -L/usr/local/lib -llzma -lpthread -L/usr/local/lib -liconv -L/usr/local/lib -licui18n -licuuc -licudata -lm -ldl -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-RCurl/00new/RCurl/libs\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (RCurl)\n",
            "* installing *source* package ‘lambda.r’ ...\n",
            "** package ‘lambda.r’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (lambda.r)\n",
            "* installing *source* package ‘RSQLite’ ...\n",
            "** package ‘RSQLite’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DbColumn.cpp -o DbColumn.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DbColumnDataSource.cpp -o DbColumnDataSource.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DbColumnDataSourceFactory.cpp -o DbColumnDataSourceFactory.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DbColumnStorage.cpp -o DbColumnStorage.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DbConnection.cpp -o DbConnection.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DbDataFrame.cpp -o DbDataFrame.o\n",
            "In file included from \u001b[01m\u001b[KDbDataFrame.cpp:7:0\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Kvendor/boost/bind.hpp:41:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.\n",
            " \u001b[01;36m\u001b[K)\u001b[m\u001b[K\n",
            " \u001b[01;36m\u001b[K^\u001b[m\u001b[K\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DbResult.cpp -o DbResult.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RcppExports.cpp -o RcppExports.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SqliteColumnDataSource.cpp -o SqliteColumnDataSource.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SqliteColumnDataSourceFactory.cpp -o SqliteColumnDataSourceFactory.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SqliteDataFrame.cpp -o SqliteDataFrame.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SqliteResult.cpp -o SqliteResult.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SqliteResultImpl.cpp -o SqliteResultImpl.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c affinity.c -o affinity.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c connection.cpp -o connection.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ext-csv.c -o ext-csv.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ext-math.c -o ext-math.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ext-regexp.c -o ext-regexp.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ext-series.c -o ext-series.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c extensions.cpp -o extensions.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c import-file.c -o import-file.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c result.cpp -o result.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rsqlite.cpp -o rsqlite.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ivendor -Ivendor/extensions -Ivendor/sqlite3 -DRSQLITE_USE_BUNDLED_SQLITE -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_STAT4 -DSQLITE_SOUNDEX -DSQLITE_USE_URI=1 -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR -DSQLITE_MAX_LENGTH=2147483647 -DHAVE_USLEEP=1 -I'/usr/local/lib/R/site-library/plogr/include' -I'/usr/lib/R/site-library/Rcpp/include'   -fvisibility=hidden -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c vendor/sqlite3/sqlite3.c -o vendor/sqlite3/sqlite3.o\n",
            "g++ -std=gnu++14 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o RSQLite.so DbColumn.o DbColumnDataSource.o DbColumnDataSourceFactory.o DbColumnStorage.o DbConnection.o DbDataFrame.o DbResult.o RcppExports.o SqliteColumnDataSource.o SqliteColumnDataSourceFactory.o SqliteDataFrame.o SqliteResult.o SqliteResultImpl.o affinity.o connection.o ext-csv.o ext-math.o ext-regexp.o ext-series.o extensions.o import-file.o result.o rsqlite.o vendor/sqlite3/sqlite3.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-RSQLite/00new/RSQLite/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (RSQLite)\n",
            "* installing *source* package ‘MatrixGenerics’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (MatrixGenerics)\n",
            "* installing *source* package ‘S4Vectors’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c AEbufs.c -o AEbufs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DataFrame_class.c -o DataFrame_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Hits_class.c -o Hits_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c LLint_class.c -o LLint_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c List_class.c -o List_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c R_init_S4Vectors.c -o R_init_S4Vectors.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Rle_class.c -o Rle_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Rle_utils.c -o Rle_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SEXP_utils.c -o SEXP_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SimpleList_class.c -o SimpleList_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c anyMissing.c -o anyMissing.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c character_utils.c -o character_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c eval_utils.c -o eval_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c hash_utils.c -o hash_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c integer_utils.c -o integer_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c logical_utils.c -o logical_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c map_ranges_to_runs.c -o map_ranges_to_runs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c raw_utils.c -o raw_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c safe_arithm.c -o safe_arithm.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sort_utils.c -o sort_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c subsetting_utils.c -o subsetting_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c vector_utils.c -o vector_utils.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o S4Vectors.so AEbufs.o DataFrame_class.o Hits_class.o LLint_class.o List_class.o R_init_S4Vectors.o Rle_class.o Rle_utils.o SEXP_utils.o SimpleList_class.o anyMissing.o character_utils.o eval_utils.o hash_utils.o integer_utils.o logical_utils.o map_ranges_to_runs.o raw_utils.o safe_arithm.o sort_utils.o subsetting_utils.o vector_utils.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-S4Vectors/00new/S4Vectors/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "in method for ‘normalizeSingleBracketReplacementValue’ with signature ‘\"List\"’: no definition for class “List”\n",
            "Creating a new generic function for ‘unname’ in package ‘S4Vectors’\n",
            "Creating a new generic function for ‘expand.grid’ in package ‘S4Vectors’\n",
            "Creating a new generic function for ‘findMatches’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘setequal’ from package ‘base’ in package ‘S4Vectors’\n",
            "in method for ‘coerce’ with signature ‘\"Hits\",\"DFrame\"’: no definition for class “DFrame”\n",
            "Creating a generic function for ‘as.factor’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘tabulate’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘cov’ from package ‘stats’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘cor’ from package ‘stats’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘smoothEnds’ from package ‘stats’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘runmed’ from package ‘stats’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘nchar’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘substr’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘substring’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘chartr’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘tolower’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘toupper’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘sub’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘gsub’ from package ‘base’ in package ‘S4Vectors’\n",
            "Creating a generic function for ‘nlevels’ from package ‘base’ in package ‘S4Vectors’\n",
            "in method for ‘coerce’ with signature ‘\"data.table\",\"DFrame\"’: no definition for class “data.table”\n",
            "Creating a generic function for ‘complete.cases’ from package ‘stats’ in package ‘S4Vectors’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (S4Vectors)\n",
            "* installing *source* package ‘Biobase’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Rinit.c -o Rinit.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c anyMissing.c -o anyMissing.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c envir.c -o envir.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c matchpt.c -o matchpt.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowMedians.c -o rowMedians.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sublist_extract.c -o sublist_extract.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o Biobase.so Rinit.o anyMissing.o envir.o matchpt.o rowMedians.o sublist_extract.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-Biobase/00new/Biobase/libs\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (Biobase)\n",
            "* installing *source* package ‘futile.logger’ ...\n",
            "** package ‘futile.logger’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (futile.logger)\n",
            "* installing *source* package ‘IRanges’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c CompressedAtomicList_utils.c -o CompressedAtomicList_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c CompressedIRangesList_class.c -o CompressedIRangesList_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c CompressedList_class.c -o CompressedList_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Grouping_class.c -o Grouping_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c IPosRanges_comparison.c -o IPosRanges_comparison.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c IRanges_class.c -o IRanges_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c IRanges_constructor.c -o IRanges_constructor.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c NCList.c -o NCList.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c R_init_IRanges.c -o R_init_IRanges.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Ranges_class.c -o Ranges_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RleViews_utils.c -o RleViews_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SimpleIRangesList_class.c -o SimpleIRangesList_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c coverage_methods.c -o coverage_methods.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c extractListFragments.c -o extractListFragments.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c inter_range_methods.c -o inter_range_methods.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o IRanges.so CompressedAtomicList_utils.o CompressedIRangesList_class.o CompressedList_class.o Grouping_class.o IPosRanges_comparison.o IRanges_class.o IRanges_constructor.o NCList.o R_init_IRanges.o Ranges_class.o RleViews_utils.o S4Vectors_stubs.o SimpleIRangesList_class.o coverage_methods.o extractListFragments.o inter_range_methods.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-IRanges/00new/IRanges/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Creating a generic function for ‘drop’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘runmed’ from package ‘stats’ in package ‘IRanges’\n",
            "Creating a generic function for ‘chartr’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘toupper’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘tolower’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘sub’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘gsub’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘startsWith’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘endsWith’ from package ‘base’ in package ‘IRanges’\n",
            "Creating a generic function for ‘smoothEnds’ from package ‘stats’ in package ‘IRanges’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (IRanges)\n",
            "* installing *source* package ‘GenomeInfoDb’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (GenomeInfoDb)\n",
            "* installing *source* package ‘XVector’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c IRanges_stubs.c -o IRanges_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Ocopy_byteblocks.c -o Ocopy_byteblocks.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RDS_random_access.c -o RDS_random_access.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c R_init_XVector.c -o R_init_XVector.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SharedDouble_class.c -o SharedDouble_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SharedInteger_class.c -o SharedInteger_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SharedRaw_class.c -o SharedRaw_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SharedVector_class.c -o SharedVector_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XDouble_class.c -o XDouble_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XInteger_class.c -o XInteger_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XRawList_comparison.c -o XRawList_comparison.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XRaw_class.c -o XRaw_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XVectorList_class.c -o XVectorList_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XVector_class.c -o XVector_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c io_utils.c -o io_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c slice_methods.c -o slice_methods.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c vector_copy.c -o vector_copy.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c view_summarization_methods.c -o view_summarization_methods.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o XVector.so IRanges_stubs.o Ocopy_byteblocks.o RDS_random_access.o R_init_XVector.o S4Vectors_stubs.o SharedDouble_class.o SharedInteger_class.o SharedRaw_class.o SharedVector_class.o XDouble_class.o XInteger_class.o XRawList_comparison.o XRaw_class.o XVectorList_class.o XVector_class.o io_utils.o slice_methods.o vector_copy.o view_summarization_methods.o -lz -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-XVector/00new/XVector/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (XVector)\n",
            "* installing *source* package ‘DelayedArray’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c R_init_DelayedArray.c -o R_init_DelayedArray.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c abind.c -o abind.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c array_selection.c -o array_selection.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c compress_atomic_vector.c -o compress_atomic_vector.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sparseMatrix_utils.c -o sparseMatrix_utils.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o DelayedArray.so R_init_DelayedArray.o S4Vectors_stubs.o abind.o array_selection.o compress_atomic_vector.o sparseMatrix_utils.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-DelayedArray/00new/DelayedArray/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Creating a new generic function for ‘rowsum’ in package ‘DelayedArray’\n",
            "Creating a new generic function for ‘aperm’ in package ‘DelayedArray’\n",
            "Creating a new generic function for ‘apply’ in package ‘DelayedArray’\n",
            "Creating a new generic function for ‘sweep’ in package ‘DelayedArray’\n",
            "Creating a new generic function for ‘scale’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dnorm’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘pnorm’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qnorm’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dbinom’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘pbinom’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qbinom’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dpois’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘ppois’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qpois’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dlogis’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘plogis’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qlogis’ from package ‘stats’ in package ‘DelayedArray’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (DelayedArray)\n",
            "* installing *source* package ‘BiocParallel’ ...\n",
            "** using staged installation\n",
            "checking for gcc... gcc\n",
            "checking whether the C compiler works... yes\n",
            "checking for C compiler default output file name... a.out\n",
            "checking for suffix of executables... \n",
            "checking whether we are cross compiling... no\n",
            "checking for suffix of object files... o\n",
            "checking whether we are using the GNU C compiler... yes\n",
            "checking whether gcc accepts -g... yes\n",
            "checking for gcc option to accept ISO C89... none needed\n",
            "checking for shm_open in -lrt... yes\n",
            "configure: creating ./config.status\n",
            "config.status: creating src/Makevars\n",
            "** libs\n",
            "g++ -std=gnu++11 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/BH/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ipcmutex.cpp -o ipcmutex.o\n",
            "g++ -std=gnu++11 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o BiocParallel.so ipcmutex.o -lrt -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-BiocParallel/00new/BiocParallel/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (BiocParallel)\n",
            "* installing *source* package ‘Biostrings’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c BAB_class.c -o BAB_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c BitMatrix.c -o BitMatrix.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c IRanges_stubs.c -o IRanges_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c MIndex_class.c -o MIndex_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c PreprocessedTB_class.c -o PreprocessedTB_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c R_init_Biostrings.c -o R_init_Biostrings.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RoSeqs_utils.c -o RoSeqs_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SparseList_utils.c -o SparseList_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XStringSetList_class.c -o XStringSetList_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XStringSet_class.c -o XStringSet_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XString_class.c -o XString_class.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c XVector_stubs.c -o XVector_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c align_needwunsQS.c -o align_needwunsQS.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c align_pairwiseAlignment.c -o align_pairwiseAlignment.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c align_utils.c -o align_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c find_palindromes.c -o find_palindromes.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c gtestsim.c -o gtestsim.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c inject_code.c -o inject_code.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c letter_frequency.c -o letter_frequency.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c lowlevel_matching.c -o lowlevel_matching.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_PWM.c -o match_PWM.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pattern.c -o match_pattern.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pattern_boyermoore.c -o match_pattern_boyermoore.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pattern_indels.c -o match_pattern_indels.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pattern_shiftor.c -o match_pattern_shiftor.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pdict.c -o match_pdict.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pdict_ACtree2.c -o match_pdict_ACtree2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pdict_Twobit.c -o match_pdict_Twobit.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_pdict_utils.c -o match_pdict_utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c match_reporting.c -o match_reporting.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c matchprobes.c -o matchprobes.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c pmatchPattern.c -o pmatchPattern.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c read_fasta_files.c -o read_fasta_files.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c read_fastq_files.c -o read_fastq_files.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c replaceAt.c -o replaceAt.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c replace_letter_at.c -o replace_letter_at.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c strutils.c -o strutils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c translate.c -o translate.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c unstrsplit_methods.c -o unstrsplit_methods.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c utils.c -o utils.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c xscat.c -o xscat.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o Biostrings.so BAB_class.o BitMatrix.o IRanges_stubs.o MIndex_class.o PreprocessedTB_class.o R_init_Biostrings.o RoSeqs_utils.o S4Vectors_stubs.o SparseList_utils.o XStringSetList_class.o XStringSet_class.o XString_class.o XVector_stubs.o align_needwunsQS.o align_pairwiseAlignment.o align_utils.o find_palindromes.o gtestsim.o inject_code.o letter_frequency.o lowlevel_matching.o match_PWM.o match_pattern.o match_pattern_boyermoore.o match_pattern_indels.o match_pattern_shiftor.o match_pdict.o match_pdict_ACtree2.o match_pdict_Twobit.o match_pdict_utils.o match_reporting.o matchprobes.o pmatchPattern.o read_fasta_files.o read_fastq_files.o replaceAt.o replace_letter_at.o strutils.o translate.o unstrsplit_methods.o utils.o xscat.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-Biostrings/00new/Biostrings/libs\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "in method for ‘twoWayAlphabetFrequencyByQuality’ with signature ‘\"QualityScaledXStringSet\"’: no definition for class “QualityScaledXStringSet”\n",
            "Creating a new generic function for ‘strsplit’ in package ‘Biostrings’\n",
            "Creating a generic function for ‘ls’ from package ‘base’ in package ‘Biostrings’\n",
            "Creating a new generic function for ‘pattern’ in package ‘Biostrings’\n",
            "Creating a new generic function for ‘offset’ in package ‘Biostrings’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (Biostrings)\n",
            "* installing *source* package ‘GenomicRanges’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c IRanges_stubs.c -o IRanges_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c R_init_GenomicRanges.c -o R_init_GenomicRanges.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c transcript_utils.c -o transcript_utils.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o GenomicRanges.so IRanges_stubs.o R_init_GenomicRanges.o S4Vectors_stubs.o transcript_utils.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-GenomicRanges/00new/GenomicRanges/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (GenomicRanges)\n",
            "* installing *source* package ‘KEGGREST’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (KEGGREST)\n",
            "* installing *source* package ‘SummarizedExperiment’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (SummarizedExperiment)\n",
            "* installing *source* package ‘AnnotationDbi’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Creating a new generic function for ‘packageName’ in package ‘AnnotationDbi’\n",
            "Creating a generic function for ‘ls’ from package ‘base’ in package ‘AnnotationDbi’\n",
            "Creating a generic function for ‘eapply’ from package ‘base’ in package ‘AnnotationDbi’\n",
            "Creating a generic function for ‘exists’ from package ‘base’ in package ‘AnnotationDbi’\n",
            "Creating a generic function for ‘sample’ from package ‘base’ in package ‘AnnotationDbi’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (AnnotationDbi)\n",
            "* installing *source* package ‘annotate’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (annotate)\n",
            "* installing *source* package ‘genefilter’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c half_range_mode.cpp -o half_range_mode.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.c -o init.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c nd.c -o nd.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c pAUC.c -o pAUC.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowPAUCs.c -o rowPAUCs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rowttests.c -o rowttests.o\n",
            "gfortran -fno-optimize-sibling-calls  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong  -c ttest.f -o ttest.o\n",
            "g++ -std=gnu++14 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o genefilter.so half_range_mode.o init.o nd.o pAUC.o rowPAUCs.o rowttests.o ttest.o -lgfortran -lm -lquadmath -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-genefilter/00new/genefilter/libs\n",
            "** R\n",
            "** data\n",
            "*** moving datasets to lazyload DB\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (genefilter)\n",
            "* installing *source* package ‘geneplotter’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (geneplotter)\n",
            "* installing *source* package ‘DESeq2’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/Rcpp/include' -I'/usr/local/lib/R/site-library/RcppArmadillo/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c DESeq2.cpp -o DESeq2.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/Rcpp/include' -I'/usr/local/lib/R/site-library/RcppArmadillo/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RcppExports.cpp -o RcppExports.o\n",
            "g++ -std=gnu++14 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o DESeq2.so DESeq2.o RcppExports.o -llapack -lblas -lgfortran -lm -lquadmath -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-DESeq2/00new/DESeq2/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (DESeq2)\n",
            "\n",
            "The downloaded source packages are in\n",
            "\t‘/tmp/RtmpBtSBUQ/downloaded_packages’\n",
            "Old packages: 'crayon', 'fansi', 'gert', 'knitr', 'magrittr', 'processx',\n",
            "  'readxl', 'sass', 'testthat', 'tinytex', 'tzdb', 'vctrs', 'cluster', 'MASS',\n",
            "  'Matrix', 'mgcv', 'nlme'\n",
            "> library(DESeq2)\n",
            "Loading required package: S4Vectors\n",
            "Loading required package: stats4\n",
            "Loading required package: BiocGenerics\n",
            "\n",
            "Attaching package: ‘BiocGenerics’\n",
            "\n",
            "The following objects are masked from ‘package:stats’:\n",
            "\n",
            "    IQR, mad, sd, var, xtabs\n",
            "\n",
            "The following objects are masked from ‘package:base’:\n",
            "\n",
            "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
            "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
            "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
            "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
            "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
            "    union, unique, unsplit, which.max, which.min\n",
            "\n",
            "\n",
            "Attaching package: ‘S4Vectors’\n",
            "\n",
            "The following objects are masked from ‘package:base’:\n",
            "\n",
            "    expand.grid, I, unname\n",
            "\n",
            "Loading required package: IRanges\n",
            "Loading required package: GenomicRanges\n",
            "Loading required package: GenomeInfoDb\n",
            "Loading required package: SummarizedExperiment\n",
            "Loading required package: MatrixGenerics\n",
            "Loading required package: matrixStats\n",
            "\n",
            "Attaching package: ‘MatrixGenerics’\n",
            "\n",
            "The following objects are masked from ‘package:matrixStats’:\n",
            "\n",
            "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
            "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
            "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
            "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
            "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
            "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
            "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
            "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
            "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
            "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
            "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
            "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
            "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
            "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
            "    rowWeightedSds, rowWeightedVars\n",
            "\n",
            "Loading required package: Biobase\n",
            "Welcome to Bioconductor\n",
            "\n",
            "    Vignettes contain introductory material; view with\n",
            "    'browseVignettes()'. To cite Bioconductor, see\n",
            "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
            "\n",
            "\n",
            "Attaching package: ‘Biobase’\n",
            "\n",
            "The following object is masked from ‘package:MatrixGenerics’:\n",
            "\n",
            "    rowMedians\n",
            "\n",
            "The following objects are masked from ‘package:matrixStats’:\n",
            "\n",
            "    anyMissing, rowMedians\n",
            "\n",
            "\u001b[?25h> \n",
            "> # Set up the conditions based on the experimental setup.\n",
            "> cond_1 = rep(\"cond1\", 2)\n",
            "\u001b[?25h> cond_2 = rep(\"cond2\", 2)\n",
            "\u001b[?25h> \n",
            "> # Read the data from the standard input.\n",
            "> countData = read.table(\"stdin\", header=TRUE, sep=\"\\t\", row.names=1 )\n",
            "\u001b[?25h> \n",
            "> # Build the dataframe from the conditions\n",
            "> samples = names(countData)\n",
            "\u001b[?25h> condition = factor(c(cond_1, cond_2))\n",
            "\u001b[?25h> colData = data.frame(samples=samples, condition=condition)\n",
            "\u001b[?25h> \n",
            "> # Create DESEq2 dataset.\n",
            "> dds = DESeqDataSetFromMatrix(countData=countData, colData=colData, design = ~condition)\n",
            "\u001b[?25h> \n",
            "> #Set the reference to be compared\n",
            "> dds$condition = relevel(dds$condition,\"cond1\")\n",
            "\u001b[?25h> \n",
            "> # Run deseq\n",
            "> dds = DESeq(dds)\n",
            "estimating size factors\n",
            "estimating dispersions\n",
            "gene-wise dispersion estimates\n",
            "mean-dispersion relationship\n",
            "final dispersion estimates\n",
            "fitting model and testing\n",
            "\u001b[?25h> \n",
            "> # Format the results.\n",
            "> res = results(dds)\n",
            "\u001b[?25h> \n",
            "> # Sort the results data frame by the padj and foldChange columns.\n",
            "> sorted = res[with(res, order(padj, -log2FoldChange)), ]\n",
            "\u001b[?25h> \n",
            "> # Turn it into a dataframe to have proper column names.\n",
            "> sorted.df = data.frame(\"id\"=rownames(sorted),sorted)\n",
            "\u001b[?25h> \n",
            "> # Write the table out.\n",
            "> write.table(sorted.df, file=\"result.txt\", sep=\"\\t\", col.names=NA, quote=FALSE)\n",
            "\u001b[?25h> \n",
            "> # Get normalized counts and write this to a file\n",
            "> nc = counts(dds,normalized=TRUE)\n",
            "\u001b[?25h> \n",
            "> # Turn it into a dataframe to have proper column names.\n",
            "> dt = data.frame(\"id\"=rownames(nc),nc)\n",
            "\u001b[?25h> \n",
            "> # Save the normalize data matrix.\n",
            "> write.table(dt, file=\"norm-matrix-deseq2.txt\", sep=\"\\t\",  row.name=FALSE, col.names=TRUE,quote=FALSE)\n",
            "\u001b[?25h> \n",
            "> \n",
            "> \n",
            "\u001b[?25h"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "Draw heatmap:<br>\n",
        "\n",
        "Added line `install.packages(\"gplots\")` to r script."
      ],
      "metadata": {
        "id": "SQYsGcz9G7AX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cat norm-matrix-deseq2.txt | R -f draw-heatmap.r"
      ],
      "metadata": {
        "id": "_ObAdCryD44z",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4ee0fa32-202b-46d5-bc03-6044a24457e1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "R version 4.1.3 (2022-03-10) -- \"One Push-Up\"\n",
            "Copyright (C) 2022 The R Foundation for Statistical Computing\n",
            "Platform: x86_64-pc-linux-gnu (64-bit)\n",
            "\n",
            "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
            "You are welcome to redistribute it under certain conditions.\n",
            "Type 'license()' or 'licence()' for distribution details.\n",
            "\n",
            "  Natural language support but running in an English locale\n",
            "\n",
            "R is a collaborative project with many contributors.\n",
            "Type 'contributors()' for more information and\n",
            "'citation()' on how to cite R or R packages in publications.\n",
            "\n",
            "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
            "'help.start()' for an HTML browser interface to help.\n",
            "Type 'q()' to quit R.\n",
            "\n",
            "> install.packages(\"gplots\")\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "also installing the dependencies ‘gtools’, ‘caTools’\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/gtools_3.9.2.tar.gz'\n",
            "Content type 'application/x-gzip' length 240402 bytes (234 KB)\n",
            "==================================================\n",
            "downloaded 234 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/caTools_1.18.2.tar.gz'\n",
            "Content type 'application/x-gzip' length 64960 bytes (63 KB)\n",
            "==================================================\n",
            "downloaded 63 KB\n",
            "\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/gplots_3.1.1.tar.gz'\n",
            "Content type 'application/x-gzip' length 520248 bytes (508 KB)\n",
            "==================================================\n",
            "downloaded 508 KB\n",
            "\n",
            "* installing *source* package ‘gtools’ ...\n",
            "** package ‘gtools’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.c -o init.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c roman2int.c -o roman2int.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c setTCPNoDelay.c -o setTCPNoDelay.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o gtools.so init.o roman2int.o setTCPNoDelay.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-gtools/00new/gtools/libs\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "*** copying figures\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (gtools)\n",
            "* installing *source* package ‘caTools’ ...\n",
            "** package ‘caTools’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c Gif2R.cpp -o Gif2R.o\n",
            "g++ -std=gnu++14 -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c GifTools.cpp -o GifTools.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c caTools_init.c -o caTools_init.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-NlA7dw/r-base-4.1.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c runfunc.c -o runfunc.o\n",
            "g++ -std=gnu++14 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o caTools.so Gif2R.o GifTools.o caTools_init.o runfunc.o -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-caTools/00new/caTools/libs\n",
            "** R\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (caTools)\n",
            "* installing *source* package ‘gplots’ ...\n",
            "** package ‘gplots’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (gplots)\n",
            "\n",
            "The downloaded source packages are in\n",
            "\t‘/tmp/Rtmpjui1dA/downloaded_packages’\n",
            "> library(gplots)\n",
            "\n",
            "Attaching package: ‘gplots’\n",
            "\n",
            "The following object is masked from ‘package:stats’:\n",
            "\n",
            "    lowess\n",
            "\n",
            "> \n",
            "> # Read normalized counts\n",
            "> data = read.table(\"stdin\", header=T, sep=\"\\t\", as.is=TRUE)\n",
            "> \n",
            "> gene = data[,1]\n",
            "> vals = as.matrix(data[,2:ncol(data)])\n",
            "> \n",
            "> # Adds a little noise to each element\n",
            "> # To avoid the clusteing function failing on zero\n",
            "> # variance datalines.\n",
            "> vals = jitter(vals, factor = 1, amount=0.00001)\n",
            "> \n",
            "> \n",
            "> # Calculate zscore\n",
            "> score = NULL\n",
            "> for (i in 1:nrow(vals)) {\n",
            "+     row=vals[i,]\n",
            "+     zscore=(row-mean(row))/sd(row)\n",
            "+     score =rbind(score,zscore)\n",
            "+ }\n",
            "> \n",
            "> row.names(score) = gene\n",
            "> zscore=score\n",
            "> \n",
            "> # Generate heatmap\n",
            "> mat = as.matrix(zscore)\n",
            "> \n",
            "> # Opent the drawing device.\n",
            "> pdf('output.pdf')\n",
            "> \n",
            "> colors = colorRampPalette(c(\"green\",\"black\",\"red\"),space=\"rgb\")(256)\n",
            "> heatmap.2(mat,col=colors,density.info=\"none\",trace=\"none\", margins=c(14,14),lhei=c(1,5))\n",
            "> \n",
            "> invisible(dev.off())\n",
            "> \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "Results:\n",
        "\n",
        "<img src='https://sun9-64.userapi.com/impg/P40aGXY8155P2Z2QP5WVJA8jm6_d8kM7c-VKrw/XPon2epkIiA.jpg?size=2100x2100&quality=96&sign=169e8842bb7a84fe84b84c7f21ae5737&type=album'>"
      ],
      "metadata": {
        "id": "FK1ErYMpLtIM"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "As we can see, some genes are highly expressed at the beginning (fermentation 0 minutes), and some others $-$ after a while (fermentation 30 minutes)."
      ],
      "metadata": {
        "id": "KkrcnArMNp_U"
      }
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "irB8Xo8JD4-d"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 3. Result Interpretation"
      ],
      "metadata": {
        "id": "n0tNtiIxOH96"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "In the result.txt file genes are sorted by adjusted p-values. So let’s take the first 50 genes from this file and keep only the first column (gene names):"
      ],
      "metadata": {
        "id": "e3gwrcjAOOfn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# original file\n",
        "!head result.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "20tlK88ROF5e",
        "outputId": "e1773279-acd3-49f7-969f-9d6a1aed0f2f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\tid\tbaseMean\tlog2FoldChange\tlfcSE\tstat\tpvalue\tpadj\n",
            "gene-YER062C\tgene-YER062C\t16724.95364409\t7.90041845725789\t0.408107183750566\t19.358685099958\t1.72225588408877e-83\t1.00476408277739e-79\n",
            "gene-YDR536W\tgene-YDR536W\t10085.4233999349\t7.87460213636111\t0.411383685208295\t19.1417463052138\t1.13404730413716e-81\t3.30801598616809e-78\n",
            "gene-YHR094C\tgene-YHR094C\t8492.75662329135\t7.88194601045555\t0.431898583145457\t18.2495296767413\t2.08698225835359e-74\t4.05848483174495e-71\n",
            "gene-YNL065W\tgene-YNL065W\t2109.04523230287\t7.78577989127413\t0.466810022269479\t16.6786905161594\t1.87279280460424e-62\t2.73146830551528e-59\n",
            "gene-YKL120W\tgene-YKL120W\t1726.73882317998\t7.26978739731226\t0.48234065089367\t15.0718944875224\t2.47901234668911e-51\t2.89251160611686e-48\n",
            "gene-YLR264W\tgene-YLR264W\t1387.41861770825\t6.48365302083557\t0.431993252552932\t15.0086904888431\t6.44067159244521e-51\t6.26247967838756e-48\n",
            "gene-YGR159C\tgene-YGR159C\t8019.03595899418\t6.76797463128207\t0.45156427420644\t14.9878434098353\t8.81699182812526e-51\t6.42979129066035e-48\n",
            "gene-YJL122W\tgene-YJL122W\t1290.7757807773\t6.51210204690814\t0.434391050899508\t14.991335648889\t8.36538677600959e-51\t6.42979129066035e-48\n",
            "gene-YNL112W\tgene-YNL112W\t5327.50114671469\t7.36457765745516\t0.493276497924626\t14.9299179840117\t2.10524877707797e-50\t1.3646690406081e-47\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# new file, top 50 genes\n",
        "!head result.txt -n 50 | cut -f 1 | cut -d \"-\" -f 2 > genes.txt\n",
        "!head genes.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "d4a5fyYiOF8K",
        "outputId": "60b51da9-7535-4205-d4df-bd3ce72b7318"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "YER062C\n",
            "YDR536W\n",
            "YHR094C\n",
            "YNL065W\n",
            "YKL120W\n",
            "YLR264W\n",
            "YGR159C\n",
            "YJL122W\n",
            "YNL112W\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Actually, if $p \\le 0.05$, we accept that gene changed it's expression. But in this case there will be to many genes to analyze. So we take only top-50 genes."
      ],
      "metadata": {
        "id": "hBd8Nc7ORBMK"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "total = 0\n",
        "\n",
        "with open('result.txt', 'r') as f:\n",
        "    _ = f.readline()\n",
        "    for line in f:\n",
        "        try:\n",
        "            p_val = float(line.strip().split('\\t')[-1])\n",
        "        except:\n",
        "            p_val = 1\n",
        "        total += 1 if p_val < 0.05 else 0\n",
        "\n",
        "print(f'Total genes that changed expression (p_adj < 0.05) = {total}')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1DStHtjmp1Da",
        "outputId": "3cabf45b-7b7f-4dc9-8e17-19ae8b2b82c3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Total genes that changed expression (p_adj < 0.05) = 3180\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "higher = []\n",
        "lower = []\n",
        "\n",
        "with open('result.txt', 'r') as f:\n",
        "    _ = f.readline()\n",
        "    for line in f:\n",
        "        try:\n",
        "            # expression became significantly lower\n",
        "            if float(line.strip().split('\\t')[3]) < 1 and float(line.strip().split('\\t')[-1]) < 0.05:\n",
        "                higher.append(line.strip().split('\\t')[0])\n",
        "            # expression became significantly higher\n",
        "            elif float(line.strip().split('\\t')[3]) > 1 and float(line.strip().split('\\t')[-1]) < 0.05:\n",
        "                lower.append(line.strip().split('\\t')[0])\n",
        "        except:\n",
        "            pass"
      ],
      "metadata": {
        "id": "_d69EOqvsg4k"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "print(f'Expression became higher in {len(higher)} genes')\n",
        "print(f'Expression became lower in {len(lower)} genes')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HHuq1pAHtXPl",
        "outputId": "7b2b58ab-aaed-466c-bfc2-75b3552ce4c5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Expression became higher in 1613 genes\n",
            "Expression became lower in 1567 genes\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "qfHpiy49Zw7c"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "gene = ['gene-'+g for g in ['YBL039C', 'YOL151W']]\n",
        "\n",
        "for g in gene:\n",
        "    print(f\"expression of {g} became {'higher' if g in higher else 'lower'}\")\n",
        "\n",
        "# ==> expression of YBL039C became lower (GO-term = 'lipid metabolic process')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Tv5xV4BKZw-W",
        "outputId": "bd09ad6e-0813-4f42-8088-434d094e27a7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "expression of gene-YBL039C became lower\n",
            "expression of gene-YOL151W became lower\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "gene = ['gene-'+g for g in ['YKR097W', 'YNL117W', 'YDR342C']]\n",
        "\n",
        "for g in gene:\n",
        "    print(f\"expression of {g} became {'higher' if g in higher else 'lower'}\")\n",
        "\n",
        "# ==> expression of them became higher (GO-terms = 'carbohydrate metabolic process' and 'carbohydrate transport'. But for some genes with this terms expression became lower!)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TEP-m_KQaZRB",
        "outputId": "bc093cf6-9fdc-4590-e5a7-a896e6d34e4d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "expression of gene-YKR097W became higher\n",
            "expression of gene-YNL117W became higher\n",
            "expression of gene-YDR342C became higher\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Description of each ID can be found on https://www.yeastgenome.org/locus/ **ID**"
      ],
      "metadata": {
        "id": "CP3Ztvqoa8gJ"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "#### 3.1 Gene ontology"
      ],
      "metadata": {
        "id": "bMcR7udaO1Kd"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Gene ontology (GO) terms are curated keywords that describe the function of individual genes and can help researchers understand what role they may be playing. The Saccharomyces Genome Database, maintained by Stanford, contains all of the GO terms associated with yeast."
      ],
      "metadata": {
        "id": "I4X8XVtPO7q4"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Using http://www.yeastgenome.org/cgi-bin/GO/goSlimMapper.pl we:<br>\n",
        "uploaded our genes.txt file $\\rightarrow$ \"Yeast GO-Slim: Process\" $\\rightarrow$ \"SELECT ALL TERMS\" $\\rightarrow$ Search"
      ],
      "metadata": {
        "id": "t7SfCHIWPCgu"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "$~$\n",
        "\n",
        "**Results:** from all genes with p-value adjusted $<0.05$ we took top49 genes. From 49 genes there were 1 currently unannotated (YLR264C) and 7 genes that could not be mapped to terms in the current GO slim set (YOR360C, YGR079W, YBL028C, YLR327C, YJL108C, YML018C, YJL107C). For each of the remaining 42 genes **GO Term (GO ID)** was found."
      ],
      "metadata": {
        "id": "dM-vWzoUSVGy"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# results here\n",
        "\n",
        "import requests\n",
        "\n",
        "response = requests.get('https://gotermfinder.yeastgenome.org/mapper_genes_1764968_slimTerms.txt')\n",
        "print(response.text)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "g6DQKZflOua_",
        "outputId": "8c96704b-f0e1-4a10-a4a1-0de0977993bf"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Your input list contains 49 genes.\n",
            "\n",
            "These 1 identifier(s) represent either invalid gene names, dubious ORFs unlikely to encode functional proteins or genes which are currently unannotated: YLR264C\n",
            "\n",
            "These 7 identifier(s) represent valid gene names that either could not be mapped to terms in the current GO slim set or are currently annotated to the root node for the slim set being used: YOR360C YGR079W YBL028C YLR327C YJL108C YML018C YJL107C\n",
            "\n",
            "GO slim terms from the biological process ontology.\n",
            "\n",
            "TERM\trRNA processing\n",
            "GOID\tGO:0006364\n",
            "NUM_ANNOTATIONS\t: 12 of 48 genes (25.00%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 327 of 6486 annotated genes (5.04%)\n",
            "ANNOTATED_GENES\t: YDR449C, YEL026W, YER127W, YGR159C, YHR066W, YHR196W, YJL069C, YMR093W, YNL112W, YNL182C, YOL041C, YOL080C\n",
            "\n",
            "TERM\tribosomal large subunit biogenesis\n",
            "GOID\tGO:0042273\n",
            "NUM_ANNOTATIONS\t: 8 of 48 genes (16.67%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 118 of 6486 annotated genes (1.82%)\n",
            "ANNOTATED_GENES\t: YCR072C, YDL063C, YHR066W, YIR012W, YJL122W, YNL182C, YOL041C, YOL080C\n",
            "\n",
            "TERM\tribosomal small subunit biogenesis\n",
            "GOID\tGO:0042274\n",
            "NUM_ANNOTATIONS\t: 7 of 48 genes (14.58%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 137 of 6486 annotated genes (2.11%)\n",
            "ANNOTATED_GENES\t: YDR449C, YEL026W, YER127W, YGR159C, YHR196W, YJL069C, YMR093W\n",
            "\n",
            "TERM\ttranscription by RNA polymerase I\n",
            "GOID\tGO:0006360\n",
            "NUM_ANNOTATIONS\t: 6 of 48 genes (12.50%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 70 of 6486 annotated genes (1.08%)\n",
            "ANNOTATED_GENES\t: YHR196W, YJL148W, YJR063W, YML043C, YMR093W, YNL248C\n",
            "\n",
            "TERM\tribosome assembly\n",
            "GOID\tGO:0042255\n",
            "NUM_ANNOTATIONS\t: 6 of 48 genes (12.50%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 64 of 6486 annotated genes (0.99%)\n",
            "ANNOTATED_GENES\t: YCR072C, YGR159C, YHR066W, YIR012W, YNL182C, YOL080C\n",
            "\n",
            "TERM\tnucleobase-containing small molecule metabolic process\n",
            "GOID\tGO:0055086\n",
            "NUM_ANNOTATIONS\t: 4 of 48 genes (8.33%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 163 of 6486 annotated genes (2.51%)\n",
            "ANNOTATED_GENES\t: YBL039C, YMR300C, YNL141W, YOL136C\n",
            "\n",
            "TERM\tcarbohydrate metabolic process\n",
            "GOID\tGO:0005975\n",
            "NUM_ANNOTATIONS\t: 4 of 48 genes (8.33%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 148 of 6486 annotated genes (2.28%)\n",
            "ANNOTATED_GENES\t: YBR105C, YER062C, YKR097W, YOL136C\n",
            "\n",
            "TERM\ttransmembrane transport\n",
            "GOID\tGO:0055085\n",
            "NUM_ANNOTATIONS\t: 3 of 48 genes (6.25%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 271 of 6486 annotated genes (4.18%)\n",
            "ANNOTATED_GENES\t: YDR536W, YHR094C, YNL065W\n",
            "\n",
            "TERM\tion transport\n",
            "GOID\tGO:0006811\n",
            "NUM_ANNOTATIONS\t: 3 of 48 genes (6.25%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 210 of 6486 annotated genes (3.24%)\n",
            "ANNOTATED_GENES\t: YKL120W, YNL065W, YNR060W\n",
            "\n",
            "TERM\tRNA catabolic process\n",
            "GOID\tGO:0006401\n",
            "NUM_ANNOTATIONS\t: 3 of 48 genes (6.25%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 152 of 6486 annotated genes (2.34%)\n",
            "ANNOTATED_GENES\t: YLR264W, YNL112W, YOR359W\n",
            "\n",
            "TERM\tregulation of translation\n",
            "GOID\tGO:0006417\n",
            "NUM_ANNOTATIONS\t: 3 of 48 genes (6.25%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 207 of 6486 annotated genes (3.19%)\n",
            "ANNOTATED_GENES\t: YLR264W, YNL112W, YOR359W\n",
            "\n",
            "TERM\ttRNA processing\n",
            "GOID\tGO:0008033\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 118 of 6486 annotated genes (1.82%)\n",
            "ANNOTATED_GENES\t: YOL124C, YPL212C\n",
            "\n",
            "TERM\ttranscription by RNA polymerase II\n",
            "GOID\tGO:0006366\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 486 of 6486 annotated genes (7.49%)\n",
            "ANNOTATED_GENES\t: YJR063W, YNL112W\n",
            "\n",
            "TERM\tresponse to chemical\n",
            "GOID\tGO:0042221\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 440 of 6486 annotated genes (6.78%)\n",
            "ANNOTATED_GENES\t: YLR224W, YOR271C\n",
            "\n",
            "TERM\tDNA-templated transcription, termination\n",
            "GOID\tGO:0006353\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 41 of 6486 annotated genes (0.63%)\n",
            "ANNOTATED_GENES\t: YJR063W, YNL112W\n",
            "\n",
            "TERM\tcarbohydrate transport\n",
            "GOID\tGO:0008643\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 36 of 6486 annotated genes (0.56%)\n",
            "ANNOTATED_GENES\t: YDR536W, YHR094C\n",
            "\n",
            "TERM\tnucleobase-containing compound transport\n",
            "GOID\tGO:0015931\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 139 of 6486 annotated genes (2.14%)\n",
            "ANNOTATED_GENES\t: YHR196W, YLR264W\n",
            "\n",
            "TERM\tlipid metabolic process\n",
            "GOID\tGO:0006629\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 297 of 6486 annotated genes (4.58%)\n",
            "ANNOTATED_GENES\t: YBL039C, YOL151W\n",
            "\n",
            "TERM\tDNA-templated transcription, elongation\n",
            "GOID\tGO:0006354\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 105 of 6486 annotated genes (1.62%)\n",
            "ANNOTATED_GENES\t: YJL148W, YNL248C\n",
            "\n",
            "TERM\tRNA modification\n",
            "GOID\tGO:0009451\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 177 of 6486 annotated genes (2.73%)\n",
            "ANNOTATED_GENES\t: YOL124C, YPL212C\n",
            "\n",
            "TERM\tregulation of DNA metabolic process\n",
            "GOID\tGO:0051052\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 119 of 6486 annotated genes (1.83%)\n",
            "ANNOTATED_GENES\t: YNL182C, YOR359W\n",
            "\n",
            "TERM\tproteolysis involved in cellular protein catabolic process\n",
            "GOID\tGO:0051603\n",
            "NUM_ANNOTATIONS\t: 2 of 48 genes (4.17%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 224 of 6486 annotated genes (3.45%)\n",
            "ANNOTATED_GENES\t: YBR105C, YLR224W\n",
            "\n",
            "TERM\tcellular amino acid metabolic process\n",
            "GOID\tGO:0006520\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 155 of 6486 annotated genes (2.39%)\n",
            "ANNOTATED_GENES\t: YLR180W\n",
            "\n",
            "TERM\ttRNA aminoacylation for protein translation\n",
            "GOID\tGO:0006418\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 36 of 6486 annotated genes (0.56%)\n",
            "ANNOTATED_GENES\t: YDR037W\n",
            "\n",
            "TERM\tDNA replication\n",
            "GOID\tGO:0006260\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 139 of 6486 annotated genes (2.14%)\n",
            "ANNOTATED_GENES\t: YNL182C\n",
            "\n",
            "TERM\torganelle assembly\n",
            "GOID\tGO:0070925\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 112 of 6486 annotated genes (1.73%)\n",
            "ANNOTATED_GENES\t: YLR180W\n",
            "\n",
            "TERM\tcytoplasmic translation\n",
            "GOID\tGO:0002181\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 197 of 6486 annotated genes (3.04%)\n",
            "ANNOTATED_GENES\t: YLR264W\n",
            "\n",
            "TERM\tresponse to osmotic stress\n",
            "GOID\tGO:0006970\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 68 of 6486 annotated genes (1.05%)\n",
            "ANNOTATED_GENES\t: YER062C\n",
            "\n",
            "TERM\tgeneration of precursor metabolites and energy\n",
            "GOID\tGO:0006091\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 80 of 6486 annotated genes (1.23%)\n",
            "ANNOTATED_GENES\t: YOL136C\n",
            "\n",
            "TERM\tmRNA processing\n",
            "GOID\tGO:0006397\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 180 of 6486 annotated genes (2.78%)\n",
            "ANNOTATED_GENES\t: YEL026W\n",
            "\n",
            "TERM\tamino acid transport\n",
            "GOID\tGO:0006865\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 46 of 6486 annotated genes (0.71%)\n",
            "ANNOTATED_GENES\t: YNL065W\n",
            "\n",
            "TERM\tDNA recombination\n",
            "GOID\tGO:0006310\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 190 of 6486 annotated genes (2.93%)\n",
            "ANNOTATED_GENES\t: YGR159C\n",
            "\n",
            "TERM\tregulation of organelle organization\n",
            "GOID\tGO:0033043\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 237 of 6486 annotated genes (3.65%)\n",
            "ANNOTATED_GENES\t: YLR180W\n",
            "\n",
            "TERM\tRNA splicing\n",
            "GOID\tGO:0008380\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 140 of 6486 annotated genes (2.16%)\n",
            "ANNOTATED_GENES\t: YEL026W\n",
            "\n",
            "TERM\tprotein targeting\n",
            "GOID\tGO:0006605\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 242 of 6486 annotated genes (3.73%)\n",
            "ANNOTATED_GENES\t: YBR105C\n",
            "\n",
            "TERM\tmonocarboxylic acid metabolic process\n",
            "GOID\tGO:0032787\n",
            "NUM_ANNOTATIONS\t: 1 of 48 genes (2.08%)\n",
            "TOTAL_NUM_ANNOTATIONS\t: 127 of 6486 annotated genes (1.96%)\n",
            "ANNOTATED_GENES\t: YOL136C\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "vxLkeANDOugc"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "W2gs973-DKEL"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}