{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Copy of Lab_notebook OLD.ipynb",
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "mto0NijBWxJU"
      },
      "source": [
        "## 0. Install all neccessary programms on Google Colab\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 576
        },
        "id": "zPZmfoZBU9-h",
        "outputId": "2b327f8b-d84b-4ac2-cbd4-f05b2e23105b"
      },
      "source": [
        "import os\n",
        "import matplotlib.pyplot as plt\n",
        "import pandas as pd\n",
        "import seaborn as sns\n",
        "\n",
        "# !sudo apt install fastqc -qq\n",
        "# !sudo apt install jellyfish -qq\n",
        "# !sudo apt install spades -qq\n",
        "# !sudo apt install barrnap \n",
        "\n",
        "############################ install QUAST ############################\n",
        "# !wget https://downloads.sourceforge.net/project/quast/quast-5.0.2.tar.gz\n",
        "# !tar -xzf quast-5.0.2.tar.gz\n",
        "# %cd 'quast-5.0.2'\n",
        "# !./setup.py install\n",
        "# %cd ..\n",
        "########################################################################\n",
        "\n",
        "############################ install Prokka ############################\n",
        "# !sudo apt-get install libdatetime-perl libxml-simple-perl libdigest-md5-perl git default-jre bioperl\n",
        "# !sudo apt-get install parallel\n",
        "# !sudo cpan Bio::Perly\n",
        "# !git clone https://github.com/tseemann/prokka.git $HOME/prokka\n",
        "# !$HOME/prokka/bin/prokka --setupdb\n",
        "\n",
        "'''\n",
        "в файле bin/prokka надо вручную поменять зависимости (MINVER) на \"2.0\" у следующих библиотек:\n",
        "  'blastp' => {\n",
        "    GETVER  => \"blastp -version\",\n",
        "    REGEXP  => qr/blastp:\\s+($BIDEC)/,\n",
        "    MINVER  => \"2.0\",\n",
        "    NEEDED  => 1,\n",
        "  },\n",
        "  'makeblastdb' => {\n",
        "    GETVER  => \"makeblastdb -version\",\n",
        "    REGEXP  => qr/makeblastdb:\\s+($BIDEC)/,\n",
        "    MINVER  => \"2.0\",\n",
        "    NEEDED  => 0,  # only if --proteins used\n",
        "  },\n",
        "'''\n",
        "########################################################################\n",
        "\n",
        "############################ install Mauve ############################\n",
        "# !wget http://darlinglab.org/mauve/snapshots/2015/2015-02-13/linux-x64/mauve_linux_snapshot_2015-02-13.tar.gz\n",
        "# !tar -xzf mauve_linux_snapshot_2015-02-13.tar.gz\n",
        "# %cd mauve_snapshot_2015-02-13\n",
        "# !./setup.py install\n",
        "# %cd ..\n",
        "\n",
        "# and java (set environment variable)\n",
        "# def install_java():\n",
        "#   !apt-get install -y openjdk-17-jdk-headless -qq > /dev/null      #install openjdk\n",
        "#   os.environ[\"JAVA_HOME\"] = \"/usr/lib/jvm/java-17-openjdk-amd64\"     #set environment variable\n",
        "#   !java -version       #check java version\n",
        "# install_java()\n",
        "########################################################################"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "The following additional packages will be installed:\n",
            "  libjellyfish-2.0-2\n",
            "The following NEW packages will be installed:\n",
            "  jellyfish libjellyfish-2.0-2\n",
            "0 upgraded, 2 newly installed, 0 to remove and 37 not upgraded.\n",
            "Need to get 418 kB of archives.\n",
            "After this operation, 808 kB of additional disk space will be used.\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 2.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libjellyfish-2.0-2:amd64.\n",
            "(Reading database ... 155219 files and directories currently installed.)\n",
            "Preparing to unpack .../libjellyfish-2.0-2_2.2.8-3build1_amd64.deb ...\n",
            "Unpacking libjellyfish-2.0-2:amd64 (2.2.8-3build1) ...\n",
            "Selecting previously unselected package jellyfish.\n",
            "Preparing to unpack .../jellyfish_2.2.8-3build1_amd64.deb ...\n",
            "Unpacking jellyfish (2.2.8-3build1) ...\n",
            "Setting up libjellyfish-2.0-2:amd64 (2.2.8-3build1) ...\n",
            "Setting up jellyfish (2.2.8-3build1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            },
            "text/plain": [
              "'\\nв файле bin/prokka надо вручную поменять зависимости (MINVER) на \"2.0\" у следующих библиотек:\\n  \\'blastp\\' => {\\n    GETVER  => \"blastp -version\",\\n    REGEXP  => qr/blastp:\\\\s+($BIDEC)/,\\n    MINVER  => \"2.0\",\\n    NEEDED  => 1,\\n  },\\n  \\'makeblastdb\\' => {\\n    GETVER  => \"makeblastdb -version\",\\n    REGEXP  => qr/makeblastdb:\\\\s+($BIDEC)/,\\n    MINVER  => \"2.0\",\\n    NEEDED  => 0,  # only if --proteins used\\n  },\\n'"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uauU5R15HXYS"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "nDQjWwggW0_4"
      },
      "source": [
        "## **00. Data preparation**"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "M0w0n2NBS5vQ"
      },
      "source": [
        "!mkdir raw_data"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jROrMzWtTSg2",
        "outputId": "a3469bce-c1b3-46cd-ad78-dc8de8ad53ba"
      },
      "source": [
        "# SRR292678 - paired end (forward and reverse)\n",
        "!wget -c https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R1_001.fastq.gz\n",
        "!wget -c https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R2_001.fastq.gz\n",
        "\n",
        "# SRR292862 – mate pair (forward and reverse)\n",
        "!wget -c https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R1_001.fastq.gz\n",
        "!wget -c https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R2_001.fastq.gz\n",
        "\n",
        "# SRR292770 – mate pair (forward and reverse)\n",
        "!wget -c https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R1_001.fastq.gz\n",
        "!wget -c https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R2_001.fastq.gz"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2021-11-19 00:48:24--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R1_001.fastq.gz\n",
            "Resolving d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)... 13.249.139.38, 13.249.139.96, 13.249.139.27, ...\n",
            "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|13.249.139.38|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 425982530 (406M) [application/x-gzip]\n",
            "Saving to: ‘SRR292678sub_S1_L001_R1_001.fastq.gz’\n",
            "\n",
            "SRR292678sub_S1_L00 100%[===================>] 406.25M  12.5MB/s    in 33s     \n",
            "\n",
            "2021-11-19 00:48:57 (12.5 MB/s) - ‘SRR292678sub_S1_L001_R1_001.fastq.gz’ saved [425982530/425982530]\n",
            "\n",
            "--2021-11-19 00:48:57--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R2_001.fastq.gz\n",
            "Resolving d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)... 13.249.139.38, 13.249.139.96, 13.249.139.27, ...\n",
            "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|13.249.139.38|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 426246191 (406M) [application/x-gzip]\n",
            "Saving to: ‘SRR292678sub_S1_L001_R2_001.fastq.gz’\n",
            "\n",
            "SRR292678sub_S1_L00 100%[===================>] 406.50M  43.5MB/s    in 9.6s    \n",
            "\n",
            "2021-11-19 00:49:08 (42.6 MB/s) - ‘SRR292678sub_S1_L001_R2_001.fastq.gz’ saved [426246191/426246191]\n",
            "\n",
            "--2021-11-19 00:49:08--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R1_001.fastq.gz\n",
            "Resolving d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)... 99.84.254.43, 99.84.254.71, 99.84.254.215, ...\n",
            "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|99.84.254.43|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 226372891 (216M) [application/x-gzip]\n",
            "Saving to: ‘SRR292862_S2_L001_R1_001.fastq.gz’\n",
            "\n",
            "SRR292862_S2_L001_R 100%[===================>] 215.89M  62.8MB/s    in 3.4s    \n",
            "\n",
            "2021-11-19 00:49:11 (62.8 MB/s) - ‘SRR292862_S2_L001_R1_001.fastq.gz’ saved [226372891/226372891]\n",
            "\n",
            "--2021-11-19 00:49:11--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R2_001.fastq.gz\n",
            "Resolving d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)... 99.84.254.43, 99.84.254.71, 99.84.254.215, ...\n",
            "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|99.84.254.43|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 231496590 (221M) [application/x-gzip]\n",
            "Saving to: ‘SRR292862_S2_L001_R2_001.fastq.gz’\n",
            "\n",
            "SRR292862_S2_L001_R 100%[===================>] 220.77M  51.9MB/s    in 4.1s    \n",
            "\n",
            "2021-11-19 00:49:16 (54.3 MB/s) - ‘SRR292862_S2_L001_R2_001.fastq.gz’ saved [231496590/231496590]\n",
            "\n",
            "--2021-11-19 00:49:16--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R1_001.fastq.gz\n",
            "Resolving d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)... 13.249.139.38, 13.249.139.96, 13.249.139.97, ...\n",
            "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|13.249.139.38|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 226840159 (216M) [application/x-gzip]\n",
            "Saving to: ‘SRR292770_S1_L001_R1_001.fastq.gz’\n",
            "\n",
            "SRR292770_S1_L001_R 100%[===================>] 216.33M  49.5MB/s    in 4.4s    \n",
            "\n",
            "2021-11-19 00:49:21 (49.3 MB/s) - ‘SRR292770_S1_L001_R1_001.fastq.gz’ saved [226840159/226840159]\n",
            "\n",
            "--2021-11-19 00:49:21--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R2_001.fastq.gz\n",
            "Resolving d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)... 13.249.139.27, 13.249.139.97, 13.249.139.96, ...\n",
            "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|13.249.139.27|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 232544438 (222M) [application/x-gzip]\n",
            "Saving to: ‘SRR292770_S1_L001_R2_001.fastq.gz’\n",
            "\n",
            "SRR292770_S1_L001_R 100%[===================>] 221.77M  62.1MB/s    in 3.8s    \n",
            "\n",
            "2021-11-19 00:49:25 (59.0 MB/s) - ‘SRR292770_S1_L001_R2_001.fastq.gz’ saved [232544438/232544438]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9mtiMHvBTqGI"
      },
      "source": [
        "!mv SRR292678sub_S1_L001_R1_001.fastq.gz raw_data/paired_end_470bp_fwd.fastq.gz\n",
        "!mv SRR292678sub_S1_L001_R2_001.fastq.gz raw_data/paired_end_470bp_rev.fastq.gz\n",
        "\n",
        "!mv SRR292862_S2_L001_R1_001.fastq.gz raw_data/mate_pair_2kb_fwd.fastq.gz\n",
        "!mv SRR292862_S2_L001_R2_001.fastq.gz raw_data/mate_pair_2kb_rev.fastq.gz\n",
        "\n",
        "!mv SRR292770_S1_L001_R1_001.fastq.gz raw_data/mate_pair_6kb_fwd.fastq.gz\n",
        "!mv SRR292770_S1_L001_R2_001.fastq.gz raw_data/mate_pair_6kb_rev.fastq.gz\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "xBsBaL9nUFe0"
      },
      "source": [
        "# unpack\n",
        "!gunzip raw_data/*.fastq.gz"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "4U_lWP2oHT4s"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "zkEbPN_QidR_"
      },
      "source": [
        "## **1. Data inspection**\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ue-r54y3eqxf"
      },
      "source": [
        "Оценим качество ридов и их количество."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "10rGQD2CU9AY"
      },
      "source": [
        "!mkdir 1_fastqc"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bnABiJJgTqLr"
      },
      "source": [
        "!fastqc -q raw_data/*.fastq -o 1_fastqc/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "KCRKSDtOi269"
      },
      "source": [
        "**Number of reads in each file:** <br>\n",
        "\n",
        "|          file name           |   fwd   |   rev   |\n",
        "|------------------------------|---------|---------|\n",
        "| SRR292678 (paired_end_470bp) | 5499346 | 5499346 |\n",
        "|   SRR292862 (mate_pair_2kb)  | 5102041 | 5102041 |\n",
        "|   SRR292770 (mate_pair_6kb)  | 5102041 | 5102041 |"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "LCAKkPpHq-1u"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "xTTfpFoCHRwc"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "F6VjEBavremb"
      },
      "source": [
        "## **2. (optional, 1 extra credit). K-mer profile and genome size estimation**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "lDAxujrluWil"
      },
      "source": [
        "For our purposes it’s OK to only analyze the paired-end library (SRR292678), and only forward file (because the results for them are almost same). <br>\n",
        "Searching for k-mer with size = 31"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "D4I0ZyQiuV0v"
      },
      "source": [
        "!mkdir 2_k-mer_count"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9yWXSrbcuWA-"
      },
      "source": [
        "# -m, --mer-len=uint32                    *Length of mer\n",
        "# -s, --size=uint64                       *Initial hash size (> avg ecoli genome size in our case)\n",
        "# -C, --canonical                          Count both strand, canonical representation (false)\n",
        "# -o, --output=string                      Output file (mer_counts.jf)\n",
        "\n",
        "\n",
        "!jellyfish count -m 31 -C -s 6M raw_data/paired_end_470bp_fwd.fastq -o 2_k-mer_count/31mer_counts_fwd.jf"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "5jIPGHJ64Kx-"
      },
      "source": [
        "# convert to .tsv\n",
        "!jellyfish histo 2_k-mer_count/31mer_counts_fwd.jf > 2_k-mer_count/31mer_fwd_hist.tsv\n",
        "\n",
        "# left col: a list of the bins (the number of times a k-mer occurs or its ‘depth’)\n",
        "# right col: the count for the number of k-mers in the data that fit into that category"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qpUAMS457vaD"
      },
      "source": [
        "Небольшой питоновский код чтобы обработать результаты jellyfish count и построить графики:"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "IJ4M0lGA4K2Z",
        "outputId": "47e6b2a2-bacb-4bec-a77d-5f3853941214"
      },
      "source": [
        "fwd = pd.read_csv('2_k-mer_count/31mer_fwd_hist.tsv', sep=' ', header=None)\n",
        "fwd.columns = ['bins', 'count']\n",
        "\n",
        "fwd.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>bins</th>\n",
              "      <th>count</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>6755622</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>2</td>\n",
              "      <td>266510</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>3</td>\n",
              "      <td>48557</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>4</td>\n",
              "      <td>18950</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>5</td>\n",
              "      <td>11926</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   bins    count\n",
              "0     1  6755622\n",
              "1     2   266510\n",
              "2     3    48557\n",
              "3     4    18950\n",
              "4     5    11926"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "NKg9rSpCFOVF"
      },
      "source": [
        "Интерпретация графиков:\n",
        "* X $-$ сколько раз 31-мер встречается в ридах\n",
        "* Y(x) $-$ число разных 31-меров, которые встречаются в ридах x раз"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "fj9SF25V4K41"
      },
      "source": [
        "fwd[5:].plot(x='bins', y='count', style='.')"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 81
        },
        "id": "idw1khrfCCcC",
        "outputId": "49af1d52-3720-4efb-916f-868c2e7aa283"
      },
      "source": [
        "fwd_count_max = fwd[5:]['count'].max()\n",
        "fwd_max = fwd[fwd['count'] == fwd_count_max]\n",
        "\n",
        "fwd_max"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>bins</th>\n",
              "      <th>count</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>61</th>\n",
              "      <td>62</td>\n",
              "      <td>89747</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "    bins  count\n",
              "61    62  89747"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "g4W2wLxZHIWF"
      },
      "source": [
        "#### **GENOME SIZE ANALYSIS**\n",
        "\n",
        "$N = \\frac{(M*L)}{(L-K+1)} = \\frac{62 * 90}{90 - 31 + 1} = 93$\n",
        "\n",
        "$Genome\\_size = \\frac{T}{N} = \\frac{5499346}{93} \\simeq 59132 $ b.p.\n",
        "\n",
        "(N: Depth of coverage, M: Kmer peak, K: Kmer-size, L: avg read length T: Total bases)\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bWojKtsZstze"
      },
      "source": [
        "Actual *E.Coli* genome size is about $4.64$ (Escherichia coli str. K-12 substr. MG1655) $-$ $5.5$ (Escherichia coli O157:H7 str. Sakai)."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Z3qoszjTHPbw"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6FPtnanOHHqq"
      },
      "source": [
        "## **3. Assembling E. coli X genome from paired reads**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "nYnYST5DojIp"
      },
      "source": [
        "Первым делом мы запускаем сборку на paired_end файлах с размером вставки (расстояние между ридами в парах ридов / distance between reads within read-pairs) 470п.н. (файл SRR292678). Ожидаем, что сборка будет не слишком хороша, так как insert size маленький и сложно будет определять последовательность контигов для их объединения в скаффолды."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "u3DGl3iFHxuN"
      },
      "source": [
        "!mkdir 3_spades"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "SruUteWQHG-O"
      },
      "source": [
        "# Verify that the software was installed correctly\n",
        "# !spades --test"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "YV2vbOSDHiQh"
      },
      "source": [
        ""
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Een2hFCBHHDm",
        "outputId": "00d0b4b4-fdc8-402b-91c3-b4980f22aca4"
      },
      "source": [
        "# -1\t<filename>\tfile with forward paired-end reads\n",
        "# -2\t<filename>\tfile with reverse paired-end reads\n",
        "# -o\t<output_dir>\tdirectory to store all the resulting files (required)\n",
        "\n",
        "!spades -1 raw_data/paired_end_470bp_fwd.fastq -2 raw_data/paired_end_470bp_rev.fastq -o 3_spades/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Command line: /usr/lib/spades/bin/spades.py\t-1\t/content/raw_data/paired_end_470bp_forw.fastq\t-2\t/content/raw_data/paired_end_470bp_rev.fastq\t-o\t/content/3_spades\t\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 3.11.1\n",
            "  Python version: 2.7.17\n",
            "  OS: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic\n",
            "\n",
            "Output dir: /content/3_spades\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
            "  Reads:\n",
            "    Library number: 1, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/content/raw_data/paired_end_470bp_forw.fastq']\n",
            "      right reads: ['/content/raw_data/paired_end_470bp_rev.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed (with gzip)\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned OFF\n",
            "  MismatchCorrector will be SKIPPED\n",
            "  Coverage cutoff is turned OFF\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/3_spades/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 12\n",
            "\n",
            "\n",
            "======= SPAdes pipeline started. Log can be found here: /content/3_spades/spades.log\n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running read error correction tool: /usr/lib/spades/bin/hammer /content/3_spades/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     0M / 14M   INFO    General                 (main.cpp                  :  83)   Starting BayesHammer, built from N/A, git revision N/A\n",
            "  0:00:00.002     0M / 14M   INFO    General                 (main.cpp                  :  84)   Loading config from /content/3_spades/corrected/configs/config.info\n",
            "  0:00:00.003     0M / 14M   INFO    General                 (memory_limit.hpp          :  52)   Memory limit set to 12 Gb\n",
            "  0:00:00.003     0M / 14M   INFO    General                 (main.cpp                  :  93)   Trying to determine PHRED offset\n",
            "  0:00:00.006     0M / 14M   INFO    General                 (main.cpp                  :  99)   Determined value is 33\n",
            "  0:00:00.009     0M / 14M   INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.009     0M / 14M   INFO    General                 (main.cpp                  : 120)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.016     0M / 14M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:00:00.016     0M / 14M   INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:00:00.016     0M / 14M   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.016     0M / 14M   INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:00:00.016     0M / 14M   INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 2097152\n",
            "  0:00:00.021     0M / 14M   INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/paired_end_470bp_forw.fastq\n",
            "  0:00:17.093     0M / 566M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 476320 reads\n",
            "  0:00:33.970     0M / 567M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 952815 reads\n",
            "  0:00:49.831     0M / 567M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 1429410 reads\n",
            "  0:01:06.109     0M / 568M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 1906672 reads\n",
            "  0:01:22.898     0M / 568M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 2383606 reads\n",
            "  0:01:44.484     0M / 568M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 2860029 reads\n",
            "  0:02:01.557     0M / 568M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 3336548 reads\n",
            "  0:02:35.777     0M / 568M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 4289331 reads\n",
            "  0:03:19.745     0M / 568M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/paired_end_470bp_rev.fastq\n",
            "  0:05:14.133     0M / 584M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 8837885 reads\n",
            "  0:06:31.886     0M / 584M  INFO   K-mer Splitting          (kmer_data.cpp             : 111)   Total 10998692 reads processed\n",
            "  0:06:32.210     0M / 584M  INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:06:48.803     0M / 714M  INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 35869396 kmers in total.\n",
            "  0:06:48.804     0M / 714M  INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:06:49.365     0M / 714M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:06:59.670     0M / 714M  INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:07:00.293     0M / 714M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 16640728 bytes occupied (3.7114 bits per kmer).\n",
            "  0:07:00.293     0M / 714M  INFO   K-mer Counting           (kmer_data.cpp             : 357)   Arranging kmers in hash map order\n",
            "  0:07:05.942     0M / 1G    INFO    General                 (main.cpp                  : 155)   Clustering Hamming graph.\n",
            "  0:14:15.568     0M / 1G    INFO    General                 (main.cpp                  : 162)   Extracting clusters\n",
            "  0:16:40.230     0M / 1G    INFO    General                 (main.cpp                  : 174)   Clustering done. Total clusters: 13059930\n",
            "  0:16:40.253     0M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 379)   Collecting K-mer information, this takes a while.\n",
            "  0:16:41.278     0M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /content/raw_data/paired_end_470bp_forw.fastq\n",
            "  0:32:12.682     0M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /content/raw_data/paired_end_470bp_rev.fastq\n",
            "  0:46:53.973     0M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 392)   Collection done, postprocessing.\n",
            "  0:46:54.131     0M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 406)   There are 35869396 kmers in total. Among them 23232864 (64.7707%) are singletons.\n",
            "  0:46:54.131     0M / 1G    INFO    General                 (main.cpp                  : 180)   Subclustering Hamming graph\n",
            "  0:47:40.551     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 7 non-read kmers were generated.\n",
            "  0:47:40.551     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
            "  0:47:40.552     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 4366674. Among them 2314470 (53.003%) are good\n",
            "  0:47:40.552     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 40749. Among them 40317 (98.9399%) are good\n",
            "  0:47:40.552     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 8817994. Among them 7328745 (83.1112%) are good\n",
            "  0:47:40.552     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 3.57255 kmers\n",
            "  0:47:40.552     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.01904\n",
            "  0:47:40.552     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 9683532\n",
            "  0:47:40.553     0M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.92945,0.0468015,0.0182831,0.00546531),(0.0176801,0.97103,0.00425595,0.00703388),(0.00700894,0.00424543,0.971047,0.0176981),(0.00545963,0.0182562,0.0467842,0.9295))\n",
            "  0:47:40.569     0M / 1G    INFO    General                 (main.cpp                  : 185)   Finished clustering.\n",
            "  0:47:40.569     0M / 1G    INFO    General                 (main.cpp                  : 204)   Starting solid k-mers expansion in 2 threads.\n",
            "  0:59:07.624     0M / 1G    INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 0 produced 53046 new k-mers.\n",
            "  1:10:28.376     0M / 1G    INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 1 produced 1043 new k-mers.\n",
            "  1:20:09.837     0M / 1G    INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 2 produced 30 new k-mers.\n",
            "  1:30:15.762     0M / 1G    INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
            "  1:30:15.762     0M / 1G    INFO    General                 (main.cpp                  : 229)   Solid k-mers finalized\n",
            "  1:30:15.762     0M / 1G    INFO    General                 (hammer_tools.cpp          : 211)   Starting read correction in 2 threads.\n",
            "  1:30:15.763     0M / 1G    INFO    General                 (hammer_tools.cpp          : 222)   Correcting pair of reads: /content/raw_data/paired_end_470bp_forw.fastq and /content/raw_data/paired_end_470bp_rev.fastq\n",
            "  1:30:16.346     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 0 of 200000 reads.\n",
            "  1:30:28.244     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
            "  1:30:28.784     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
            "  1:30:29.153     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 1 of 200000 reads.\n",
            "  1:30:42.056     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 1\n",
            "  1:30:42.601     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 1\n",
            "  1:30:42.972     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 2 of 200000 reads.\n",
            "  1:30:55.762     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 2\n",
            "  1:30:56.228     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 2\n",
            "  1:30:56.574     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 3 of 200000 reads.\n",
            "  1:31:07.689     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 3\n",
            "  1:31:08.145     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 3\n",
            "  1:31:08.512     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 4 of 200000 reads.\n",
            "  1:31:19.103     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 4\n",
            "  1:31:19.570     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 4\n",
            "  1:31:19.906     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 5 of 200000 reads.\n",
            "  1:31:30.564     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 5\n",
            "  1:31:31.050     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 5\n",
            "  1:31:31.396     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 6 of 200000 reads.\n",
            "  1:31:43.545     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 6\n",
            "  1:31:44.051     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 6\n",
            "  1:31:44.395     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 7 of 200000 reads.\n",
            "  1:31:58.302     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 7\n",
            "  1:31:58.861     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 7\n",
            "  1:31:59.168     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 8 of 200000 reads.\n",
            "  1:32:09.830     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 8\n",
            "  1:32:10.364     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 8\n",
            "  1:32:10.722     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 9 of 200000 reads.\n",
            "  1:32:21.106     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 9\n",
            "  1:32:21.597     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 9\n",
            "  1:32:21.961     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 10 of 200000 reads.\n",
            "  1:32:32.250     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 10\n",
            "  1:32:32.731     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 10\n",
            "  1:32:33.117     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 11 of 200000 reads.\n",
            "  1:32:43.482     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 11\n",
            "  1:32:43.977     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 11\n",
            "  1:32:44.325     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 12 of 200000 reads.\n",
            "  1:32:54.752     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 12\n",
            "  1:32:55.247     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 12\n",
            "  1:32:55.596     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 13 of 200000 reads.\n",
            "  1:33:06.188     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 13\n",
            "  1:33:06.669     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 13\n",
            "  1:33:07.017     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 14 of 200000 reads.\n",
            "  1:33:17.477     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 14\n",
            "  1:33:17.948     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 14\n",
            "  1:33:18.299     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 15 of 200000 reads.\n",
            "  1:33:28.707     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 15\n",
            "  1:33:29.173     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 15\n",
            "  1:33:29.523     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 16 of 200000 reads.\n",
            "  1:33:39.996     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 16\n",
            "  1:33:40.471     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 16\n",
            "  1:33:40.822     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 17 of 200000 reads.\n",
            "  1:33:51.427     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 17\n",
            "  1:33:51.931     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 17\n",
            "  1:33:52.293     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 18 of 200000 reads.\n",
            "  1:34:02.912     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 18\n",
            "  1:34:03.403     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 18\n",
            "  1:34:03.738     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 19 of 200000 reads.\n",
            "  1:34:14.132     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 19\n",
            "  1:34:14.629     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 19\n",
            "  1:34:14.957     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 20 of 200000 reads.\n",
            "  1:34:25.265     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 20\n",
            "  1:34:25.785     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 20\n",
            "  1:34:26.110     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 21 of 200000 reads.\n",
            "  1:34:36.421     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 21\n",
            "  1:34:37.033     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 21\n",
            "  1:34:37.277     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 22 of 200000 reads.\n",
            "  1:34:47.554     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 22\n",
            "  1:34:48.025     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 22\n",
            "  1:34:48.369     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 23 of 200000 reads.\n",
            "  1:34:59.067     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 23\n",
            "  1:34:59.544     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 23\n",
            "  1:34:59.886     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 24 of 200000 reads.\n",
            "  1:35:11.020     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 24\n",
            "  1:35:11.525     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 24\n",
            "  1:35:11.907     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 25 of 200000 reads.\n",
            "  1:35:22.797     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 25\n",
            "  1:35:23.288     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 25\n",
            "  1:35:23.637     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 26 of 200000 reads.\n",
            "  1:35:34.336     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 26\n",
            "  1:35:34.810     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 26\n",
            "  1:35:35.011     0M / 1G    INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 27 of 99346 reads.\n",
            "  1:35:42.820     0M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 27\n",
            "  1:35:43.044     0M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 27\n",
            "  1:35:43.200     0M / 1G    INFO    General                 (hammer_tools.cpp          : 270)   Correction done. Changed 935573 bases in 788580 reads.\n",
            "  1:35:43.200     0M / 1G    INFO    General                 (hammer_tools.cpp          : 271)   Failed to correct 0 bases out of 989881557.\n",
            "  1:35:43.257     0M / 1G    INFO    General                 (main.cpp                  : 262)   Saving corrected dataset description to /content/3_spades/corrected/corrected.yaml\n",
            "  1:35:43.264     0M / 1G    INFO    General                 (main.cpp                  : 269)   All done. Exiting.\n",
            "\n",
            "== Compressing corrected reads (with gzip)\n",
            "\n",
            "== Dataset description file was created: /content/3_spades/corrected/corrected.yaml\n",
            "\n",
            "\n",
            "===== Read error correction finished. \n",
            "\n",
            "\n",
            "===== Assembling started.\n",
            "\n",
            "\n",
            "== Running assembler: K21\n",
            "\n",
            "  0:00:00.000     0M / 27M   INFO    General                 (main.cpp                  :  76)   Loading config from /content/3_spades/K21/configs/config.info\n",
            "  0:00:00.001     0M / 27M   INFO    General                 (memory_limit.hpp          :  52)   Memory limit set to 12 Gb\n",
            "  0:00:00.001     0M / 27M   INFO    General                 (main.cpp                  :  89)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.001     0M / 27M   INFO    General                 (main.cpp                  :  90)   Maximum k-mer length: 128\n",
            "  0:00:00.001     0M / 27M   INFO    General                 (main.cpp                  :  91)   Assembling dataset (/content/3_spades/dataset.info) with K=21\n",
            "  0:00:00.001     0M / 27M   INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
            "  0:00:00.001     0M / 27M   INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
            "  0:00:00.001     0M / 27M   INFO    General                 (launch.hpp                :  61)   Two-step RR enabled: 0\n",
            "  0:00:00.001     0M / 27M   INFO   StageManager             (stage.cpp                 : 126)   STAGE == Construction\n",
            "  0:00:00.002     0M / 27M   INFO    General                 (read_converter.hpp        :  84)   Converting reads to binary format for library #0 (takes a while)\n",
            "  0:00:00.002     0M / 27M   INFO    General                 (read_converter.hpp        :  85)   Converting paired reads\n",
            "  0:00:00.196     0M / 106M  INFO    General                 (binary_converter.hpp      : 139)   16384 reads processed\n",
            "  0:00:00.301     0M / 106M  INFO    General                 (binary_converter.hpp      : 139)   32768 reads processed\n",
            "  0:00:00.515     0M / 106M  INFO    General                 (binary_converter.hpp      : 139)   65536 reads processed\n",
            "  0:00:00.955     0M / 138M  INFO    General                 (binary_converter.hpp      : 139)   131072 reads processed\n",
            "  0:00:01.826     0M / 203M  INFO    General                 (binary_converter.hpp      : 139)   262144 reads processed\n",
            "  0:00:03.739     0M / 247M  INFO    General                 (binary_converter.hpp      : 139)   524288 reads processed\n",
            "  0:00:07.556     0M / 264M  INFO    General                 (binary_converter.hpp      : 139)   1048576 reads processed\n",
            "  0:00:14.909     0M / 296M  INFO    General                 (binary_converter.hpp      : 139)   2097152 reads processed\n",
            "  0:00:29.684     0M / 298M  INFO    General                 (binary_converter.hpp      : 139)   4194304 reads processed\n",
            "  0:00:39.104     0M / 299M  INFO    General                 (binary_converter.hpp      : 159)   5498906 reads written\n",
            "  0:00:39.414     0M / 299M  INFO    General                 (read_converter.hpp        :  94)   Converting single reads\n",
            "  0:00:39.536     0M / 363M  INFO    General                 (binary_converter.hpp      : 159)   415 reads written\n",
            "  0:00:39.544     0M / 363M  INFO    General                 (graph_construction.hpp    : 120)   Constructing DeBruijn graph for k=21\n",
            "  0:00:39.545     0M / 363M  INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 4 buckets using 2 threads. This might take a while.\n",
            "  0:00:39.545     0M / 363M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:39.545     0M / 363M  INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:00:39.545     0M / 363M  INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 16777216\n",
            "  0:01:02.111     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 3886091 reads\n",
            "  0:01:24.134     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 7773435 reads\n",
            "  0:01:46.161     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 11659710 reads\n",
            "  0:02:11.826     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 15546471 reads\n",
            "  0:02:33.911     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 19433117 reads\n",
            "  0:02:48.624     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 21996454 reads\n",
            "  0:02:48.625     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 297)   Adding contigs from previous K\n",
            "  0:02:48.781     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 310)   Used 21996454 reads. Maximum read length 90\n",
            "  0:02:48.782     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 311)   Average read length 89.9999\n",
            "  0:02:48.782     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:02:49.912     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 6272563 kmers in total.\n",
            "  0:02:49.912     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:02:49.972     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:02:49.973     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:02:49.973     0M / 2G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:02:49.973     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:02:49.973     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 2097152\n",
            "  0:02:52.153     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 381)   Processed 6272563 kmers\n",
            "  0:02:52.154     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 386)   Used 6272563 kmers.\n",
            "  0:02:52.640     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:02:52.874     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 6272518 kmers in total.\n",
            "  0:02:52.874     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:02:52.929     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:02:54.048     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:02:54.119     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 2916896 bytes occupied (3.72022 bits per kmer).\n",
            "  0:02:54.123     0M / 2G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  87)   Building k-mer extensions from k+1-mers\n",
            "  0:02:56.693     0M / 2G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  91)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:02:56.694     0M / 2G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
            "  0:03:01.875     0M / 2G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   310446 22-mers were removed by early tip clipper\n",
            "  0:03:01.876     0M / 2G    INFO    General                 (graph_construction.hpp    : 136)   Condensing graph\n",
            "  0:03:01.876     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:03:05.236     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 32960 sequences extracted\n",
            "  0:03:07.122     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:03:07.773     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 5 loops collected\n",
            "  0:03:07.821     0M / 2G    INFO    General                 (graph_construction.hpp    : 141)   Building index with from graph\n",
            "  0:03:07.821     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:03:07.822     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:03:07.822     0M / 2G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:03:07.822     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:03:07.822     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 2097152\n",
            "  0:03:08.786     0M / 2G    INFO    General                 (edge_index_builders.hpp   :  75)   Processed 65891 edges\n",
            "  0:03:08.787     0M / 2G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 65891 sequences.\n",
            "  0:03:09.237     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:03:09.448     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 5962117 kmers in total.\n",
            "  0:03:09.449     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:03:09.498     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:03:10.510     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:03:10.570     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 2772872 bytes occupied (3.72065 bits per kmer).\n",
            "  0:03:10.748     0M / 2G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\n",
            "  0:03:12.093     0M / 2G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\n",
            "  0:03:12.116     0M / 2G    INFO    General                 (graph_construction.hpp    : 173)   Filling coverage index\n",
            "  0:03:12.116     0M / 2G    INFO    General                 (edge_index_builders.hpp   : 183)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:07:43.310     0M / 2G    INFO    General                 (graph_construction.hpp    : 175)   Filling coverage and flanking coverage from index\n",
            "  0:07:43.711     0M / 2G    INFO    General                 (construction.cpp          :  30)   Figured out: read length = 90\n",
            "  0:07:43.711     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == EC Threshold Finding\n",
            "  0:07:44.144     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 3\n",
            "  0:07:44.144     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 142\n",
            "  0:07:44.144     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 142. Coverage mad: 51.891\n",
            "  0:07:44.144     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:07:44.382     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:07:44.978     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:07:46.583     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:07:49.963     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:07:52.358     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 138.186. Fitted coverage std. dev: 48.8926\n",
            "  0:07:52.364     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.846675\n",
            "  0:07:52.365     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 35\n",
            "  0:07:52.365     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 35\n",
            "  0:07:52.365     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5167636\n",
            "  0:07:52.365     0M / 2G    INFO    General                 (genomic_info_filler.cpp   : 127)   Mean coverage was calculated as 138.186\n",
            "  0:07:52.365     0M / 2G    INFO    General                 (genomic_info_filler.cpp   : 141)   EC coverage threshold value was calculated as 35\n",
            "  0:07:52.365     0M / 2G    INFO    General                 (genomic_info_filler.cpp   : 142)   Trusted kmer low bound: 0\n",
            "  0:07:52.365     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Simplification\n",
            "  0:07:52.402     0M / 2G    INFO    General                 (simplification.cpp        : 387)   Graph simplification started\n",
            "  0:07:52.402     0M / 2G    INFO    General                 (simplification.cpp        :  77)   PROCEDURE == InitialCleaning\n",
            "  0:07:52.402     0M / 2G    INFO    General                 (graph_simplification.hpp  : 645)   Flanking coverage based disconnection disabled\n",
            "  0:07:52.402     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Self conjugate edge remover\n",
            "  0:07:52.411     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.412     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Initial isolated edge remover\n",
            "  0:07:52.445     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 7828 times\n",
            "  0:07:52.446     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Initial tip clipper\n",
            "  0:07:52.458     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 212 times\n",
            "  0:07:52.458     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Initial ec remover\n",
            "  0:07:52.619     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 4107 times\n",
            "  0:07:52.620     0M / 2G    INFO    General                 (graph_simplification.hpp  : 617)   Creating parallel br instance\n",
            "  0:07:52.620     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:07:52.620     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.626     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 20 times\n",
            "  0:07:52.626     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.764     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 2687 times\n",
            "  0:07:52.764     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.771     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 95 times\n",
            "  0:07:52.771     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:07:52.771     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.772     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 1 times\n",
            "  0:07:52.772     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.776     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 40 times\n",
            "  0:07:52.776     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.779     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 79 times\n",
            "  0:07:52.779     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:07:52.779     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.779     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.780     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.784     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 56 times\n",
            "  0:07:52.784     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.785     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 26 times\n",
            "  0:07:52.785     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:07:52.785     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.785     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.786     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.787     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 24 times\n",
            "  0:07:52.787     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.787     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 20 times\n",
            "  0:07:52.787     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:07:52.788     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.788     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.788     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.788     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 9 times\n",
            "  0:07:52.788     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 8 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 1 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 6 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 1 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.789     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 6 times\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 6 times\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:07:52.790     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.793     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.793     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.800     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 5 times\n",
            "  0:07:52.800     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.801     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 5 times\n",
            "  0:07:52.802     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:07:52.802     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.802     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.802     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.802     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 1 times\n",
            "  0:07:52.802     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.802     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.803     0M / 2G    INFO    General                 (simplification.cpp        : 167)   PROCEDURE == Post simplification\n",
            "  0:07:52.804     0M / 2G    INFO    General                 (graph_simplification.hpp  : 447)   Disconnection of relatively low covered edges disabled\n",
            "  0:07:52.804     0M / 2G    INFO    General                 (graph_simplification.hpp  : 477)   Complex tip clipping disabled\n",
            "  0:07:52.804     0M / 2G    INFO    General                 (graph_simplification.hpp  : 617)   Creating parallel br instance\n",
            "  0:07:52.804     0M / 2G    INFO    General                 (simplification.cpp        : 301)   Iteration 1\n",
            "  0:07:52.804     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.805     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.805     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.814     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.814     0M / 2G    INFO    General                 (simplification.cpp        : 301)   Iteration 2\n",
            "  0:07:52.814     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:52.815     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.816     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:52.822     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:52.823     0M / 2G    INFO    General                 (simplification.cpp        : 333)   Disrupting self-conjugate edges\n",
            "  0:07:52.826     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Simplification Cleanup\n",
            "  0:07:52.826     0M / 2G    INFO    General                 (simplification.cpp        : 547)   Removing isolated edges\n",
            "  0:07:52.828     0M / 2G    INFO    General                 (simplification.cpp        : 570)   Counting average coverage\n",
            "  0:07:52.829     0M / 2G    INFO    General                 (simplification.cpp        : 575)   Average coverage = 145.521\n",
            "  0:07:52.829     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Contig Output\n",
            "  0:07:52.830     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /content/3_spades//K21/assembly_graph_with_scaffolds.gfa\n",
            "  0:07:52.864     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K21/before_rr.fasta\n",
            "  0:07:52.901     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  50)   Outputting FastG graph to /content/3_spades//K21/assembly_graph.fastg\n",
            "  0:07:53.002     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K21/simplified_contigs.fasta\n",
            "  0:07:53.041     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K21/final_contigs.fasta\n",
            "  0:07:53.078     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Contig Output\n",
            "  0:07:53.078     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /content/3_spades//K21/assembly_graph_with_scaffolds.gfa\n",
            "  0:07:53.105     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K21/before_rr.fasta\n",
            "  0:07:53.153     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  50)   Outputting FastG graph to /content/3_spades//K21/assembly_graph.fastg\n",
            "  0:07:53.267     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K21/simplified_contigs.fasta\n",
            "  0:07:53.313     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K21/final_contigs.fasta\n",
            "  0:07:53.360     0M / 2G    INFO    General                 (launch.hpp                : 137)   SPAdes finished\n",
            "  0:07:53.371     0M / 2G    INFO    General                 (main.cpp                  : 110)   Assembling time: 0 hours 7 minutes 53 seconds\n",
            "\n",
            "== Running assembler: K33\n",
            "\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (main.cpp                  :  76)   Loading config from /content/3_spades/K33/configs/config.info\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (memory_limit.hpp          :  52)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (main.cpp                  :  89)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (main.cpp                  :  90)   Maximum k-mer length: 128\n",
            "  0:00:00.001     0M / 26M   INFO    General                 (main.cpp                  :  91)   Assembling dataset (/content/3_spades/dataset.info) with K=33\n",
            "  0:00:00.001     0M / 26M   INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
            "  0:00:00.001     0M / 26M   INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
            "  0:00:00.001     0M / 26M   INFO    General                 (launch.hpp                :  61)   Two-step RR enabled: 0\n",
            "  0:00:00.001     0M / 26M   INFO   StageManager             (stage.cpp                 : 126)   STAGE == Construction\n",
            "  0:00:00.001     0M / 26M   INFO    General                 (read_converter.hpp        :  67)   Binary reads detected\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (graph_construction.hpp    : 120)   Constructing DeBruijn graph for k=33\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 4 buckets using 2 threads. This might take a while.\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 8388608\n",
            "  0:00:13.274     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 2353737 reads\n",
            "  0:00:25.489     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 4707248 reads\n",
            "  0:00:37.749     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 7061046 reads\n",
            "  0:00:50.084     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 9414310 reads\n",
            "  0:01:02.431     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 11768136 reads\n",
            "  0:01:14.762     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 14122161 reads\n",
            "  0:01:27.062     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 16476057 reads\n",
            "  0:01:39.944     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 18829567 reads\n",
            "  0:01:52.269     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 21183561 reads\n",
            "  0:01:56.840     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 21996454 reads\n",
            "  0:01:56.841     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 297)   Adding contigs from previous K\n",
            "  0:01:58.310     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 310)   Used 21996454 reads. Maximum read length 90\n",
            "  0:01:58.311     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 311)   Average read length 89.9999\n",
            "  0:01:58.311     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:02:00.845     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 6242403 kmers in total.\n",
            "  0:02:00.845     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:02:01.055     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:02:01.055     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:02:01.056     0M / 2G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:02:01.056     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:02:01.056     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 1048576\n",
            "  0:02:03.559     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 381)   Processed 6242403 kmers\n",
            "  0:02:03.559     0M / 2G    INFO    General                 (kmer_splitters.hpp        : 386)   Used 6242403 kmers.\n",
            "  0:02:03.826     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:02:04.258     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 6246463 kmers in total.\n",
            "  0:02:04.258     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:02:04.361     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:02:06.611     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:02:06.874     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 2904720 bytes occupied (3.72015 bits per kmer).\n",
            "  0:02:06.883     0M / 2G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  87)   Building k-mer extensions from k+1-mers\n",
            "  0:02:09.974     0M / 2G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  91)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:02:09.974     0M / 2G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
            "  0:02:15.967     0M / 2G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   366647 34-mers were removed by early tip clipper\n",
            "  0:02:15.967     0M / 2G    INFO    General                 (graph_construction.hpp    : 136)   Condensing graph\n",
            "  0:02:15.967     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:02:19.944     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 22195 sequences extracted\n",
            "  0:02:22.116     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:02:22.780     0M / 2G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 3 loops collected\n",
            "  0:02:22.830     0M / 2G    INFO    General                 (graph_construction.hpp    : 141)   Building index with from graph\n",
            "  0:02:22.831     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:02:22.831     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:02:22.831     0M / 2G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:02:22.831     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:02:22.832     0M / 2G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 1048576\n",
            "  0:02:23.828     0M / 2G    INFO    General                 (edge_index_builders.hpp   :  75)   Processed 44386 edges\n",
            "  0:02:23.828     0M / 2G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 44386 sequences.\n",
            "  0:02:24.054     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:02:24.438     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 5875756 kmers in total.\n",
            "  0:02:24.438     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:02:24.530     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:02:25.841     0M / 2G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:02:25.946     0M / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 2732920 bytes occupied (3.72094 bits per kmer).\n",
            "  0:02:26.026     0M / 2G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\n",
            "  0:02:27.428     0M / 2G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\n",
            "  0:02:27.491     0M / 2G    INFO    General                 (graph_construction.hpp    : 173)   Filling coverage index\n",
            "  0:02:27.492     0M / 2G    INFO    General                 (edge_index_builders.hpp   : 183)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:07:06.461     0M / 2G    INFO    General                 (graph_construction.hpp    : 175)   Filling coverage and flanking coverage from index\n",
            "  0:07:06.587     0M / 2G    INFO    General                 (construction.cpp          :  30)   Figured out: read length = 90\n",
            "  0:07:06.588     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == EC Threshold Finding\n",
            "  0:07:06.905     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 3\n",
            "  0:07:06.905     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 116\n",
            "  0:07:06.906     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 116. Coverage mad: 44.478\n",
            "  0:07:06.906     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:07:07.063     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:07:07.480     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:07:08.770     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:07:11.407     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:07:12.720     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 114.126. Fitted coverage std. dev: 41.5272\n",
            "  0:07:12.725     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.793703\n",
            "  0:07:12.726     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 29\n",
            "  0:07:12.726     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 29\n",
            "  0:07:12.726     0M / 2G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5176487\n",
            "  0:07:12.726     0M / 2G    INFO    General                 (genomic_info_filler.cpp   : 127)   Mean coverage was calculated as 114.126\n",
            "  0:07:12.726     0M / 2G    INFO    General                 (genomic_info_filler.cpp   : 141)   EC coverage threshold value was calculated as 29\n",
            "  0:07:12.726     0M / 2G    INFO    General                 (genomic_info_filler.cpp   : 142)   Trusted kmer low bound: 0\n",
            "  0:07:12.727     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Simplification\n",
            "  0:07:12.730     0M / 2G    INFO    General                 (simplification.cpp        : 387)   Graph simplification started\n",
            "  0:07:12.730     0M / 2G    INFO    General                 (simplification.cpp        :  77)   PROCEDURE == InitialCleaning\n",
            "  0:07:12.730     0M / 2G    INFO    General                 (graph_simplification.hpp  : 645)   Flanking coverage based disconnection disabled\n",
            "  0:07:12.730     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Self conjugate edge remover\n",
            "  0:07:12.737     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:12.737     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Initial isolated edge remover\n",
            "  0:07:12.765     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 8233 times\n",
            "  0:07:12.765     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Initial tip clipper\n",
            "  0:07:12.776     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 252 times\n",
            "  0:07:12.776     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Initial ec remover\n",
            "  0:07:12.878     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 1903 times\n",
            "  0:07:12.879     0M / 2G    INFO    General                 (graph_simplification.hpp  : 617)   Creating parallel br instance\n",
            "  0:07:12.879     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:07:12.879     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:12.882     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 17 times\n",
            "  0:07:12.882     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.018     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 1671 times\n",
            "  0:07:13.018     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.023     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 38 times\n",
            "  0:07:13.023     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:07:13.023     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.023     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:07:13.023     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.026     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 29 times\n",
            "  0:07:13.026     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.030     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 69 times\n",
            "  0:07:13.030     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:07:13.030     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.030     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:07:13.030     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.037     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 47 times\n",
            "  0:07:13.037     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.039     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 28 times\n",
            "  0:07:13.039     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:07:13.039     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.039     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 1 times\n",
            "  0:07:13.039     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 23 times\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 9 times\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 5 times\n",
            "  0:07:13.041     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.042     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 4 times\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 3 times\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:07:13.043     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.046     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.046     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.048     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 3 times\n",
            "  0:07:13.048     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 167)   PROCEDURE == Post simplification\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (graph_simplification.hpp  : 447)   Disconnection of relatively low covered edges disabled\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (graph_simplification.hpp  : 477)   Complex tip clipping disabled\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (graph_simplification.hpp  : 617)   Creating parallel br instance\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 301)   Iteration 1\n",
            "  0:07:13.049     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.051     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.051     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.053     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.053     0M / 2G    INFO    General                 (simplification.cpp        : 301)   Iteration 2\n",
            "  0:07:13.053     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:07:13.054     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.054     0M / 2G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:07:13.056     0M / 2G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:07:13.056     0M / 2G    INFO    General                 (simplification.cpp        : 333)   Disrupting self-conjugate edges\n",
            "  0:07:13.057     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Simplification Cleanup\n",
            "  0:07:13.057     0M / 2G    INFO    General                 (simplification.cpp        : 547)   Removing isolated edges\n",
            "  0:07:13.058     0M / 2G    INFO    General                 (simplification.cpp        : 570)   Counting average coverage\n",
            "  0:07:13.059     0M / 2G    INFO    General                 (simplification.cpp        : 575)   Average coverage = 119.29\n",
            "  0:07:13.059     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Contig Output\n",
            "  0:07:13.059     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /content/3_spades//K33/assembly_graph_with_scaffolds.gfa\n",
            "  0:07:13.091     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K33/before_rr.fasta\n",
            "  0:07:13.134     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  50)   Outputting FastG graph to /content/3_spades//K33/assembly_graph.fastg\n",
            "  0:07:13.240     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K33/simplified_contigs.fasta\n",
            "  0:07:13.286     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K33/final_contigs.fasta\n",
            "  0:07:13.330     0M / 2G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Contig Output\n",
            "  0:07:13.330     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /content/3_spades//K33/assembly_graph_with_scaffolds.gfa\n",
            "  0:07:13.353     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K33/before_rr.fasta\n",
            "  0:07:13.397     0M / 2G    INFO    General                 (contig_output_stage.cpp   :  50)   Outputting FastG graph to /content/3_spades//K33/assembly_graph.fastg\n",
            "  0:07:13.482     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K33/simplified_contigs.fasta\n",
            "  0:07:13.531     0M / 2G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K33/final_contigs.fasta\n",
            "  0:07:13.566     0M / 2G    INFO    General                 (launch.hpp                : 137)   SPAdes finished\n",
            "  0:07:13.615     0M / 2G    INFO    General                 (main.cpp                  : 110)   Assembling time: 0 hours 7 minutes 13 seconds\n",
            "\n",
            "== Running assembler: K55\n",
            "\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (main.cpp                  :  76)   Loading config from /content/3_spades/K55/configs/config.info\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (memory_limit.hpp          :  52)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (main.cpp                  :  89)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (main.cpp                  :  90)   Maximum k-mer length: 128\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (main.cpp                  :  91)   Assembling dataset (/content/3_spades/dataset.info) with K=55\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (launch.hpp                :  61)   Two-step RR enabled: 0\n",
            "  0:00:00.000     0M / 26M   INFO    General                 (launch.hpp                :  79)   Will need read mapping, kmer mapper will be attached\n",
            "  0:00:00.000     0M / 26M   INFO   StageManager             (stage.cpp                 : 126)   STAGE == Construction\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (read_converter.hpp        :  67)   Binary reads detected\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (graph_construction.hpp    : 120)   Constructing DeBruijn graph for k=55\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 4 buckets using 2 threads. This might take a while.\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:00:00.002     0M / 26M   INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 8388608\n",
            "  0:00:13.768     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 3833260 reads\n",
            "  0:00:26.834     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 7665764 reads\n",
            "  0:00:40.808     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 11499043 reads\n",
            "  0:00:54.079     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 15331448 reads\n",
            "  0:01:07.106     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 19164179 reads\n",
            "  0:01:16.740     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 291)   Processed 21996454 reads\n",
            "  0:01:16.740     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 297)   Adding contigs from previous K\n",
            "  0:01:18.348     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 310)   Used 21996454 reads. Maximum read length 90\n",
            "  0:01:18.348     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 311)   Average read length 89.9999\n",
            "  0:01:18.348     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:01:19.899     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 6099249 kmers in total.\n",
            "  0:01:19.899     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:01:20.072     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:01:20.073     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:01:20.073     0M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:01:20.073     0M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:01:20.073     0M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 1048576\n",
            "  0:01:22.565     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 381)   Processed 6099249 kmers\n",
            "  0:01:22.566     0M / 1G    INFO    General                 (kmer_splitters.hpp        : 386)   Used 6099249 kmers.\n",
            "  0:01:23.405     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:01:24.475     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 6107707 kmers in total.\n",
            "  0:01:24.475     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:01:24.568     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:01:32.483     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:01:32.596     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 2840416 bytes occupied (3.72044 bits per kmer).\n",
            "  0:01:32.599     0M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  87)   Building k-mer extensions from k+1-mers\n",
            "  0:01:34.983     0M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  91)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:01:34.983     0M / 1G    INFO    General                 (graph_construction.hpp    : 136)   Condensing graph\n",
            "  0:01:34.984     0M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:01:38.246     0M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 148417 sequences extracted\n",
            "  0:01:40.183     0M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:01:40.831     0M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:01:41.160     0M / 1G    INFO    General                 (graph_construction.hpp    : 141)   Building index with from graph\n",
            "  0:01:41.160     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:01:41.161     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:01:41.161     0M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:01:41.161     0M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:01:41.161     0M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 1048576\n",
            "  0:01:42.316     0M / 1G    INFO    General                 (edge_index_builders.hpp   :  75)   Processed 296833 edges\n",
            "  0:01:42.317     0M / 1G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 296833 sequences.\n",
            "  0:01:43.327     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:01:44.305     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 6099249 kmers in total.\n",
            "  0:01:44.306     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:01:44.400     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:01:45.443     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:01:45.657     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 2837998 bytes occupied (3.72242 bits per kmer).\n",
            "  0:01:45.819     0M / 1G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\n",
            "  0:01:47.549     0M / 1G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\n",
            "  0:01:47.576     0M / 1G    INFO    General                 (graph_construction.hpp    : 173)   Filling coverage index\n",
            "  0:01:47.576     0M / 1G    INFO    General                 (edge_index_builders.hpp   : 183)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:05:30.407     0M / 1G    INFO    General                 (graph_construction.hpp    : 175)   Filling coverage and flanking coverage from index\n",
            "  0:05:31.182     0M / 1G    INFO    General                 (construction.cpp          :  30)   Figured out: read length = 90\n",
            "  0:05:31.182     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == EC Threshold Finding\n",
            "  0:05:31.492     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 8\n",
            "  0:05:31.492     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 70\n",
            "  0:05:31.492     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 70. Coverage mad: 29.652\n",
            "  0:05:31.492     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:05:31.591     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:05:31.859     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:05:32.757     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:05:34.372     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:05:35.374     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 70.0199. Fitted coverage std. dev: 26.7788\n",
            "  0:05:35.377     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.270451\n",
            "  0:05:35.378     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 20\n",
            "  0:05:35.378     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 20\n",
            "  0:05:35.378     0M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5138731\n",
            "  0:05:35.378     0M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   Mean coverage was calculated as 70.0199\n",
            "  0:05:35.378     0M / 1G    INFO    General                 (genomic_info_filler.cpp   : 141)   EC coverage threshold value was calculated as 20\n",
            "  0:05:35.378     0M / 1G    INFO    General                 (genomic_info_filler.cpp   : 142)   Trusted kmer low bound: 0\n",
            "  0:05:35.378     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Gap Closer\n",
            "  0:05:35.378     0M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
            "  0:05:35.578     0M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
            "  0:05:58.694     0M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 5498906 paired reads\n",
            "  0:05:58.694     0M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
            "  0:05:58.728     0M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
            "  0:05:59.019     0M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 8 gaps after checking 249 candidates\n",
            "  0:05:59.037     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Simplification\n",
            "  0:05:59.084     0M / 1G    INFO    General                 (simplification.cpp        : 387)   Graph simplification started\n",
            "  0:05:59.085     0M / 1G    INFO    General                 (simplification.cpp        :  77)   PROCEDURE == InitialCleaning\n",
            "  0:05:59.085     0M / 1G    INFO    General                 (graph_simplification.hpp  : 645)   Flanking coverage based disconnection disabled\n",
            "  0:05:59.085     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Self conjugate edge remover\n",
            "  0:05:59.128     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:05:59.128     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Initial isolated edge remover\n",
            "  0:05:59.223     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 9900 times\n",
            "  0:05:59.224     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Initial tip clipper\n",
            "  0:06:00.474     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 66987 times\n",
            "  0:06:00.474     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Initial ec remover\n",
            "  0:06:00.486     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 168 times\n",
            "  0:06:00.488     0M / 1G    INFO    General                 (graph_simplification.hpp  : 617)   Creating parallel br instance\n",
            "  0:06:00.488     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:06:00.488     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.491     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 15 times\n",
            "  0:06:00.491     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.855     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 957 times\n",
            "  0:06:00.855     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.857     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.857     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:06:00.857     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.857     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.858     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.858     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.858     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.863     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 40 times\n",
            "  0:06:00.863     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:06:00.863     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.863     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 5 times\n",
            "  0:06:00.864     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.887     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 34 times\n",
            "  0:06:00.888     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.890     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 12 times\n",
            "  0:06:00.890     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:06:00.890     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.890     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.890     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.898     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 4 times\n",
            "  0:06:00.898     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.898     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 3 times\n",
            "  0:06:00.898     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:06:00.899     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.899     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.899     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.899     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.899     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.900     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 3 times\n",
            "  0:06:00.900     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:06:00.900     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.900     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.900     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.901     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:06:00.901     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.902     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:06:00.902     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:06:00.902     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.902     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 2 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 1 times\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:06:00.903     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.904     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.904     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.907     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 5 times\n",
            "  0:06:00.907     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 410)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Low coverage edge remover\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 167)   PROCEDURE == Post simplification\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (graph_simplification.hpp  : 447)   Disconnection of relatively low covered edges disabled\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (graph_simplification.hpp  : 477)   Complex tip clipping disabled\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (graph_simplification.hpp  : 617)   Creating parallel br instance\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 301)   Iteration 1\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.908     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.909     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.909     0M / 1G    INFO    General                 (simplification.cpp        : 301)   Iteration 2\n",
            "  0:06:00.909     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Tip clipper\n",
            "  0:06:00.909     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.909     0M / 1G    INFO    General                 (simplification.cpp        : 356)   Running Bulge remover\n",
            "  0:06:00.910     0M / 1G    INFO    General                 (simplification.cpp        : 358)   Triggered 0 times\n",
            "  0:06:00.910     0M / 1G    INFO    General                 (simplification.cpp        : 333)   Disrupting self-conjugate edges\n",
            "  0:06:00.911     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Gap Closer\n",
            "  0:06:00.911     0M / 1G    INFO    General                 (graph_pack.hpp            :  99)   Index refill\n",
            "  0:06:00.912     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:06:00.912     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:06:00.912     0M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:06:00.912     0M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 2 Gb\n",
            "  0:06:00.912     0M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 1048576\n",
            "  0:06:01.824     0M / 1G    INFO    General                 (edge_index_builders.hpp   :  75)   Processed 2508 edges\n",
            "  0:06:01.824     0M / 1G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 2508 sequences.\n",
            "  0:06:02.846     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:06:03.925     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 5287053 kmers in total.\n",
            "  0:06:03.925     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:06:04.020     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:06:12.652     0M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:06:12.753     0M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 2459920 bytes occupied (3.72218 bits per kmer).\n",
            "  0:06:12.825     0M / 1G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\n",
            "  0:06:14.239     0M / 1G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\n",
            "  0:06:14.240     0M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
            "  0:06:14.241     0M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
            "  0:06:38.819     0M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 5498906 paired reads\n",
            "  0:06:38.820     0M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
            "  0:06:38.821     0M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
            "  0:06:38.826     0M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 10 gaps after checking 125 candidates\n",
            "  0:06:39.644     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Simplification Cleanup\n",
            "  0:06:39.645     0M / 1G    INFO    General                 (simplification.cpp        : 547)   Removing isolated edges\n",
            "  0:06:39.654     0M / 1G    INFO    General                 (simplification.cpp        : 570)   Counting average coverage\n",
            "  0:06:39.655     0M / 1G    INFO    General                 (simplification.cpp        : 575)   Average coverage = 72.7372\n",
            "  0:06:39.656     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Mismatch Correction\n",
            "  0:06:39.656     0M / 1G    INFO    General                 (graph_pack.hpp            : 108)   Normalizing k-mer map. Total 132740 kmers to process\n",
            "  0:06:39.831     0M / 1G    INFO    General                 (graph_pack.hpp            : 110)   Normalizing done\n",
            "  0:07:54.843     0M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 187)   Finished collecting potential mismatches positions\n",
            "  0:07:54.900     0M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 284)   All edges processed\n",
            "  0:07:54.902     0M / 1G    INFO    General                 (mismatch_correction.cpp   :  26)   Corrected 4 nucleotides\n",
            "  0:07:54.902     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Contig Output\n",
            "  0:07:54.902     0M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /content/3_spades//K55/assembly_graph_with_scaffolds.gfa\n",
            "  0:07:54.927     0M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K55/before_rr.fasta\n",
            "  0:07:54.968     0M / 1G    INFO    General                 (contig_output_stage.cpp   :  50)   Outputting FastG graph to /content/3_spades//K55/assembly_graph.fastg\n",
            "  0:07:55.104     0M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K55/simplified_contigs.fasta\n",
            "  0:07:55.154     0M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K55/intermediate_contigs.fasta\n",
            "  0:07:55.196     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Paired Information Counting\n",
            "  0:07:55.201     0M / 1G    INFO    General                 (graph_pack.hpp            : 108)   Normalizing k-mer map. Total 132922 kmers to process\n",
            "  0:07:55.363     0M / 1G    INFO    General                 (graph_pack.hpp            : 110)   Normalizing done\n",
            "  0:07:55.364     0M / 1G    INFO    General                 (pair_info_count.cpp       : 317)   Min edge length for estimation: 58171\n",
            "  0:07:55.365     0M / 1G    INFO    General                 (pair_info_count.cpp       : 328)   Estimating insert size for library #0\n",
            "  0:07:55.365     0M / 1G    INFO    General                 (pair_info_count.cpp       : 176)   Estimating insert size (takes a while)\n",
            "  0:07:55.458     0M / 1G    INFO    General                 (short_read_mapper.hpp     :  88)   Selecting usual mapper\n",
            "  0:07:57.520     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 200000 reads\n",
            "  0:07:57.579     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 400000 reads\n",
            "  0:07:59.652     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 600000 reads\n",
            "  0:07:59.707     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 800000 reads\n",
            "  0:08:01.781     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 1000000 reads\n",
            "  0:08:01.844     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 1200000 reads\n",
            "  0:08:08.072     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 2200000 reads\n",
            "  0:08:18.437     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 4200000 reads\n",
            "  0:08:24.378     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  99)   Total 5498906 reads processed\n",
            "  0:08:24.482     0M / 1G    INFO    General                 (pair_info_count.cpp       : 194)   Edge pairs: 67108864 (rough upper limit)\n",
            "  0:08:24.483     0M / 1G    INFO    General                 (pair_info_count.cpp       : 198)   2636627 paired reads (47.9482% of all) aligned to long edges\n",
            "  0:08:24.515     0M / 1G    INFO    General                 (pair_info_count.cpp       : 351)     Insert size = 469.025, deviation = 14.6939, left quantile = 450, right quantile = 486, read length = 90\n",
            "  0:08:24.786     0M / 1G    INFO    General                 (pair_info_count.cpp       : 368)   Filtering data for library #0\n",
            "  0:08:24.786     0M / 1G    INFO    General                 (short_read_mapper.hpp     :  88)   Selecting usual mapper\n",
            "  0:08:26.924     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 200000 reads\n",
            "  0:08:26.947     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 400000 reads\n",
            "  0:08:29.055     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 600000 reads\n",
            "  0:08:29.108     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 800000 reads\n",
            "  0:08:31.197     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 1000000 reads\n",
            "  0:08:31.219     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 1200000 reads\n",
            "  0:08:37.475     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 2200000 reads\n",
            "  0:08:48.514     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 4200000 reads\n",
            "  0:08:54.290     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  99)   Total 5498906 reads processed\n",
            "  0:08:54.290     0M / 1G    INFO    General                 (pair_info_count.cpp       : 380)   Mapping library #0\n",
            "  0:08:54.290     0M / 1G    INFO    General                 (pair_info_count.cpp       : 382)   Mapping paired reads (takes a while)\n",
            "  0:08:54.291     0M / 1G    INFO    General                 (pair_info_count.cpp       : 275)   Left insert size quantile 450, right insert size quantile 486, filtering threshold 2, rounding threshold 0\n",
            "  0:08:54.297     0M / 1G    INFO    General                 (short_read_mapper.hpp     :  88)   Selecting usual mapper\n",
            "  0:08:56.600     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 200000 reads\n",
            "  0:08:56.601     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 400000 reads\n",
            "  0:08:58.918     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 600000 reads\n",
            "  0:08:58.955     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 800000 reads\n",
            "  0:09:01.267     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 1000000 reads\n",
            "  0:09:01.353     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 1200000 reads\n",
            "  0:09:08.327     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 2200000 reads\n",
            "  0:09:20.254     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  81)   Processed 4200000 reads\n",
            "  0:09:26.938     0M / 1G    INFO    General                 (sequence_mapper_notifier.h:  99)   Total 5498906 reads processed\n",
            "  0:09:27.038     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Distance Estimation\n",
            "  0:09:27.038     0M / 1G    INFO    General                 (distance_estimation.cpp   : 224)   Processing library #0\n",
            "  0:09:27.038     0M / 1G    INFO    General                 (distance_estimation.cpp   : 130)   Weight Filter Done\n",
            "  0:09:27.038     0M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
            "  0:09:27.064     0M / 1G    INFO    General                 (distance_estimation.cpp   :  35)   Filtering info\n",
            "  0:09:27.064     0M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 10687\n",
            "  0:09:27.069     0M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
            "  0:09:27.101     0M / 1G    INFO    General                 (distance_estimation.cpp   : 172)   Refining clustered pair information\n",
            "  0:09:27.105     0M / 1G    INFO    General                 (distance_estimation.cpp   : 174)   The refining of clustered pair information has been finished\n",
            "  0:09:27.105     0M / 1G    INFO    General                 (distance_estimation.cpp   : 176)   Improving paired information\n",
            "  0:09:27.165     0M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 1673; contradictional = 204\n",
            "  0:09:27.222     0M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 78; contradictional = 38\n",
            "  0:09:27.222     0M / 1G    INFO    General                 (distance_estimation.cpp   : 183)   Filling scaffolding index\n",
            "  0:09:27.223     0M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
            "  0:09:27.272     0M / 1G    INFO    General                 (distance_estimation.cpp   :  35)   Filtering info\n",
            "  0:09:27.272     0M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 2523\n",
            "  0:09:27.274     0M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
            "  0:09:27.274     0M / 1G    INFO    General                 (distance_estimation.cpp   : 228)   Clearing raw paired index\n",
            "  0:09:27.309     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Repeat Resolving\n",
            "  0:09:27.309     0M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
            "  0:09:27.309     0M / 1G    INFO    General                 (launcher.cpp              : 478)   ExSPAnder repeat resolving tool started\n",
            "  0:09:27.313     0M / 1G    INFO    General                 (launcher.cpp              : 393)   Creating main extenders, unique edge length = 2000\n",
            "  0:09:27.314     0M / 1G    INFO    General                 (extenders_logic.cpp       : 275)   Estimated coverage of library #0 is 72.7372\n",
            "  0:09:27.317     0M / 1G    INFO    General                 (extenders_logic.cpp       : 275)   Estimated coverage of library #0 is 72.7372\n",
            "  0:09:27.319     0M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
            "  0:09:27.319     0M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
            "  0:09:27.319     0M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
            "  0:09:27.319     0M / 1G    INFO    General                 (launcher.cpp              : 421)   Total number of extenders is 3\n",
            "  0:09:27.319     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 0 paths from 1127 (0%)\n",
            "  0:09:27.427     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 113 paths from 1127 (10%)\n",
            "  0:09:27.435     0M / 1G    INFO    General                 (path_extender.hpp         : 865)   Processed 128 paths from 1127 (11%)\n",
            "  0:09:27.487     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 226 paths from 1127 (20%)\n",
            "  0:09:27.492     0M / 1G    INFO    General                 (path_extender.hpp         : 865)   Processed 256 paths from 1127 (22%)\n",
            "  0:09:27.526     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 339 paths from 1127 (30%)\n",
            "  0:09:27.541     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 452 paths from 1127 (40%)\n",
            "  0:09:27.547     0M / 1G    INFO    General                 (path_extender.hpp         : 865)   Processed 512 paths from 1127 (45%)\n",
            "  0:09:27.548     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 565 paths from 1127 (50%)\n",
            "  0:09:27.553     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 678 paths from 1127 (60%)\n",
            "  0:09:27.558     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 791 paths from 1127 (70%)\n",
            "  0:09:27.567     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 904 paths from 1127 (80%)\n",
            "  0:09:27.567     0M / 1G    INFO    General                 (path_extender.hpp         : 867)   Processed 1017 paths from 1127 (90%)\n",
            "  0:09:27.568     0M / 1G    INFO    General                 (path_extender.hpp         : 865)   Processed 1024 paths from 1127 (90%)\n",
            "  0:09:27.568     0M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
            "  0:09:27.568     0M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
            "  0:09:27.572     0M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
            "  0:09:27.573     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 299)   Removing overlaps\n",
            "  0:09:27.573     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 302)   Sorting paths\n",
            "  0:09:27.573     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 309)   Marking overlaps\n",
            "  0:09:27.573     0M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 132)   Marking start/end overlaps\n",
            "  0:09:27.578     0M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 135)   Marking remaining overlaps\n",
            "  0:09:27.583     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 312)   Splitting paths\n",
            "  0:09:27.586     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 317)   Deduplicating paths\n",
            "  0:09:27.588     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 319)   Overlaps removed\n",
            "  0:09:27.590     0M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
            "  0:09:27.590     0M / 1G    INFO    General                 (launcher.cpp              : 287)   Traversing tandem repeats\n",
            "  0:09:27.595     0M / 1G    INFO    General                 (launcher.cpp              : 297)   Traversed 1 loops\n",
            "  0:09:27.595     0M / 1G    INFO    General                 (launcher.cpp              : 428)   Closing gaps in paths\n",
            "  0:09:27.597     0M / 1G    INFO    General                 (launcher.cpp              : 456)   Gap closing completed\n",
            "  0:09:27.599     0M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
            "  0:09:27.599     0M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
            "  0:09:27.599     0M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
            "  0:09:27.599     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 299)   Removing overlaps\n",
            "  0:09:27.600     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 302)   Sorting paths\n",
            "  0:09:27.600     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 309)   Marking overlaps\n",
            "  0:09:27.600     0M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 132)   Marking start/end overlaps\n",
            "  0:09:27.601     0M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 135)   Marking remaining overlaps\n",
            "  0:09:27.602     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 312)   Splitting paths\n",
            "  0:09:27.602     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 317)   Deduplicating paths\n",
            "  0:09:27.603     0M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 319)   Overlaps removed\n",
            "  0:09:27.604     0M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
            "  0:09:27.605     0M / 1G    INFO    General                 (launcher.cpp              : 531)   ExSPAnder repeat resolving tool finished\n",
            "  0:09:27.619     0M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Contig Output\n",
            "  0:09:27.620     0M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /content/3_spades//K55/assembly_graph_with_scaffolds.gfa\n",
            "  0:09:27.644     0M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/3_spades//K55/before_rr.fasta\n",
            "  0:09:27.680     0M / 1G    INFO    General                 (contig_output_stage.cpp   :  50)   Outputting FastG graph to /content/3_spades//K55/assembly_graph.fastg\n",
            "  0:09:27.765     0M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /content/3_spades//K55/final_contigs.paths\n",
            "  0:09:27.950     0M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /content/3_spades//K55/scaffolds.paths\n",
            "  0:09:28.363     0M / 1G    INFO    General                 (launch.hpp                : 137)   SPAdes finished\n",
            "  0:09:28.623     0M / 1G    INFO    General                 (main.cpp                  : 110)   Assembling time: 0 hours 9 minutes 28 seconds\n",
            "\n",
            "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
            "\n",
            " * Corrected reads are in /content/3_spades/corrected/\n",
            " * Assembled contigs are in /content/3_spades/contigs.fasta\n",
            " * Assembled scaffolds are in /content/3_spades/scaffolds.fasta\n",
            " * Assembly graph is in /content/3_spades/assembly_graph.fastg\n",
            " * Assembly graph in GFA format is in /content/3_spades/assembly_graph_with_scaffolds.gfa\n",
            " * Paths in the assembly graph corresponding to the contigs are in /content/3_spades/contigs.paths\n",
            " * Paths in the assembly graph corresponding to the scaffolds are in /content/3_spades/scaffolds.paths\n",
            "\n",
            "======= SPAdes pipeline finished.\n",
            "\n",
            "SPAdes log can be found here: /content/3_spades/spades.log\n",
            "\n",
            "Thank you for using SPAdes!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bgDWQoF-qAq6"
      },
      "source": [
        "<br><br><hr>\n",
        "Я делал работу в разные дни, и сохранял промежуточные результаты себе на Гугл Диск. Так что следующий кусок кода для меня :)\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0BQ71ReztM_B",
        "outputId": "ad4bd1c8-1b86-4611-9243-39dcefb06ca2"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "VxXRXdd5JVrC"
      },
      "source": [
        "# Заархивировать и скопировать на Гугл Диск\n",
        "# !zip -r /content/3_spades.zip /content/3_spades\n",
        "# !cp 3_spades.zip drive/MyDrive/Desktop/3_spades.zip"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "z4tOfzBZ00DJ"
      },
      "source": [
        "# Вынуть с гугл диска и разархивировать\n",
        "# !cp drive/MyDrive/Desktop/3_spades.zip 3_spades.zip\n",
        "# !unzip 3_spades.zip -qq\n",
        "# !cp -r content/3_spades 3_spades/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "WUqs55iW8KKW"
      },
      "source": [
        "<hr>\n",
        "<br><br>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "P13b3IfRvc7u"
      },
      "source": [
        "Результаты [QUAST](http://cab.cc.spbu.ru/quast/):\n",
        "\n",
        "Для файлов contigs.fasta и scaffolds.fasta (НАДО ЛИ ЗАПУСКАТЬ ДЛЯ ВТОРОГО ФАЙЛА?)\n",
        "\n",
        "\\# contigs (>= 500bp):\t206 || 214 <br>\n",
        "N50:\t105346 ||\t105346 <br>\n",
        "L50:\t15 || 15\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "7UydqQpVze5a"
      },
      "source": [
        "Можно также получить отчёт используя консольный quast:"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "rzh0JJBEqdGb"
      },
      "source": [
        "!mkdir 3_quast_results\n",
        "!quast-5.0.2/quast.py 3_spades/contigs.fasta -o 3_quast_results"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "SIexhg7B0gI2"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "GMTqTp_o0hCj"
      },
      "source": [
        "## **3a. (Optional, 1 extra credit). Effect of read correction.**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "C-KL0bNI0t2r"
      },
      "source": [
        "SPAdes работает в два этапа: исправление ошибок и сборка.<br>\n",
        "В папке 3_spades/corrected лежат исправленные риды.<br>\n",
        "Посмотрим на профиль k-меров (аналогично пункту 2.), но для исправленных ридов."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "ju9xUt4A0fXw"
      },
      "source": [
        "!mkdir 3a_k-mer_corrected"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "B1CqWHar0fal"
      },
      "source": [
        "# jellyfish count\n",
        "# -m, --mer-len=uint32                    *Length of mer\n",
        "# -s, --size=uint64                       *Initial hash size (> avg ecoli genome size in our case)\n",
        "# -C, --canonical                          Count both strand, canonical representation (false)\n",
        "# -o, --output=string                      Output file (mer_counts.jf)\n",
        "\n",
        "!gunzip -k /content/3_spades/corrected/paired_end_470bp_fwd.00.0_0.cor.fastq.gz\n",
        "!jellyfish count -m 31 -C -s 6M 3_spades/corrected/paired_end_470bp_fwd.00.0_0.cor.fastq \\\n",
        "-o 3a_k-mer_corrected/31mer_corrected_fwd.jf"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qWuZN8Ys8iAi"
      },
      "source": [
        "<br>\n",
        "\n",
        "Convert results of *jellyfish count* to .tsv:"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "7oa99fCB0ffz"
      },
      "source": [
        "!jellyfish histo 3a_k-mer_corrected/31mer_corrected_fwd.jf > 3a_k-mer_corrected/31mer_fwd_hist.tsv\n",
        "\n",
        "# left col: a list of the bins (the number of times a k-mer occurs or its ‘depth’)\n",
        "# right col: the count for the number of k-mers in the data that fit into that category"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ym73p4Jn7tgp"
      },
      "source": [
        "<br>\n",
        "\n",
        "Небольшой питоновский код чтобы обработать результаты jellyfish count и построить графики:"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "NnYJRQNN0fib"
      },
      "source": [
        "# read data from files\n",
        "\n",
        "fwd_corr = pd.read_csv('3a_k-mer_corrected/31mer_fwd_hist.tsv', sep=' ', header=None)\n",
        "fwd_corr.columns = ['bins', 'count']"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 590
        },
        "id": "OerRwdxT0fkr",
        "outputId": "21886491-ec8d-4ad1-984d-bf72a1519008"
      },
      "source": [
        "# plot data\n",
        "\n",
        "fwd.plot(x='bins', \n",
        "         y='count', \n",
        "         style='.', \n",
        "         ylim = (0, 0.4e6),\n",
        "         title = '31-mers',\n",
        "         xlabel = 'bins (depth of k-mer)',\n",
        "         ylabel = 'number of k-mers with depth = X',\n",
        "         legend = False)\n",
        "\n",
        "fwd_corr.plot(x='bins', \n",
        "         y='count', \n",
        "         style='.', \n",
        "         ylim = (0, 0.4e6),\n",
        "         title = '31-mers after correction',\n",
        "         xlabel = 'bins (depth of k-mer)',\n",
        "         ylabel = 'number of k-mers with depth = X',\n",
        "         legend = False)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "data": {
            "text/plain": [
              "<matplotlib.axes._subplots.AxesSubplot at 0x7f842eb86690>"
            ]
          },
          "execution_count": 98,
          "metadata": {},
          "output_type": "execute_result"
        },
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZcAAAEWCAYAAACqitpwAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3dfZyVdZ3/8debQQRvuZEIQe6UNbFWhElxtdJuFN02rNxSa6VWpVJ3c2srrH5pd7u1blnulmbqhmaSmSbrasR6k1mBziiCaMSIkhAKwijeLbef3x/X9+A1h5kzZ2bOmZkzvJ+Px/U41/W57r4XB86H6/v9Xt9LEYGZmVkl9evpApiZWd/j5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJWAZJ+LGmtpE2S/ijpnBQfIOlmSU9JCknH93BRzbqFk4tZZfwrMC4i9gPeA3xN0tS07n7gw8AzPVU4AEn9e/L8tntxcjGrgIhYFhGbC4tpOjgitkTEdyLifmB7e8eRdLyk1ZI+K2lduhs6VdIp6Y5oo6TP57bvJ2m2pCckbZB0k6Shad24dLd0tqQ/AXdLGpjusjZIel7Sg5JGVOGPxHZzTi5mFSLp+5JeAf4ArAXu6OShXg8MBEYBXwJ+SHbnMxV4C/D/JI1P2/4DcCrwNuBAoBn4XtHx3gYcBpwEzAT2Bw4ChgEfB17tZDnN2uTkYlYhEXEesC9ZArgF2Fx6jzZtBb4eEVuBucABwHcj4sWIWAY8BhyRtv048IWIWJ3unC4BTiuqArskIl6OiFfTsYcBh0TE9ohojIhNnSynWZucXMwqKP1g3w+MBj5RaltJYyS9VJhyqzZERKEKrXBX8Wxu/avAPml+LHBrquJ6HnicrPotX9X1dG7+emA+MFfSnyX9m6Q9OnKNZuVwcjGrjv7AwaU2iIg/RcQ+hamT53kaODkiBuemgRGxJn+q3Dm3RsSXI2IS8FfAu4GzOnluszY5uZh1kaTXSTpd0j6S6iSdBJwB3JXW7ylpYNp8QGpUV4VOfyXwdUlj07mGS5pRoqwnSHqTpDpgE1k12Y4KlcVsJycXs64Lsiqw1WQN6v8OXBgR89L65WRVWaPIqqReJavOqoTvAvOAX0l6EVgIHF1i+9cDN5MllseBX5NVlZlVlPyyMDMzqzTfuZiZWcVVPbmkOuiHJd2elsdLWiSpSdJPJQ1I8T3TclNaPy53jItSfHmqzy7Ep6dYk6TZuXir5zAzs+7RHXcunySr2y34JnBZRBxCVj99doqfDTSn+GVpOyRNAk4HDgemA99PCauO7GGxk4FJwBlp21LnMDOzblDV5CJpNPDXwNVpWcDbyRoUAeaQPV0MMCMtk9a/I20/A5gbEZsj4kmgCTgqTU0RsTIitpA9bDajnXOYmVk3qPZAdt8BPkv21DJkTwY/HxHb0vJqsh40pM+nASJim6QX0vajyHrA0Mo+TxfFj27nHC1ImgXMAth7772nvuENb+jEJZqZ7b4aGxufi4jhxfGqJRdJ7wbWRURjbx1mPCKuAq4CqK+vj4aGhh4ukZlZbZG0qrV4Ne9cjgXeI+kUskH49iPrkz9YUv90ZzEaKDxJvIZsML3VaVyk/YENuXhBfp/W4htKnMPMzLpB1dpcIuKiiBgdEePIGuTvjogPAfcAp6XNZgK3pfl5aZm0/u7IHsKZB5yeepONByYCDwAPAhNTz7AB6Rzz0j5tncPMzLpBTzzn8jngU5KayNpHrknxa4BhKf4pYDZk78kAbiIbCfaXwPlpcMBtwAVkTzw/DtyUti11DjMz6wZ+Qj9xm4uZWcdJaoyI+uK4n9A3M7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKc3IxM7OKq1pykTRQ0gOSHpG0TNKXU/xHkp6UtDhNk1Ncki6X1CRpiaQpuWPNlLQiTTNz8amSlqZ9LpekFB8qaUHafoGkIdW6TjMz21U171w2A2+PiCOAycB0SdPSus9ExOQ0LU6xk4GJaZoFXAFZogAuBo4GjgIuziWLK4Bzc/tNT/HZwF0RMRG4Ky2bmVk3qVpyicxLaXGPNEWJXWYA16X9FgKDJY0ETgIWRMTGiGgGFpAlqpHAfhGxMCICuA44NXesOWl+Ti5uZmbdoKptLpLqJC0G1pEliEVp1ddT1ddlkvZMsVHA07ndV6dYqfjqVuIAIyJibZp/BhhRqWsyM7P2tZlcJH1a0i7rJQ2TdE05B4+I7RExGRgNHCXpjcBFwBuANwNDgc91quRlSnc1rd4xSZolqUFSw/r166tZDDOz3UqpO5dDgYckHVsISDoPaASWduQkEfE8cA8wPSLWpqqvzcB/kbWjAKwBDsrtNjrFSsVHtxIHeDZVm5E+17VRrqsioj4i6ocPH96RSzIzsxLaTC4RMQs4D/hPSddLehA4DjgmIr7T3oElDZc0OM0PAt4F/CH3oy+ytpBH0y7zgLNSr7FpwAupams+cKKkIakh/0Rgflq3SdK0dKyzgNtyxyr0KpuZi5uZWTfo3876R4EHyHph9QM+nWvLaM9IYI6kurTvTRFxu6S7JQ0HBCwGPp62vwM4BWgCXgE+ChARGyV9FXgwbfeViNiY5s8DfgQMAu5ME8A3gJsknQ2sAj5QZpnNzKwClDVJtLJC+jvgy8APgG8BRwDfA/4I/HNEtFrVVKvq6+ujoaGhw/s1rmpm4coNTJswjKlj/TiNme1eJDVGRH1xvNSdy2nACRGxKi03SjqG7E5jITCh8sWsLY2rmvnQ1QvZsm0HA/r344ZzpjnBmJlRus1lRi6xFGIREVeQPdC421u4cgNbtu1gR8DWbTtYuHJDTxfJzKxX6NRzLhHhfrvAtAnDGNC/H3WCPfr3Y9qEYT1dJDOzXqG9Bn0rYerYIdxwzjS3uZiZFXFy6aKpY4c4qZiZFSmrWkzSG/KfZmZmpZTb5vKTok8zM7M2dbRBX1UphZmZ9Sl+E6WZmVWck4uZmVVcR5NLqZd9mZmZAeUnFxV9mpmZtanc5PKWok8zM7M2lZVcIuKl/KeZmVkpbtA3M7OKc3IxM7OKc3IxM7OKa3fgSknHApcAY9P2Inu1y27/sjAzM2tdOaMiXwP8E9AIbK9ucczMrC8op1rshYi4MyLWRcSGwtTeTpIGSnpA0iOSlkn6coqPl7RIUpOkn0oakOJ7puWmtH5c7lgXpfhySSfl4tNTrEnS7Fy81XOYmVn3aDO5SJoiaQpwj6RLJR1TiKV4ezYDb4+II4DJwHRJ04BvApdFxCFAM3B22v5soDnFL0vbIWkScDpwODAd+L6kOkl1wPeAk4FJwBlpW0qcw8zMukGparFvFS3X5+YDeHupA0dEAIXnYvZIU2G/M1N8Dll7zhXAjDQPcDPwn5KU4nMjYjPwpKQm4Ki0XVNErASQNBeYIenxEucwM7Nu0GZyiYgTACRNKPyAF0gqqzE/3V00AoeQ3WU8ATwfEdvSJquBUWl+FPB0Ovc2SS8Aw1J8Ye6w+X2eLoofnfZp6xzF5ZsFzAIYM2ZMOZdkZmZlKKfN5eZWYj8r5+ARsT0iJgOjye42etWbLCPiqoioj4j64cOH93RxzMz6jDbvXNIrjQ8H9pf0vtyq/YCBHTlJRDwv6R7gGGCwpP7pzmI0sCZttgY4CFgtqT+wP7AhFy/I79NafEOJc5iZWTcodedyKPBuYDDwN7lpCnBueweWNFzS4DQ/CHgX8DhwD3Ba2mwmcFuan5eWSevvTu0284DTU2+y8cBE4AHgQWBi6hk2gKzRf17ap61zmJlZNyjV5nIbcJukYyLi95049khgTmp36QfcFBG3S3oMmCvpa8DDZM/RkD6vTw32G8mSBRGxTNJNwGPANuD8iNgOIOkCYD5QB1wbEcvSsT7XxjnMzKwbKPuPfokNssb77wLTyHp7/R74p+JG/lpXX18fDQ0NPV0MM7OaIqkxIuqL4+U06P8EuInsTuRAssb8GytbPDMz60vKSS57RcT1EbEtTT+mgw36Zma2eylnbLE709Aqc8mqxT4I3CFpKEBEbKxi+czMrAaVk1w+kD4/VhQ/nSzZeHRkMzNrod3kEhHju6MgZmbWd7Tb5iJpL0lflHRVWp4o6d3VL5qZmdWqchr0/wvYAvxVWl4DfK1qJTIzs5pXTnI5OCL+DdgKEBGvkL2N0szMrFXlJJctafiWAJB0MNm7WszMzFpVTm+xi4FfAgdJugE4FvhINQtlZma1rZzeYgskPUQ2/IuAT0bEc1UvmZmZ1axSQ+4Xv8p4bfocI2lMRDxUvWKZmVktK+c1xwPJXnH8CNmdy18CDWTvZjEzM9tFmw36EXFCetXxWmBKemPjVOBI/PItMzMroZzeYodGxNLCQkQ8ChxWvSKZmVmtK6e32BJJVwM/TssfApZUr0hmZlbrykkuHwU+AXwyLd8HXFG1EpmZWc0rpyvy/wGXpcnMzKxd5bS5dIqkgyTdI+kxScskfTLFL5G0RtLiNJ2S2+ciSU2Slks6KRefnmJN6d0yhfh4SYtS/KeSBqT4nmm5Ka0fV63rNDOzXVUtuQDbgE9HxCSyBzDPlzQprbssIian6Q6AtO504HBgOvB9SXWS6oDvAScDk4Azcsf5ZjrWIUAzcHaKnw00p/hlaTszM+smVUsuEbG28KBlRLwIPA6MKrHLDGBuRGyOiCeBJuCoNDVFxMqI2EL2RswZkgS8Hbg57T8HODV3rDlp/mbgHWl7MzPrBuW8z+UvJP1Q0q8k3V2YOnKSVC11JLAohS6QtETStZKGpNgo4OncbqtTrK34MOD5iNhWFG9xrLT+hbR9cblmSWqQ1LB+/fqOXJKZmZVQzp3Lz4CHgC8Cn8lNZZG0D/Bz4MKI2ETW0+xgYDLZA5rfKrF7VUXEVenh0Prhw4f3VDHMzPqccroib4uITnU9lrQHWWK5ISJuAYiIZ3PrfwjcnhbXAAfldh/NayMBtBbfAAyW1D/dneS3LxxrtaT+wP5pezMz6wZt3rlIGippKPDfks6TNLIQS/GSUhvHNcDjEfHtXHxkbrP3Ao+m+XnA6amn13hgIvAA8CAwMfUMG0DW6D8vIgK4Bzgt7T8TuC13rJlp/jTg7rS9mZl1g1J3Lo1kLwgrNITnq8ICmNDOsY8F/g5YKmlxin2erLfX5HSMp4CPAUTEMkk3AY+R9TQ7PyK2A0i6AJgP1AHXRsSydLzPAXMlfQ14mCyZkT6vl9QEbCRLSGZm1k3U3n/oJQ1MD1KWjNW6+vr6aGho6OlimJnVFEmNEVFfHC+nQf93ZcbMzMyA0i8Lez1Zl95Bko7kteqx/YC9uqFsZmZWo0q1uZwEfISsF9a3c/EXydpOzMzMWtVmcomIOcAcSe+PiJ93Y5nMzKzGlaoW+3BE/BgYJ+lTxevz3YvNzMzySlWL7Z0+9+mOgpiZWd9RqlrsB2n2m32t27GZmVVXOcO/PCrpWeA3abo/Il6obrHMzKyWtfucS3onyhnAUuCvgUdyT9ybmZntot07F0mjyYZyeQtwBLAMuL/K5TIzsxpWTrXYn8gGj/yXiPh4lctjZmZ9QDnDvxwJXAecKen3kq6TdHZ7O5mZ2e6r3TuXiHhE0hPAE2RVYx8G3sZrIxCbmZm1UE6bSwOwJ9lglb8B3hoRq6pdMDMzq13ltLmcHBF+wbyZmZWtnK7ITixmZtYh5TTom5mZdYiTi5mZVVy7yUXS30raN81/UdItkqaUsd9Bku6R9JikZZI+meJDJS2QtCJ9DklxSbpcUpOkJflzSJqZtl8haWYuPlXS0rTP5ZJU6hxmZtY9yrlz+X8R8aKk44B3knVBvqKM/bYBn46IScA04HxJk4DZwF0RMRG4Ky0DnAxMTNOswjkkDQUuBo4GjgIuziWLK4Bzc/tNT/G2zmFmZt2gnOSyPX3+NXBVRPwPMKC9nSJibUQ8lOZfBB4ne23yDGBO2mwOcGqanwFcF5mFwGBJI8neiLkgIjZGRDOwAJie1u0XEQsjIsge9Mwfq7VzmJlZNygnuayR9APgg8AdkvYsc7+dJI0je9J/ETAiItamVc8AI9L8KODp3G6rU6xUfHUrcUqco7hcsyQ1SGpYv96d4szMKqWcJPEBYD5wUkQ8DwwFPlPuCSTtA/wcuDAiNuXXpTuOKL+4HVfqHBFxVUTUR0T98OHDq1kMM7PdSsmHKCXVAQ9FxBsKsXRHsLbtvVrsvwdZYrkhIm5J4WcljYyItalqa12KrwEOyu0+OsXWAMcXxe9N8dGtbF/qHGZm1g1K3rlExHZguaQxHT1w6rl1DfB4RHw7t2oeUOjxNRO4LRc/K/Uamwa8kBLZfOBESUNSQ/6JwPy0bpOkaelcZxUdq7VzmJlZNyhn+JchwDJJDwAvF4IR8Z529jsW+Dtgae7lYp8HvgHclEZWXkVW7QZwB3AK0AS8Anw0nWejpK+SDfsP8JWI2JjmzwN+BAwC7kwTJc5hZmbdQFmTRIkNpLe1Fo+IX1elRD2kvr4+GhoaeroYZmY1RVJjRNQXx8sZcv/XksYCEyPifyXtBdRVo5BmZtY3lPOE/rnAzcAPUmgU8ItqFsrMzGpbOV2RzydrP9kEEBErgNdVs1BmZlbbykkumyNiS2FBUn+q/GyKmZnVtnKSy68lfR4YJOldwM+A/65usczMrJaVk1xmA+uBpcDHyLoMf7GahTIzs9pWTm+xHcAP02RmZtaucnqLvVvSw5I2Stok6UVJm9rbz8zMdl/lPKH/HeB9wNJo74lLMzMzymtzeRp41InFzMzKVc6dy2fJ3uPya2BzIVg0GKWZmdlO5SSXrwMvAQMp4w2UZmZm5SSXAyPijVUviZmZ9RnltLncIenEqpfEzMz6jHKSyyeAX0p6NXVDdldkMzMrqZyHKPftjoKYmVnfUc6dy06SLqlSOczMrA/pUHIB2nu1sZmZWYeTi8reULpW0jpJj+Zil0haI2lxmk7JrbtIUpOk5ZJOysWnp1iTpNm5+HhJi1L8p5IGpPieabkprR/XwWs0M7MuKmdssaG5xakpNr6MY/8ImN5K/LKImJymO9LxJgGnA4enfb4vqU5SHfA94GRgEnBG2hbgm+lYhwDNwNkpfjbQnOKXpe3MzKwblXPn8t+S9oNshOT0497u+1wi4j5gY5nlmAHMjYjNEfEk0AQclaamiFiZXlg2F5ghScDbyV6/DDAHODV3rDlp/mbgHWl7MzPrJuUkl38hSzD7SJpK9rKwD3fhnBdIWpKqzYak2CiyMcwKVqdYW/FhwPMRsa0o3uJYaf0LaftdSJolqUFSw/r167twSWZmltducomI/yGrXvoVWVXXeyNicSfPdwVwMDAZWAt8q5PHqYiIuCoi6iOifvjw4T1ZFDOzPqXN51wk/QeQHwl5f+AJsjsPIuIfO3qyiHg2d/wfArenxTXAQblNR6cYbcQ3AIMl9U93J/ntC8daLal/KveGjpbVzMw6r9RDlA1Fy41dPZmkkRGxNi2+Fyj0JJsH/ETSt4EDgYnAA2S90yamDgRryBr9z4yIkHQPcBpZO8xM4LbcsWYCv0/r7/brAszMulebySUi5rS1rhySbgSOBw6QtBq4GDhe0mSyO6KngI+lcy2TdBPwGLANOD8itqfjXADMB+qAayNiWTrF54C5kr4GPAxck+LXANdLaiLrUHB6V67DzMw6Tv5Pfaa+vj4aGopv1szMrBRJjRFRXxzv6EOUZmZm7WozuUi6Pn1+svuKY2ZmfUGpO5epkg4E/l7SEElD81N3FdDMzGpPqd5iVwJ3ARPIeorln3KPFDczM9tFm3cuEXF5RBxG1kNrQkSMz01OLGZm1qZyXhb2CUlHAG9JofsiYkl1i2VmZrWsnFGR/xG4AXhdmm6Q9A/VLpiZmdWudu9cgHOAoyPiZQBJ3yR7+v0/qlkwMzOrXeU85yJge255Ox14aZiZme1+yrlz+S9gkaRb0/KpvDbUipmZ2S7KadD/tqR7geNS6KMR8XBVS2VmZjWtnDsXIuIh4KEql8XMzPoIjy1mZmYV5+RiZmYVVzK5SKpLL+UyMzMrW8nkkl7YtUPS/t1UHjMz6wPKadB/CVgqaQHwciEYEf9YtVKZmVlNKye53JImMzOzsrTboB8Rc4CbgIURMacwtbefpGslrZP0aC42VNICSSvS55AUl6TLJTVJWiJpSm6fmWn7FZJm5uJTJS1N+1wuSaXOYWZm3aecgSv/BlgM/DItT5Y0r4xj/wiYXhSbDdwVERPJ3hUzO8VPBiamaRZwRTrXUOBi4GjgKODiXLK4Ajg3t9/0ds5hZmbdpJyuyJeQ/bA/DxARiynjRWERcR+wsSg8Ayjc9cwhG0qmEL8uMguBwZJGAicBCyJiY0Q0AwuA6WndfhGxMCICuK7oWK2dw8zMukk5yWVrRLxQFNvRyfONiIi1af4ZYESaHwU8ndtudYqViq9uJV7qHLuQNEtSg6SG9evXd+JyzMysNeUkl2WSzgTqJE2U9B/A77p64nTHEV09TlfOERFXRUR9RNQPHz68mkUxM9utlJNc/gE4HNgM3AhsAi7s5PmeTVVapM91Kb4GOCi33egUKxUf3Uq81DnMzKyblNNb7JWI+ALwDuCEiPhCRPxfJ883Dyj0+JoJ3JaLn5V6jU0DXkhVW/OBEyUNSQ35JwLz07pNkqalXmJnFR2rtXOYmVk3afc5F0lvBq4F9k3LLwB/HxGN7ex3I3A8cICk1WS9vr4B3CTpbGAV8IG0+R3AKUAT8ArwUYCI2Cjpq8CDabuvREShk8B5ZD3SBgF3pokS5+h1frLoT3zvnhW8unUHH5g6mtmnHNbTRTIzqwhlzRIlNpCWAOdHxG/S8nHA9yPiL7uhfN2mvr4+Ghoauu18Z12ziPtWPNcidurkA/nO6Ud2WxnMzLpKUmNE1BfHy2lz2V5ILAARcT+wrZKF291cOPfhXRILwC8W/5nGVc09UCIzs8pqs1os95T8ryX9gKwxP4APAvdWv2h9U+OqZn6x+M9trv/5Q6uZOtaDCphZbSvV5vKtouWLc/NV7ULcl93y0OqS6x/2nYuZ9QFtJpeIOKE7C7K7eKid5PH4My/SuKrZdy9mVtPK6S02mKyr77j89h5yv+MaVzXz+DMvtogdsM8AnntpS4vYlb9+gh+etUv7mJlZzSinQf8OssSyFGjMTdZBrVWJfepdhzJ8nwEtYo/9uXi0HTOz2lLO+1wGRsSnql6S3UBxldjYoXtx5tFjuHf5On712LM7439+/v9cNWZmNa2cO5frJZ0raWR6V8rQNBS+dUBrVWKD99oDgI+97WCUiwdZrzEzs1pVTnLZAlwK/J7XqsS672nDPqK1KrEPvnkMAFPHDuHN41repTz34uZuKZeZWTWUUy32aeCQiNj1qT8rW3Hf7TePG8KZR4/ZuTxxxL488NRr1WbD992zm0pmZlZ55dy5FMb7si54ZXPLQQ2mjml5p3L4gfu3WN53z3LyvplZ71TOL9jLwGJJ95ANuw+4K3JHNK5qZt4jLZ/KX7Z2U4vl5le2IF67w/nhb1byrsNf70Z9M6tJ5SSXX6TJOmnhyg3sKKoXO/mNI1ssT5swjH6C7Wm77eGhYMysdrWbXCJiTnvbWGlD9mr5HMupkw9s0d4CWaP+1LFDWrS7uFHfzGpVOU/oP0krY4lFxISqlKgPWlb0UOTebbSnuFHfzPqKcqrF8uOQDAT+FvBzLjmNq5pZuHID0yYMa7Uaa8WzLZ9vWd/GHYkb9c2sryinWmxDUeg7khqBL1WnSLWlcVUzH7p6IVu27WBA/37ccM60FgmmcVUzDUVP5h/Qxh1JcaP+1fc/6UZ9M6tJ5VSLTckt9iO7k/F/qZOFKzewZdsOdgRs3baDhSs3tEgGtzy0ukVjfj/B+6eMbvVY0yYMo66f2JZ22L4jdjmemVktKOc5l2/lpn8FptLF99JLekrSUkmLJTWk2FBJCyStSJ9DUlySLpfUJGlJPtlJmpm2XyFpZi4+NR2/Ke2rXUtRGdMmDGNA/37UCfbo349pE4a1WF9cBVafGu5bM3XsEM45bvzO5WDXzgBmZrWgnGqxar3X5YSip/5nA3dFxDckzU7LnwNOBiam6WjgCuDoNL7ZxWR3UgE0SpoXEc1pm3OBRWSjOk8H7qzGRUwdO4QbzpnWZpvL86+0HE5/cDvJYlPRw5aPeoRkM6tB5VSL7Qm8n13f5/KVCpdlBnB8mp9D9irlz6X4dRERwEJJgyWNTNsuiIiNqZwLgOmS7gX2i4iFKX4dcCpVSi7wWjfiYh1pbykovsWq2i2XmVkVlVMtdhvZD/w2sqf1C1NXBPArSY2SZqXYiIhYm+afAUak+VHA07l9V6dYqfjqVuLdriPtLQXuMWZmfUE5v1yjI2J6hc97XESskfQ6YIGkP+RXRkRI2uXZmkpLiW0WwJgxY9rZuuOKL+Adh41ot3Hew8CYWV9Qzp3L7yS9qZInjYg16XMdcCtwFPBsqu4ifa5Lm68BDsrtPjrFSsVHtxJvrRxXRUR9RNQPHz68q5e1izcW3YWccOjr2t2nMAxMQWEYGDOzWlJOcjmOrLF8eeqttVTSks6eUNLekvYtzAMnAo8C84BCj6+ZZNVxpPhZqdfYNOCFVH02HzhR0pDUs+xEYH5at0nStNRL7KzcsbrVvcvXtVguflK/NVPHDuEdh41oEXO7i5nVmnKqxU6u8DlHALem3sH9gZ9ExC8lPQjcJOlsYBWvdXe+AziF14b+/yhARGyU9FXgwbTdVwqN+8B5wI+AQWQN+VVrzG9L46pm/vfxZ1vEyq3nO/7Q17V47XFxO4yZWW9XTlfkVZU8YUSsBI5oJb4BeEcr8QDOb+NY1wLXthJvAN7Y5cJ2QWca8wuKux/fs3zdLgNdmpn1ZuVUi1kndOThyWLF1WB3/2EdjUVdms3MejMnl27S3sOTee+bMpq6XKv+jjQMjJlZrXBy6SbtPTyZN3XsEM71MDBmVsOcXKqgcVUzd+d6ivWvU9ntLQUeBg/1RW0AAAzfSURBVMbMapmTSxXc8tBqtm1/rTX/7Ye+rsMPQRa/hbKp6J0wZma9mZNLFbT1MrCOKH4LZcOqZjfqm1nNcHKpguKRkDvjfVNGt3hSf4ef1DezGuLkUmGdGQm5NVPHDqG+qCqtuKrMzKy38pC7FdC4qnnn+1y68vBkseLuy5W4IzIz6w5OLl3UuKqZD129kC3bdjCgfz/+clTLoVo68vBksbbaXTxCspn1dq4W66KFKzewZdsOdgRs3baDlc+1fNVNRx6eLOZ2FzOrVU4uXTRtwjAG9O9HnUCC515qWXXVmfaWgtbaXdwl2cxqgZNLF00dO4Qvvftw/uqQA9hn4B67rO9se0tB8Z3Pg0+5S7KZ9X5uc+mixlXNfOX2Zfzf1h27rHvzuM63txQUt7sEWdWY213MrDfznUsXLVy5gc2tJBaA2Scf1uXjv6+VO5/frniuy8c1M6smJ5cuGrLXgFZfAnbY6/etyN3F1LFDOOR1+7SIrdr4Ct+44/EuH9vMrFqcXLqouY1nT46sYLXV3x87fpfYlfetdNuLmfVaTi5dNG3CsFbjXW3Izzvz6DFMGrnvLvEPX73QCcbMeiUnly6aOnYIH3/rhBaxj791QsUb3L966pt2ib26dQfvv+J3nHXNooqey8ysq5S9or7vkTQd+C5QB1wdEd8otX19fX00NDR0+nw/WfQn7nx0LSe/cWTV3nd/4dyH+cXiP5e1rciGntkR2fM3B+wzgAvfeWjVymZmuydJjRFRv0u8LyYXSXXAH4F3AauBB4EzIuKxtvbpanLpLu/61r2sWP9y+xt2UD4Ztfc3oiPb+ti1U5ZI863tV9gGwZ51/di6I9i2PYgUH9i/H9sDduR+T/pJDNoj2/bVzdvZkTtWXT+QRD+JfsDWHcG+A+t4/X6DWP/SZjZv28HowYM4aOheDN93T943ZXSXawMaVzXz84dW89yLmytyzPyYgm0dp1L/6SyUXdClcrd2nHKuo5TdLbkcA1wSESel5YsAIuJf29qnVpILVC/BmPVWA/r348Zzp3XpR/WMq37PltxL/LpyzOIxBW84Z9fj/GTRn/j8rUt3Lv/Le9/UqQRTXPbOlru141zyN4fzlduXlbyO9uxuyeU0YHpEnJOW/w44OiIuKNpuFjArLR4KLO/kKQ8AuvXhk7r9ho+qG7TfCCS1v7VZjQvY/nLzmu0vbXimM7vX7TPs9XV7DxlF/l9LF47Z4ngRsf3l5/9cfJz+Q0dN7Ddg0H6F5R1bXt20beOaFV0ueyfL3dpxdmx9dVO/PQbt99p1NL+w/aWNT3SwiGMjYnhxcLd+Qj8irgKu6upxJDW0lrlrka+ld/K19E596VqgstfTV3uLrQEOyi2PTjEzM+sGfTW5PAhMlDRe0gDgdGBeD5fJzGy30SerxSJim6QLgPlkXZGvjYhlVTxll6vWehFfS+/ka+md+tK1QAWvp0826JuZWc/qq9ViZmbWg5xczMys4pxcukDSdEnLJTVJmt3T5WmPpIMk3SPpMUnLJH0yxYdKWiBpRfockuKSdHm6viWSpvTsFexKUp2khyXdnpbHS1qUyvzT1KEDSXum5aa0flxPlrs1kgZLulnSHyQ9LumYWv1uJP1T+jv2qKQbJQ2sle9G0rWS1kl6NBfr8PcgaWbafoWkmb3oWi5Nf8eWSLpV0uDcuovStSyXdFIu3vHfuojw1ImJrKPAE8AEYADwCDCpp8vVTplHAlPS/L5kQ+RMAv4NmJ3is4FvpvlTgDvJRuyYBizq6Wto5Zo+BfwEuD0t3wScnuavBD6R5s8DrkzzpwM/7emyt3Itc4Bz0vwAYHAtfjfAKOBJYFDuO/lIrXw3wFuBKcCjuViHvgdgKLAyfQ5J80N6ybWcCPRP89/MXcuk9Du2JzA+/b7Vdfa3rsf/ItbqBBwDzM8tXwRc1NPl6uA13EY2/tpyYGSKjQSWp/kfkI3JVth+53a9YSJ7fuku4O3A7ekf+HO5fzg7vyOynoPHpPn+aTv19DXkrmX/9IOsonjNfTcpuTydflj7p+/mpFr6boBxRT/IHfoegDOAH+TiLbbryWspWvde4IY03+I3rPC9dPa3ztVinVf4B1SwOsVqQqp6OBJYBIyIiLVp1TPAiDTf26/xO8BnYeeYiMOA5yNiW1rOl3fntaT1L6Tte4vxwHrgv1I139WS9qYGv5uIWAP8O/AnYC3Zn3UjtfvdQMe/h177/RT5e7I7L6jwtTi57IYk7QP8HLgwIjbl10X2X5Ne3z9d0ruBdRHR2NNlqZD+ZNUXV0TEkcDLZNUvO9XQdzMEmEGWMA8E9gam92ihKqhWvof2SPoCsA24oRrHd3LpvJocYkbSHmSJ5YaIuCWFn5U0Mq0fCaxL8d58jccC75H0FDCXrGrsu8BgSYWHg/Pl3Xktaf3+wIbuLHA7VgOrI6Lw5rebyZJNLX437wSejIj1EbEVuIXs+6rV7wY6/j305u8HSR8B3g18KCVLqPC1OLl0Xs0NMSNJwDXA4xHx7dyqeUChN8tMsraYQvys1CNmGvBCrmqgR0XERRExOiLGkf3Z3x0RHwLuAU5LmxVfS+EaT0vb95r/fUbEM8DTkg5NoXcAj1GD3w1Zddg0SXulv3OFa6nJ7ybp6PcwHzhR0pB0J3diivU4ZS9S/Czwnoh4JbdqHnB66r03HpgIPEBnf+t6stGs1ieyniJ/JOtJ8YWeLk8Z5T2O7HZ+CbA4TaeQ1W/fBawA/hcYmrYX8L10fUuB+p6+hjau63he6y02If2DaAJ+BuyZ4gPTclNaP6Gny93KdUwGGtL38wuyXkY1+d0AXwb+ADwKXE/WA6kmvhvgRrK2oq1kd5Rnd+Z7IGvPaErTR3vRtTSRtaEUfgOuzG3/hXQty4GTc/EO/9Z5+BczM6s4V4uZmVnFObmYmVnFObmYmVnFObmYmVnFObmYmVnFOblYnyJpXH4E2KJ1V0uaVIFzjFQahbmVdfdKqu/kcU/Nl68rx0r7X5pGJr60KH6JpH/u7HE7WZbhkn7Znee0ntUnX3Ns1pqIOKdCh/oU8MMKHSvvVLJBHh+r0PFmkT2Psb1Cx+sUSf0jYr2ktZKOjYjf9mR5rHv4zsX6ov6SblD2TpSbJe0FLe8EJL0k6euSHpG0UNKIFP9bZe8geUTSfW0c//3AL9P2gyTNTee6FRhU2EjSiZJ+L+khST9LY7oh6SlJ/yZpqaQHJB0i6a+A9wCXSlos6eB0mL9N2/xR0luKC5KeDL80lXmppA+m+DxgH6CxEGuNpHMl3SlpUFF8nLJ3fvwonfsGSe+U9Ftl7yc5Km23t7J3hjygbMDNGSn+EUnzJN1N9vAhZA+Gfqitsljf4uRifdGhwPcj4jBgE9n7QortDSyMiCOA+4BzU/xLwEkp/p7indKwGM0RsTmFPgG8ks51MTA1bXcA8EXgnRExhezJ+0/lDvVCRLwJ+E/gOxHxO7IhNT4TEZMj4om0Xf+IOAq4MB2/2PvInuw/gmxMr0sljYyI9wCvpmP9tLU/JEkXkI0vdWpEvNrKJocA3wLekKYzyUZ5+Gfg82mbL5AN13IUcEI6/95p3RTgtIh4W1puAHZJkNY3OblYX/R0rurlx2Q/iMW2kFVBQTYc/Lg0/1vgR5LOJXtJUrGRZEPjF7w1nYOIWEI2dAtkL46aBPxW0mKy8ajG5va7Mfd5TIlrKQwumi9j3nHAjRGxPSKeBX4NvLnE8QrOAk4m+/Hf3MY2T0bE0ojYASwD7opsSI+lubKcCMxO13gv2VAuY9K6BRGxMXe8dWSjJNtuwG0u1hcVj2nU2hhHW+O1sY+2k/4tRMTHJR0N/DVZldLUiMiP0Psq2Q9oe0T243pGGWUsNQZT4Yd/ZxkrZCnZHc9o4ElJBwH/ndZdSVbtl086O3LLO3JlEfD+iFieP3j6M3y56JwDyf78bDfgOxfri8ZIKtwNnAncX+6Okg6OiEUR8SWyO5SDijb5Iy3vIO5L50DSG4G/TPGFwLGSDknr9pb0F7n9Ppj7/H2af5Hs9dMd8Rvgg5LqJA0nu5N6oIz9HgY+BsyTdGBEPJ2q0CZHxJUdOP984B/S6MdIOrLEtn9BNpCl7QacXKwvWg6cL+lxspGFr+jAvpemhvFHgd+RvS98p4h4GXiikDTSsfdJ5/oKWfUVEbGe7L3xN0paQpZA3pA71JAU/yTwTyk2F/hMahg/mPLcSlYV9whwN/DZyIbvb1dE3E/WfvI/qY2oM74K7AEskbQsLbflBOB/OnkeqzEeFdmsgyS9F5gaEV/s5P5PkQ3N/lxFC9bLpd53MyKiuafLYtXnNhezDoqIWyX1tne892qpyu7bTiy7D9+5mJlZxbnNxczMKs7JxczMKs7JxczMKs7JxczMKs7JxczMKu7/A5tKKu/ze3SrAAAAAElFTkSuQmCC\n",
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZcAAAEWCAYAAACqitpwAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deZwV1Zn/8c+XRsCdRluCgIDKaDAZFTqKP03ikiiaRIwxiWaRJEay6IxONjXJjGabiXESE2firhGNa4wLMRrDqFmMgnYrgqjEdkEgRBBacAsKPL8/6lwoLt23bzf3dnOb7/v1qldXPXWq6lRfuE9XnVOnFBGYmZlVUp+eroCZmfU+Ti5mZlZxTi5mZlZxTi5mZlZxTi5mZlZxTi5mZlZxTi5mPUTS9yW9JOnvPV2XniTpm5Iu7+l6WGU5uViPk/RLSYskrZD0V0mfz63rJ+lmSc9LCkkH92BVK0bSLsBXgTER8TZJn5F0f0/Xq9okHSxpQT4WEf8ZEZ9vbxurTU4utin4L2BkRGwHHA18X9K43Pr7gU8BPfoXvqS+FdzdLsDSiFhciZ1VuG7F+67rrmNZ7+HkYj0uIuZExMrCYpp2S+vejIifRsT9wOqO9lX4y1jSNyQtTldEx0g6Kl0VLZP0zVz5PpLOlPSMpKWSbpI0KK0bma6WTpL0AnCvpAHpSmuppJclPSxpcDt1Kez3FUlPSPpwir8PmAbsLOlVSTcCFwMHpOWXU7n+kv5b0guSXpR0saQti87zjHRb7Rft1OFkSU/m6jA2xd8u6Q/pHOZIOjq3zVWSLpJ0p6TXgEPSleMZkmYBr0nqK2m8pAfSPh7LX1VKGiTpF5L+JqlV0m2Stgbuyp33q5J2lnSOpF/mtj061enlVMe359Y9L+lrkmZJWi7pRkkDOvp3YT0gIjx56vEJuBB4nSyxPAJs00aZBcDBHeznYGAV8B/AFsDJwBLgOmBbYC/gDWBUKn8aMB0YBvQHLgGuT+tGpvpcDWwNbAl8AfgNsBVQB4wDtmunLh8Fdib7I+7jwGvAkFw9F+TKfga4v2j784GpwKBU998A/1V0nuemem/ZzvEXAu8CBOwOjEi/lxbgm0A/4FDgFWCPtN1VwHLgwFT3AcDzwExgePo9DAWWAkelMu9Pyw1pH78FbgTq0/He29Z5p9g5wC/T/D+l39P703bfSHXtl9Y/DzyUfq+DgCeBL/b0v19Pbfz77+kKePJUmNKX9UHAt4Et2lhfbnJ5A6hLy9umBLF/rkwzcEyafxI4LLduCPAW0Jd1yWXX3PrPAQ8A/9yF85sJTMzVs93kkpLBa8BuudgBwHO57d8EBpQ43t3AaW3E3012i7FPLnY9cE6avwq4umib54HP5ZbPAK5p43iT0u9wDVDfzudTKrn8O3BTbl0fsgR5cK4en8qt/xFwcU//2/W04eTbYrbJiIjVkd3+GgZ8qaPyknbJ3V55NbdqaUQUbqG9kX6+mFv/BrBNmh8B3JpuwbxMlmxWA/lbXfNz89eQfYnekG75/EjSFu3U70RJM3P7fgewY0fnlTSQXR0157b/XYoXLImIf5TYx3DgmTbiOwPzI2JNLjaP7GqkYD4bysdGAB8t1C3V7yCyxDIcWBYRrSXq1p6dU10ASHWcX1S3fNvb66z7LG0T4oY52xT1JbW5lBIRL7DxXyzzyf4i/0vxCkkjC4fKHfMt4DvAd9L6O4G5wBVF244ALgMOAx6MiNWSZpJdkbSleHjyl8iS4F4RsbDMbYrNp+3f49+A4ZL65BLMLsBfO9h3Pjaf7Mrl5OJCkoYAgyQNjIiXO1nnvwHvzO1LZMmqvd+BbaJ85WI9StJOko6XtI2kOklHACcA9+TK9M812vZLjertfUl31sXAD1IyQFKDpIkl6nuIpHcq60G1guwW2po2im5N9kW6JG33WbIrl/a8CAyT1A/W/sV+GXC+pJ3SPoam30+5Lge+JmmcMrun85xB9hf/NyRtkRriPwTc0Il9/xL4kKQj0uc2IHUyGBYRi8ga7i+UVJ+O8Z7cee4gaft29nsT8AFJh6Urwq8CK8luRVoNcXKxnhZkt8AWAK3AfwOnR8TUXJm5ZH/FDyW7JfUG2W2ZSvgZWaP57yW9Qta4v3+J8m8DbiZLLE8CfyS7VbaeiHgC+DHwINkX6juBDa6Ocu4F5gB/l/RSip1B1pg9XdIK4P+APco9sYj4FfADss4MrwC3AYMi4k2yZHIk2RXShcCJEfFUJ/Y9H5hI1ilgCdmVzNdZ953yabLE+xSwGDg9bfcUWfvOs+l22s5F+51L1u38f1LdPgR8KNXZaohSo5iZmVnF+MrFzMwqrurJJd2PfVTSHWl5lKQZklrSA1D9Urx/Wm5J60fm9nFWis/N33OWNCHFWiSdmYu3eQwzM+se3XHlchrZvemCc4HzI2J3snvsJ6X4SUBrip+fyiFpDHA82cNvE8gaCetSg+rPye4bjwFOSGVLHcPMzLpBVZOLpGHAB8h6rRS6FR5K1iAKMAU4Js1PTMuk9Yel8hOBGyJiZUQ8R9bAuV+aWiLi2dTYdwMwsYNjmJlZN6j2cy4/JRu+Ydu0vAPwckSsSssLWPdw1FDSQ1oRsUrS8lR+KFkPHtrYZn5RfP8OjrEeSZOByQBbb731uD333LMLp2hmtvlqbm5+KSIaiuPVHEn1g8DiiGjWJjpMekRcClwK0NjYGE1NTT1cIzOz2iJpXlvxal65HAgcLekosoHvtiN7pmCgpL7pymIY6568XUj2JO4CZUN6b082EF4hXpDfpq340hLHMDOzblC1NpeIOCsihkXESLIG+Xsj4pPAfcBxqdgk4PY0PzUtk9bfG9lDOFOB41NvslHAaLJRUR8GRqeeYf3SMaambdo7hpmZdYOeeM7lDOArklrI2kcKYzJdQTYsRAvwFeBMyN71QTYkxBNkA/edkgY4XAWcSvbE9pNkI6nO6eAYZmbWDfyEfuI2FzOzzpPUHBGNxXE/oW9mZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhVXteQiaYCkhyQ9JmmOpO+k+FWSnpM0M037pLgkXSCpRdIsSWNz+5ok6ek0TcrFx0manba5QJJSfJCkaan8NEn11TpPMzPbUDWvXFYCh0bE3sA+wARJ49O6r0fEPmmamWJHAqPTNBm4CLJEAZwN7A/sB5ydSxYXASfntpuQ4mcC90TEaOCetGxmZt2kasklMq+mxS3SFCU2mQhcnbabDgyUNAQ4ApgWEcsiohWYRpaohgDbRcT0iAjgauCY3L6mpPkpubiZmXWDqra5SKqTNBNYTJYgZqRVP0i3vs6X1D/FhgLzc5svSLFS8QVtxAEGR8SiNP93YHClzsnMzDrWbnKR9FVJG6yXtIOkK8rZeUSsjoh9gGHAfpLeAZwF7Am8CxgEnNGlmpcpXdW0ecUkabKkJklNS5YsqWY1zMw2K6WuXPYAHpF0YCEg6ctAMzC7MweJiJeB+4AJEbEo3fpaCfyCrB0FYCEwPLfZsBQrFR/WRhzgxXTbjPRzcTv1ujQiGiOisaGhoTOnZGZmJbSbXCJiMvBl4H8lXSPpYeAg4ICI+GlHO5bUIGlgmt8SeD/wVO5LX2RtIY+nTaYCJ6ZeY+OB5enW1t3A4ZLqU0P+4cDdad0KSePTvk4Ebs/tq9CrbFIubmZm3aBvB+sfBx4i64XVB/hqri2jI0OAKZLq0rY3RcQdku6V1AAImAl8MZW/EzgKaAFeBz4LEBHLJH0PeDiV+25ELEvzXwauArYE7koTwA+BmySdBMwDPlZmnc3MrAKUNUm0sUL6NPAd4BLgx8DewM+BvwJfi4g2bzXVqsbGxmhqaurpapiZ1RRJzRHRWBwvdeVyHHBIRMxLy82SDiC70pgO7Fr5apqZWW/QbnKJiIltxAK4SNLNVa2VmZnVtC495xIR7rdrZmbt8sCVZmZWcU4uZmZWcWUlF0l75n+amZmVUu6Vy3VFP83MzNrV2dtiqkotzMysV3Gbi5mZVZyTi5mZVVxnk0upl32ZmZkB5ScXFf00MzNrV7nJ5d1FP83MzNpVVnKJiFfzP83MzEpxg76ZmVWck4uZmVWck4uZmVVcR685RtKBwDnAiFReZK928cvCzMysTR0mF+AK4N+AZmB1datjZma9QTm3xZZHxF0RsTgilhamjjaSNEDSQ5IekzRH0ndSfJSkGZJaJN0oqV+K90/LLWn9yNy+zkrxuZKOyMUnpFiLpDNz8TaPYWZm3aPd5CJprKSxwH2SzpN0QCGW4h1ZCRwaEXsD+wATJI0HzgXOj4jdgVbgpFT+JKA1xc9P5ZA0Bjge2AuYAFwoqU5SHfBz4EhgDHBCKkuJY5iZWTcodVvsx0XLjbn5AA4tteOICKDwXMwWaSps94kUn0LWnnMRMDHNA9wM/K8kpfgNEbESeE5SC7BfKtcSEc8CSLoBmCjpyRLHMDOzbtBucomIQwAk7Vr4Ai+QVFZjfrq6aAZ2J7vKeAZ4OSJWpSILgKFpfigwPx17laTlwA4pPj232/w284vi+6dt2jtGcf0mA5MBdtlll3JOyczMylBOm8vNbcR+Vc7OI2J1ROwDDCO72tik3mQZEZdGRGNENDY0NPR0dczMeo12r1zSK433AraXdGxu1XbAgM4cJCJelnQfcAAwUFLfdGUxDFiYii0EhgMLJPUFtgeW5uIF+W3aii8tcQwzM+sGpa5c9gA+CAwEPpSbxgInd7RjSQ2SBqb5LYH3A08C9wHHpWKTgNvT/NS0TFp/b2q3mQocn3qTjQJGAw8BDwOjU8+wfmSN/lPTNu0dw8zMukGpNpfbgdslHRARD3Zh30OAKandpQ9wU0TcIekJ4AZJ3wceJXuOhvTzmtRgv4wsWRARcyTdBDwBrAJOiYjVAJJOBe4G6oArI2JO2tcZ7RzDzMy6gbI/9EsUyBrvfwaMJ+vt9SDwb8WN/LWusbExmpqaeroaZmY1RVJzRDQWx8tp0L8OuInsSmRnssb86ytbPTMz603KSS5bRcQ1EbEqTb+kkw36Zma2eSlnbLG70tAqN5DdFvs4cKekQQARsayK9TMzsxpUTnL5WPr5haL48WTJxqMjm5nZejpMLhExqjsqYmZmvUeHbS6StpL0bUmXpuXRkj5Y/aqZmVmtKqdB/xfAm8D/S8sLge9XrUZmZlbzykkuu0XEj4C3ACLidbK3UZqZmbWpnOTyZhq+JQAk7Ub2rhYzM7M2ldNb7Gzgd8BwSdcCBwKfqWalzMystpXTW2yapEfIhn8RcFpEvFT1mpmZWc0qNeR+8auMF6Wfu0jaJSIeqV61zMyslpXzmuMBZK84fozsyuWfgSayd7OYmZltoN0G/Yg4JL3qeBEwNr2xcRywL375lpmZlVBOb7E9ImJ2YSEiHgfeXr0qmZlZrSunt9gsSZcDv0zLnwRmVa9KZmZW68pJLp8FvgSclpb/BFxUtRqZmVnNK6cr8j+A89NkZmbWoXLaXLpE0nBJ90l6QtIcSael+DmSFkqamaajctucJalF0lxJR+TiE1KsJb1bphAfJWlGit8oqV+K90/LLWn9yGqdp5mZbahqyQVYBXw1IsaQPYB5iqQxad35EbFPmu4ESOuOB/YCJgAXSqqTVAf8HDgSGAOckNvPuWlfuwOtwEkpfhLQmuLnp3JmZtZNqpZcImJR4UHLiHgFeBIYWmKTicANEbEyIp4DWoD90tQSEc9GxJtkb8ScKEnAocDNafspwDG5fU1J8zcDh6XyZmbWDcp5n8s/SbpM0u8l3VuYOnOQdFtqX2BGCp0qaZakKyXVp9hQYH5uswUp1l58B+DliFhVFF9vX2n98lS+uF6TJTVJalqyZElnTsnMzEoo58rlV8AjwLeBr+emskjaBvg1cHpErCDrabYbsA/ZA5o/LrF5VUXEpenh0MaGhoaeqoaZWa9TTlfkVRHRpa7HkrYgSyzXRsQtABHxYm79ZcAdaXEhMDy3+TDWjQTQVnwpMFBS33R1ki9f2NcCSX2B7VN5MzPrBu1euUgaJGkQ8BtJX5Y0pBBL8ZJSG8cVwJMR8ZNcfEiu2IeBx9P8VOD41NNrFDAaeAh4GBideob1I2v0nxoRAdwHHJe2nwTcntvXpDR/HHBvKm9mZt2g1JVLM9kLwgoN4flbYQHs2sG+DwQ+DcyWNDPFvknW22uftI/ngS8ARMQcSTcBT5D1NDslIlYDSDoVuBuoA66MiDlpf2cAN0j6PvAoWTIj/bxGUguwjCwhmZlZN1FHf9BLGpAepCwZq3WNjY3R1NTU09UwM6spkpojorE4Xk6D/gNlxszMzIDSLwt7G1mX3i0l7cu622PbAVt1Q93MzKxGlWpzOQL4DFkvrJ/k4q+QtZ2YmZm1qd3kEhFTgCmSPhIRv+7GOpmZWY0rdVvsUxHxS2CkpK8Ur893LzYzM8srdVts6/Rzm+6oiJmZ9R6lbotdkmbP7W3djs3MrLrKGf7lcUkvAn9O0/0Rsby61TIzs1rW4XMu6Z0oJwCzgQ8Aj+WeuDczM9tAh1cukoaRDeXybmBvYA5wf5XrZWZmNayc22IvkA0e+Z8R8cUq18fMzHqBcoZ/2Re4GviEpAclXS3ppI42MjOzzVeHVy4R8ZikZ4BnyG6NfQp4L+tGIDYzM1tPOW0uTUB/ssEq/wy8JyLmVbtiZmZWu8ppczkyIvyCeTMzK1s5XZGdWMzMrFPKadA3MzPrFCcXMzOruA6Ti6SPSto2zX9b0i2Sxpax3XBJ90l6QtIcSael+CBJ0yQ9nX7Wp7gkXSCpRdKs/DEkTUrln5Y0KRcfJ2l22uYCSSp1DDMz6x7lXLn8e0S8Iukg4H1kXZAvKmO7VcBXI2IMMB44RdIY4EzgnogYDdyTlgGOBEanaXLhGJIGAWcD+wP7AWfnksVFwMm57SakeHvHMDOzblBOclmdfn4AuDQifgv062ijiFgUEY+k+VeAJ8lemzwRmJKKTQGOSfMTgasjMx0YKGkI2Rsxp0XEsohoBaYBE9K67SJiekQE2YOe+X21dQwzM+sG5SSXhZIuAT4O3Cmpf5nbrSVpJNmT/jOAwRGxKK36OzA4zQ8F5uc2W5BipeIL2ohT4hjF9ZosqUlS05Il7hRnZlYp5SSJjwF3A0dExMvAIODr5R5A0jbAr4HTI2JFfl264ojyq9t5pY4REZdGRGNENDY0NFSzGmZmm5WSD1FKqgMeiYg9C7F0RbCo/a3W234LssRybUTcksIvShoSEYvSra3FKb4QGJ7bfFiKLQQOLor/IcWHtVG+1DHMzKwblLxyiYjVwFxJu3R2x6nn1hXAkxHxk9yqqUChx9ck4PZc/MTUa2w8sDwlsruBwyXVp4b8w4G707oVksanY51YtK+2jmFmZt2gnOFf6oE5kh4CXisEI+LoDrY7EPg0MDv3crFvAj8EbkojK88ju+0GcCdwFNACvA58Nh1nmaTvkQ37D/DdiFiW5r8MXAVsCdyVJkocw8zMuoGyJokSBaT3thWPiD9WpUY9pLGxMZqamnq6GmZmNUVSc0Q0FsfLGXL/j5JGAKMj4v8kbQXUVaOSZmbWO5TzhP7JwM3AJSk0FLitmpUyM7PaVk5X5FPI2k9WAETE08BO1ayUmZnVtnKSy8qIeLOwIKkvVX42xczMals5yeWPkr4JbCnp/cCvgN9Ut1pmZlbLykkuZwJLgNnAF8i6DH+7mpUyM7PaVk5vsTXAZWkyMzPrUDm9xT4o6VFJyyStkPSKpBUdbWdmZpuvcp7Q/ylwLDA7Onri0szMjPLaXOYDjzuxmJlZucq5cvkG2Xtc/gisLASLBqM0MzNbq5zk8gPgVWAAZbyB0szMrJzksnNEvKPqNTEzs16jnDaXOyUdXvWamJlZr1FOcvkS8DtJb6RuyO6KbGZmJZXzEOW23VERMzPrPcq5cllL0jlVqoeZmfUinUouQEevNjYzM+t0clHZBaUrJS2W9Hgudo6khZJmpumo3LqzJLVImivpiFx8Qoq1SDozFx8laUaK3yipX4r3T8staf3ITp6jmZltpHLGFhuUWxyXYqPK2PdVwIQ24udHxD5pujPtbwxwPLBX2uZCSXWS6oCfA0cCY4ATUlmAc9O+dgdagZNS/CSgNcXPT+XMzKwblXPl8htJ20E2QnL6cu/wfS4R8SdgWZn1mAjcEBErI+I5oAXYL00tEfFsemHZDcBESQIOJXv9MsAU4Jjcvqak+ZuBw1J5MzPrJuUkl/8kSzDbSBpH9rKwT23EMU+VNCvdNqtPsaFkY5gVLEix9uI7AC9HxKqi+Hr7SuuXp/IbkDRZUpOkpiVLlmzEKZmZWV6HySUifkt2e+n3ZLe6PhwRM7t4vIuA3YB9gEXAj7u4n4qIiEsjojEiGhsaGnqyKmZmvUq7z7lI+h8gPxLy9sAzZFceRMS/dvZgEfFibv+XAXekxYXA8FzRYSlGO/GlwEBJfdPVSb58YV8LJPVN9V7a2bqamVnXlXqIsqlouXljDyZpSEQsSosfBgo9yaYC10n6CbAzMBp4iKx32ujUgWAhWaP/JyIiJN0HHEfWDjMJuD23r0nAg2n9vX5dgJlZ92o3uUTElPbWlUPS9cDBwI6SFgBnAwdL2ofsiuh54AvpWHMk3QQ8AawCTomI1Wk/pwJ3A3XAlRExJx3iDOAGSd8HHgWuSPErgGsktZB1KDh+Y87DzMw6T/6jPtPY2BhNTcUXa2ZmVoqk5ohoLI539iFKMzOzDrWbXCRdk36e1n3VMTOz3qDUlcs4STsDn5NUL2lQfuquCpqZWe0p1VvsYuAeYFeynmL5p9wjxc3MzDbQ7pVLRFwQEW8n66G1a0SMyk1OLGZm1q5yXhb2JUl7A+9OoT9FxKzqVsvMzGpZOaMi/ytwLbBTmq6V9C/VrpiZmdWuDq9cgM8D+0fEawCSziV7+v1/qlkxMzOrXeU85yJgdW55NZ14aZiZmW1+yrly+QUwQ9KtafkY1g21YmZmtoFyGvR/IukPwEEp9NmIeLSqtTIzs5pWzpULEfEI8EiV62JmZr2ExxYzM7OKc3IxM7OKK5lcJNWll3KZmZmVrWRySS/sWiNp+26qj5mZ9QLlNOi/CsyWNA14rRCMiH+tWq3MzKymlZNcbkmTmZlZWTps0I+IKcBNwPSImFKYOtpO0pWSFkt6PBcbJGmapKfTz/oUl6QLJLVImiVpbG6bSan805Im5eLjJM1O21wgSaWOYWZm3aecgSs/BMwEfpeW95E0tYx9XwVMKIqdCdwTEaPJ3hVzZoofCYxO02TgonSsQcDZwP7AfsDZuWRxEXBybrsJHRzDzMy6STldkc8h+2J/GSAiZlLGi8Ii4k/AsqLwRKBw1TOFbCiZQvzqyEwHBkoaAhwBTIuIZRHRCkwDJqR120XE9IgI4OqifbV1DDMz6yblJJe3ImJ5UWxNF483OCIWpfm/A4PT/FBgfq7cghQrFV/QRrzUMTYgabKkJklNS5Ys6cLpmJlZW8pJLnMkfQKokzRa0v8AD2zsgdMVR2zsfjbmGBFxaUQ0RkRjQ0NDNatiZrZZKSe5/AuwF7ASuB5YAZzexeO9mG5pkX4uTvGFwPBcuWEpVio+rI14qWOYmVk3Kae32OsR8S3gMOCQiPhWRPyji8ebChR6fE0Cbs/FT0y9xsYDy9OtrbuBwyXVp4b8w4G707oVksanXmInFu2rrWOYmVk36fA5F0nvAq4Etk3Ly4HPRURzB9tdDxwM7ChpAVmvrx8CN0k6CZgHfCwVvxM4CmgBXgc+CxARyyR9D3g4lftuRBQ6CXyZrEfalsBdaaLEMTY5zfNamf7sUsbvugPjRrjHtJn1HsqaJUoUkGYBp0TEn9PyQcCFEfHP3VC/btPY2BhNTU3ddrzrZrzAt26bTeHXP2bItnzvmHc6yZhZTZHUHBGNxfFy2lxWFxILQETcD6yqZOU2N83zWvnmresSC8ATi17huIseoHlea89VzMysQtpNLpLGpifl/yjpEkkHS3qvpAuBP3RbDXuhf79tdpvxAH5415PdWxkzsyoo1eby46Lls3PzVe1C3JtdN+MFnlj0SrvrH36+leZ5rb49ZmY1rd3kEhGHdGdFNhc3PvxCh2V+/cgCJxczq2nl9BYbSNbVd2S+vIfc75qXX3+rwzIvvbKyG2piZlY95Qy5fycwHZhN14d9MbJbYvOWvb5e7P1jBrP89Td56Pl1Dfkvv/5md1fNzKyiykkuAyLiK1WvyWbgrscXrbcs4Ivv3Y1bHlmwXnJpmud2FzOrbeV0Rb5G0smShqR3pQxKQ+FbJ6lo+d2jd2TciHqOHTuMPrmVayJrdzEzq1XlJJc3gfOAB4HmNHXf04a9RPO8Vv7c8tJ6sUKXu3Ej6mksukpxu4uZ1bJybot9Fdg9Il7qsKS165ZHFlA8GMKR7xiydn7gVv3WW+d2FzOrZeVcuRTG+7KNsKToSuTtb9uWT+y/y9rlhm37r7e+0O5iZlaLykkurwEz01P6FxSmalest9u36DaY213MrDcp57bYbWmyLmqe18q9T724XuwdO2+/3vK4EfW87+2D+f0T68oVdwAwM6sVHSaXiJjSURkrbfqzS1lV9IRQaxttKgfvsdN6yWWvogRkZlYrOrwtJuk5Sc8WT91Rud7ilTfWfyq/ro8Yv+sOG5R7/G/L11u+b65fomlmtamc22L5cfoHAB8F/JxLJ/zfU+sniVE7bNXmA5LFt8HufWqxH6Y0s5pUzmuOl+amhRHxU+AD3VC3XqF5XivPLH51vdiohm3aLHvs2GHU5Vr116wJpj+7tKr1MzOrhnIGrhybW+xDdiVTzhWPkZ5vyS0Xhnxpy7gR9Zx80Cgu/lN21zGA+qLnX8zMakE5SSL/XpdVwPNs5HvpJT0PvAKsBlZFRGMaUuZGstGXnwc+FhGtkgT8DDiK7Hmbz0TEI2k/k4Bvp91+v9D5QNI44CpgS7KBN0+Ljt7nXCXFz7e8a2R9ydtcK1au/5LP4nYYM7NaUE5vsTWp8SAAAA+2SURBVGq91+WQoqf+zwTuiYgfSjozLZ8BHAmMTtP+wEXA/ikZnU12JRVAs6SpEdGaypwMzCBLLhOAu6p0HiXtWPRw5O6Dty1Zvrjdxd2RzawWlXNbrD/wETZ8n8t3K1yXicDBaX4K2auUz0jxq9OVx3RJAyUNSWWnRcSyVM9pwARJfwC2i4jpKX41cAw9lFy267/+r7j4+ZZixd2Pt+3vO5BmVnvKeUL/drIv+FVkT+sXpo0RwO8lNUuanGKDI6IwJv3fgcFpfigwP7ftghQrFV/QRrzbNc9r5bL7n1u7LNp+viWv9fU317taufz+5zwMjJnVnHL+LB4WERMqfNyDImKhpJ2AaZKeyq+MiJBU9TaSlNgmA+yyyy4dlO68Wx5ZwOo1606jTzvPt+SN33UH6vqIVWm71anHmLsjm1ktKefK5QFJ76zkQSNiYfq5GLgV2A94Md3uIv0sPByyEBie23xYipWKD2sj3lY9Lo2IxohobGho2NjT2nD/RcuH7rlTh0li3Ih6Pn/QqPX24R5jZlZrykkuB5E1ls+VNEvSbEmzunpASVtL2rYwDxwOPA5MBSalYpPIbseR4icqMx5Ynm6f3Q0cLqleUn3az91p3QpJ41NPsxNz++pWxe0tu+24dVnbuceYmdW6cm6LHVnhYw4Gbs2+9+kLXBcRv5P0MHCTpJOAeazr7nwnWTfkwtD/nwWIiGWSvgc8nMp9t9C4D3yZdV2R76KHGvMfLHoAcs6iFWVtV/yiML84zMxqTTldkedV8oAR8SywdxvxpcBhbcQDOKWdfV0JXNlGvAl4x0ZXdiM0z2vd4Ioj/3KwUorf7WJmVmvKuS1mXZA15q9b3m9k/XovByvl2LHD6Fu3rs/YvXMXu8eYmdUUJ5cqKX4yv/g1xqWMG1HPoXvstHZ51erwi8PMrKY4uVTJyx08z9JZbncxs1ri5FIFzfNaN7iNVTwMTGdVOlmZmVWTk0sVTH92KatzD7nUCT4ydlj7G7ShuFG/qY2EZWa2qXJyqYLiN09+aO+dO/2E/bFjh5F7tQtrAre7mFnNcHKpguI3Tz73UueHYhs3op7GooTkdhczqxVOLhXW1psnd9puQJf2VdzDzO0uZlYrnFwqrDNvnuyI213MrFY5uVRY8WCV7xszuMsjGrvdxcxqlZNLhb1eNOhkuYNVtsXtLmZWq5xcKqh5Xiu3z/zberFyB6tsj9tdzKwWOblspOZ5rfz8vhaa57Vy7l1PbnBbrNzBKttT3O7y8PNudzGzTZ9f0L4Rmue18snLp/PmqjX07SPeXL1+atlxm35lD1bZnmPHDuO6GS+sTVoBXPzHZ7jsxMaN2q+ZWTX5ymUjTH92KW+uWsOagLdWb/hW5n132fhXE48bUc/O9VuuF3vouaXtlDYz2zQ4uWyE8bvuQL++2a9ww9TS9S7IxfYast16y8vfWMXpNzxakX2bmVWDk8tGGDeingl7va3Ndbs3bN3lLsjFvtBGkrpt5t/c9mJmmywnl400/dm2b1F97qBdK3aMcSPqGTNk2w3in7p8uhOMmW2SnFw20i6Dttogts+w7Te6Ib/Y94555waxN95aw0cueoATr5hR0WOZmW0sZa+o730kTQB+BtQBl0fED0uVb2xsjKampk4fp3leKx+75IG1rzR+z+gdufqk/Ttf4TKcfsOj3Fb0HE05BGzVr45Pjx/BmUe9vfIVM7PNlqTmiNig+2qvTC6S6oC/Au8HFgAPAydExBPtbdPV5AJZgpn+7FLG77pDxdpZ2vP+H/+Bp5d0fpTlahDQR9mwNLX2r6g3171S55bfDyXm6/pkx4mAvn1Ev7o+rCG7LbJy9RpWrY619VAq369vHSMGbcXrb65m0fI36NtH1NX1YdjArGfkiyv+weDtBjB2RD177bw9ra+/WfH/X83zWvn1Iwt46ZWVNGzbn2PHDtuo/bf3PVAc7+r3RWG7+q36dfn30dY+gI36/trckssBwDkRcURaPgsgIv6rvW02Jrl0t00pwZh1hz6Cfn37cO3nx1ckwTTPa+WESx9c79m0fn37cP3JXdt//pm3fD2L4//xwb347h1zNihX7v5XvrWGoGu/j7b20bePQGLV6s7VJ29zSy7HARMi4vNp+dPA/hFxalG5ycDktLgHMLeLh9wReKmL23ZJ3XYNQ+u23G4wkjoubdYLRMTq117+2+pXl/59Y3dVt80Ob6vbun4o+f89Aatfa13Ywf7b/L++3v5y9SyOr3nrH6/02WLL7YrLda2+nft9tHfOAJ2tT5EREdFQHNysn9CPiEuBSzd2P5Ka2srctcbnsenpLefi89j0VPtcemtvsYXA8NzysBQzM7Nu0FuTy8PAaEmjJPUDjgem9nCdzMw2G73ytlhErJJ0KnA3WVfkKyNiThUPudG31jYRPo9NT285F5/Hpqeq59IrG/TNzKxn9dbbYmZm1oOcXMzMrOKcXDaCpAmS5kpqkXRmT9enFEnDJd0n6QlJcySdluKDJE2T9HT6WZ/iknRBOrdZksb27BlsSFKdpEcl3ZGWR0makep8Y+rMgaT+abklrR/Zk/XOkzRQ0s2SnpL0pKQDavEzkfRv6d/V45KulzSgVj4PSVdKWizp8Vys05+BpEmp/NOSJm0i53Fe+rc1S9Ktkgbm1p2VzmOupCNy8cp8r0WEpy5MZB0FngF2BfoBjwFjerpeJeo7BBib5rclGx5nDPAj4MwUPxM4N80fBdxF9njVeGBGT59DG+f0FeA64I60fBNwfJq/GPhSmv8ycHGaPx64safrnjuHKcDn03w/YGCtfSbAUOA5YMvc5/CZWvk8gPcAY4HHc7FOfQbAIODZ9LM+zddvAudxONA3zZ+bO48x6TurPzAqfZfVVfJ7rcf/YdbqBBwA3J1bPgs4q6fr1Yn630429tpcYEiKDQHmpvlLyMZjK5RfW25TmMieXboHOBS4I/1nfyn3H2nt50PWa/CANN83ldMmcA7bpy9lFcVr6jNJyWV++mLtmz6PI2rp8wBGFn0pd+ozAE4ALsnF1yvXU+dRtO7DwLVpfr3vq8JnUsnvNd8W67rCf6iCBSm2yUu3IfYFZgCDI2JRWvV3YHCa39TP76fAN4A0HjU7AC9HxKq0nK/v2nNJ65en8j1tFLAE+EW6vXe5pK2psc8kIhYC/w28ACwi+/02U3ufR15nP4NN8rMp8jmyqy7ohvNwctnMSNoG+DVwekSsyK+L7E+VTb5vuqQPAosjormn67KR+pLdxrgoIvYFXiO7BbNWLXwmqT1iIlmy3BnYGpjQo5WqoFr4DDoi6VvAKuDa7jqmk0vX1dwQM5K2IEss10bELSn8oqQhaf0QYHGKb8rndyBwtKTngRvIbo39DBgoqfBgcL6+a88lrd8eaPsVot1rAbAgIgpve7uZLNnU2mfyPuC5iFgSEW8Bt5B9RrX2eeR19jPYVD8bJH0G+CDwyZQooRvOw8ml62pqiBlJAq4AnoyIn+RWTQUKPVsmkbXFFOInpt4x44HludsEPSoizoqIYRExkuz3fm9EfBK4DzguFSs+l8I5HpfK9/hfohHxd2C+pD1S6DDgCWrvM3kBGC9pq/TvrHAeNfV5FOnsZ3A3cLik+nQld3iK9ShlL038BnB0RLyeWzUVOD713BsFjAYeopLfaz3ZiFbrE1nPkb+S9a74Vk/Xp4O6HkR2aT8LmJmmo8judd8DPA38HzAolRfw83Rus4HGnj6Hds7rYNb1Fts1/QdpAX4F9E/xAWm5Ja3ftafrnav/PkBT+lxuI+tpVHOfCfAd4CngceAasl5INfF5ANeTtRW9RXY1eVJXPgOyNo2WNH12EzmPFrI2lML/+Ytz5b+VzmMucGQuXpHvNQ//YmZmFefbYmZmVnFOLmZmVnFOLmZmVnFOLmZmVnFOLmZmVnFOLtarSBqZHxW2aN3lksZU4BhDlEZibmPdHyQ1dnG/x+TrtzH7Stufl0YqPq8ofo6kr3V1v12sS4Ok33XnMa1n9crXHJu1JSI+X6FdfQW4rEL7yjuGbNDHJyq0v8lkz2esrtD+ukRS34hYImmRpAMj4i89WR/rHr5ysd6or6Rrlb0f5WZJW8H6VwKSXpX0A0mPSZouaXCKf1TZO0kek/Sndvb/EeB3qfyWkm5Ix7oV2LJQSNLhkh6U9IikX6Vx3ZD0vKQfSZot6SFJu0v6f8DRwHmSZkraLe3mo6nMXyW9u7gi6Unx81KdZ0v6eIpPBbYBmguxtkg6WdJdkrYsio9U9h6Qq9Kxr5X0Pkl/Ufa+kv1Sua2VvUfkIWWDb05M8c9ImirpXrKHESF7SPST7dXFehcnF+uN9gAujIi3AyvI3h9SbGtgekTsDfwJODnF/wM4IsWPLt4oDZXRGhErU+hLwOvpWGcD41K5HYFvA++LiLFkT+F/Jber5RHxTuB/gZ9GxANkw2x8PSL2iYhnUrm+EbEfcHraf7FjyZ7y35tsjK/zJA2JiKOBN9K+bmzrlyTpVLIxp46JiDfaKLI78GNgzzR9gmykh68B30xlvkU2fMt+wCHp+FundWOB4yLivWm5CdggQVrv5ORivdH83K2XX5J9IRZ7k+wWFGTDw49M838BrpJ0MtmLk4oNIRsmv+A96RhExCyyYVwge5HUGOAvkmaSjU81Irfd9bmfB5Q4l8IAo/k65h0EXB8RqyPiReCPwLtK7K/gROBIsi//le2UeS4iZkfEGmAOcE9kQ3rMztXlcODMdI5/IBvaZZe0blpELMvtbzHZqMm2GXCbi/VGxWMatTXG0Vuxbuyj1aT/CxHxRUn7Ax8gu6U0LiLyI/a+QfYF2hGRfbmeUEYdS43BVPjiX1vHCplNdsUzDHhO0nDgN2ndxWS3/fJJZ01ueU2uLgI+EhFz8ztPv8PXio45gOz3Z5sBX7lYb7SLpMLVwCeA+8vdUNJuETEjIv6D7ApleFGRv7L+FcSf0jGQ9A7gn1N8OnCgpN3Tuq0l/VNuu4/nfj6Y5l8hewV1Z/wZ+LikOkkNZFdSD5Wx3aPAF4CpknaOiPnpFto+EXFxJ45/N/AvaTRkJO1bouw/kQ1saZsBJxfrjeYCp0h6kmyU4Ys6se15qWH8ceABsneIrxURrwHPFJJG2vc26VjfJbt9RUQsIXuP/PWSZpElkD1zu6pP8dOAf0uxG4Cvp4bx3SjPrWS34h4D7gW+EdlQ/h2KiPvJ2k9+m9qIuuJ7wBbALElz0nJ7DgF+28XjWI3xqMhmnSTpw8C4iPh2F7d/nmyo9pcqWrFNXOp9NzEiWnu6LlZ9bnMx66SIuFXSpvbO901aumX3EyeWzYevXMzMrOLc5mJmZhXn5GJmZhXn5GJmZhXn5GJmZhXn5GJmZhX3/wH3N1B0fUqrbwAAAABJRU5ErkJggg==\n",
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Xc2DuASY7M1V"
      },
      "source": [
        "Как будто коррекция просто поправила первые три точки (????????????????????????????????????????????????)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8sX5xz2k85oV"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "lIAzkOEc87Ib"
      },
      "source": [
        "## **4. Impact of reads with large insert size**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "2HGGzE9S9ehd"
      },
      "source": [
        "Теперь запустим SPAdes используя все три библиотеки ридов (SRR292678 as a paired ends,  SRR292862 and SRR292770 as a mate pairs).\n",
        "У них разные insert sizes, и мы полагаем что это нам поможет, since the library with a small insert size can resolve short repeats, whereas the library with a larger insert size can resolve longer repeats. "
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9pzmR7tX9wcz"
      },
      "source": [
        "!mkdir 4_spades3libs"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "XWtKllM-7MgV",
        "outputId": "477b908c-3cae-4c5e-ef4b-aaee768e0f35"
      },
      "source": [
        "# -m/--memory\t<int>\t  \tRAM limit for SPAdes in Gb (terminates if exceeded)\n",
        "#\t\t\t\t                [default: 250]\n",
        "# --pe<#>-1\t<filename>\tfile with forward reads for paired-end library number <#> (<#> = 1,2,..,9)\n",
        "# --pe<#>-2\t<filename>\tfile with reverse reads for paired-end library number <#> (<#> = 1,2,..,9)\n",
        "# --mp<#>-1\t<filename>\tfile with forward reads for mate-pair library number <#> (<#> = 1,2,..,9)\n",
        "# --mp<#>-2\t<filename>\tfile with reverse reads for mate-pair library number <#> (<#> = 1,2,..,9)\n",
        "# -o\t<output_dir>\t    directory to store all the resulting files (required)\n",
        "\n",
        "!spades \\\n",
        "--memory 11 \\\n",
        "--continue \\\\\n",
        "--pe1-1 raw_data/paired_end_470bp_fwd.fastq --pe1-2 raw_data/paired_end_470bp_rev.fastq \\\n",
        "--mp2-1 raw_data/mate_pair_2kb_fwd.fastq --mp2-2 raw_data/mate_pair_2kb_rev.fastq \\\n",
        "--mp3-1 raw_data/mate_pair_6kb_fwd.fastq --mp3-2 raw_data/mate_pair_6kb_rev.fastq \\\n",
        "-o 4_spades3libs/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "======= SPAdes pipeline continued. Log can be found here: /content/4_spades3libs/spades.log\n",
            "\n",
            "Restored from Command line: /usr/lib/spades/bin/spades.py\t--memory\t11\t--pe1-1\t/content/raw_data/paired_end_470bp_fwd.fastq\t--pe1-2\t/content/raw_data/paired_end_470bp_rev.fastq\t--mp2-1\t/content/raw_data/mate_pair_2kb_fwd.fastq\t--mp2-2\t/content/raw_data/mate_pair_2kb_rev.fastq\t--mp3-1\t/content/raw_data/mate_pair_6kb_fwd.fastq\t--mp3-2\t/content/raw_data/mate_pair_6kb_rev.fastq\t-o\t/content/4_spades3libs\n",
            "\n",
            "Command line: /usr/lib/spades/bin/spades.py\t--memory\t11\t--pe1-1\t/content/raw_data/paired_end_470bp_fwd.fastq\t--pe1-2\t/content/raw_data/paired_end_470bp_rev.fastq\t--mp2-1\t/content/raw_data/mate_pair_2kb_fwd.fastq\t--mp2-2\t/content/raw_data/mate_pair_2kb_rev.fastq\t--mp3-1\t/content/raw_data/mate_pair_6kb_fwd.fastq\t--mp3-2\t/content/raw_data/mate_pair_6kb_rev.fastq\t-o\t/content/4_spades3libs\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 3.11.1\n",
            "  Python version: 2.7.17\n",
            "  OS: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic\n",
            "\n",
            "Output dir: /content/4_spades3libs\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
            "  Reads:\n",
            "    Library number: 1, library type: mate-pairs\n",
            "      orientation: rf\n",
            "      left reads: ['/content/raw_data/mate_pair_2kb_fwd.fastq']\n",
            "      right reads: ['/content/raw_data/mate_pair_2kb_rev.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "    Library number: 2, library type: mate-pairs\n",
            "      orientation: rf\n",
            "      left reads: ['/content/raw_data/mate_pair_6kb_fwd.fastq']\n",
            "      right reads: ['/content/raw_data/mate_pair_6kb_rev.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "    Library number: 3, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/content/raw_data/paired_end_470bp_fwd.fastq']\n",
            "      right reads: ['/content/raw_data/paired_end_470bp_rev.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed (with gzip)\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned OFF\n",
            "  MismatchCorrector will be SKIPPED\n",
            "  Coverage cutoff is turned OFF\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/4_spades3libs/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 11\n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running read error correction tool: /usr/lib/spades/bin/hammer /content/4_spades3libs/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     0M / 14M   INFO    General                 (main.cpp                  :  83)   Starting BayesHammer, built from N/A, git revision N/A\n",
            "  0:00:00.000     0M / 14M   INFO    General                 (main.cpp                  :  84)   Loading config from /content/4_spades3libs/corrected/configs/config.info\n",
            "  0:00:00.002     0M / 14M   INFO    General                 (memory_limit.hpp          :  52)   Memory limit set to 11 Gb\n",
            "  0:00:00.002     0M / 14M   INFO    General                 (main.cpp                  :  93)   Trying to determine PHRED offset\n",
            "  0:00:00.002     0M / 14M   INFO    General                 (main.cpp                  :  99)   Determined value is 33\n",
            "  0:00:00.004     0M / 14M   INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.004     0M / 14M   INFO    General                 (main.cpp                  : 120)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.005     0M / 14M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
            "  0:00:00.005     0M / 14M   INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 32 buckets using 2 threads. This might take a while.\n",
            "  0:00:00.005     0M / 14M   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.005     0M / 14M   INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 1.83333 Gb\n",
            "  0:00:00.006     0M / 14M   INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 2097152\n",
            "  0:00:00.008     0M / 14M   INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/mate_pair_2kb_fwd.fastq\n",
            "  0:00:15.562     0M / 566M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 1151298 reads\n",
            "  0:00:30.818     0M / 567M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 2302841 reads\n",
            "  0:00:46.347     0M / 567M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 3451922 reads\n",
            "  0:01:01.853     0M / 567M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 4603083 reads\n",
            "  0:01:08.524     0M / 567M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 5102041 reads\n",
            "  0:01:08.524     0M / 567M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/mate_pair_2kb_rev.fastq\n",
            "  0:01:23.958     0M / 583M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 6254294 reads\n",
            "  0:01:39.527     0M / 583M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 7404674 reads\n",
            "  0:01:55.157     0M / 583M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 8555923 reads\n",
            "  0:02:10.541     0M / 583M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 9708729 reads\n",
            "  0:02:17.296     0M / 583M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/mate_pair_6kb_fwd.fastq\n",
            "  0:03:26.888     0M / 584M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/mate_pair_6kb_rev.fastq\n",
            "  0:03:58.755     0M / 584M  INFO   K-mer Splitting          (kmer_data.cpp             : 106)   Processed 17608819 reads\n",
            "  0:04:36.943     0M / 584M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/paired_end_470bp_fwd.fastq\n",
            "  0:07:24.331     0M / 585M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /content/raw_data/paired_end_470bp_rev.fastq\n",
            "  0:10:12.398     0M / 585M  INFO   K-mer Splitting          (kmer_data.cpp             : 111)   Total 31406856 reads processed\n",
            "  0:10:12.635     0M / 585M  INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
            "  0:11:45.096     0M / 851M  INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 169592000 kmers in total.\n",
            "  0:11:45.096     0M / 851M  INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
            "  0:11:50.111     0M / 851M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
            "  0:12:40.633     0M / 851M  INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
            "  0:12:45.197     0M / 851M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 78648912 bytes occupied (3.71003 bits per kmer).\n",
            "  0:12:45.197     0M / 851M  INFO   K-mer Counting           (kmer_data.cpp             : 357)   Arranging kmers in hash map order\n",
            "tcmalloc: large alloc 1356742656 bytes == 0x564693786000 @  0x7fd9fa508887 0x564647f11b6e 0x564647edac0b 0x7fd9f8d1fbf7 0x564647ee0bfa\n",
            "tcmalloc: large alloc 1356742656 bytes == 0x5646e456a000 @  0x7fd9fa508887 0x564647edacc8 0x7fd9f8d1fbf7 0x564647ee0bfa\n",
            "  0:13:14.369     0M / 3G    INFO    General                 (main.cpp                  : 155)   Clustering Hamming graph.\n",
            "  0:41:02.030     0M / 3G    INFO    General                 (main.cpp                  : 162)   Extracting clusters\n",
            "  0:59:50.161     0M / 8G    INFO    General                 (main.cpp                  : 174)   Clustering done. Total clusters: 112762733\n",
            "  0:59:50.218     0M / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 379)   Collecting K-mer information, this takes a while.\n",
            "tcmalloc: large alloc 4070211584 bytes == (nil) @  0x7fd9fa508887 0x564647ee5b37 0x564647f1474c 0x564647edaf8b 0x7fd9f8d1fbf7 0x564647ee0bfa\n",
            "Not enough memory to run BayesHammer. std::bad_alloc\n",
            "\n",
            "\n",
            "== Error ==  system call for: \"['/usr/lib/spades/bin/hammer', '/content/4_spades3libs/corrected/configs/config.info']\" finished abnormally, err code: 4\n",
            "\n",
            "In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru\n",
            "or report an issue on our GitHub repository github.com/ablab/spades\n",
            "Please provide us with params.txt and spades.log files from the output directory.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "wbY09FfuzW1c"
      },
      "source": [
        "<br><hr>\n",
        "\n",
        "Удивительно, но 12Gb RAM на колабе не потянули сборку трёх библиотек. Поэтому воспользуемся precomputed results."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "MCCCR0sM0IP7"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "MPN8Rp2Y7Mnm"
      },
      "source": [
        "# Вынуть с гугл диска и разархивировать\n",
        "!cp 'drive/MyDrive/Desktop/ИБ2021/ИБ/Projects/Project3/three_libs_spades_out.zip' 4_spades3libs.zip\n",
        "!unzip -j -qq 4_spades3libs.zip -d 4_spades3libs"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "PNviDEMAzmwn"
      },
      "source": [
        "<hr><br>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QbBPwGZp0wyr"
      },
      "source": [
        "Результаты [QUAST](http://cab.cc.spbu.ru/quast/):\n",
        "\n",
        "Для файлов contigs.fasta и scaffolds.fasta (НАДО ЛИ ЗАПУСКАТЬ ДЛЯ ВТОРОГО ФАЙЛА?)\n",
        "\n",
        "\\# contigs (>= 500bp):\t105 || 90 <br>\n",
        "N50:\t335515 ||\t2815616 <br>\n",
        "L50:\t6 || 1\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-5mPl5Yb1-wu"
      },
      "source": [
        "Для отчёта: Несмотря на уменьшение общего числа контигов $\\ge 500$bp (с $206$ до $105$), их длины заметно выросли, что видно по большему значению $N50$ метрики ($105346 \\rightarrow 335515$), which means higher quality of three-library assemble."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "N9mlCK0U3ffK"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "tOYCgr7Y3hbC"
      },
      "source": [
        "## **5. Genome Annotation**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "njoJBgQB3iqs"
      },
      "source": [
        "Мы секвенировали и собрали геном *E.coli* X. Теперь неоходимо его аннотировать при помощи **Prokka** на основе следующей идеи: сходные гены в разных организмах выполняют сходные функции.\n",
        "\n",
        "**Prokka** находит в контигах предполагаемые гены, а затем при помощи BLAST ищет для каждого гена в базе данных RefSeq белок, который ген мог бы кодировать.\n",
        "\n",
        "(**Prokka** identifies the coordinates of putative genes within contigs and then uses BLAST for similarity-based annotation using all proteins from sequenced bacterial genomes in the RefSeq database.)"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "rfZnlhuW3dfN"
      },
      "source": [
        "!mkdir 5_prokka_annotation"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tvAehppYDURV",
        "outputId": "0a066d4b-22db-422d-9681-15cd1ffcef97"
      },
      "source": [
        "# --outdir [X]       Output folder [auto] (default '')\n",
        "\n",
        "!$HOME/prokka/bin/prokka --compliant --prefix scaffolds --force /content/4_spades3libs/scaffolds.fasta --outdir /content/5_prokka_annotation"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[19:46:17] This is prokka 1.14.6\n",
            "[19:46:17] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
            "[19:46:17] Homepage is https://github.com/tseemann/prokka\n",
            "[19:46:17] Local time is Mon Nov 15 19:46:17 2021\n",
            "[19:46:17] You are not telling me who you are!\n",
            "[19:46:17] Operating system is linux\n",
            "[19:46:17] You have BioPerl 1.007002\n",
            "[19:46:17] System has 2 cores.\n",
            "[19:46:17] Option --cpu asked for 8 cores, but system only has 2\n",
            "[19:46:17] Will use maximum of 2 cores.\n",
            "[19:46:17] Annotating as >>> Bacteria <<<\n",
            "[19:46:17] Generating locus_tag from '/content/4_spades3libs/scaffolds.fasta' contents.\n",
            "[19:46:17] Setting --locustag IAEMKGFD from MD5 2ae640fd6def9ac6a7454ed12b9bdadd\n",
            "[19:46:17] Enabling options to ensure Genbank/ENA/DDJB submission compliance.\n",
            "[19:46:17] Re-using existing --outdir /content/5_prokka_annotation\n",
            "[19:46:17] Using filename prefix: PROKKA_11152021.XXX\n",
            "[19:46:17] Setting HMMER_NCPU=1\n",
            "[19:46:17] Writing log to: /content/5_prokka_annotation/PROKKA_11152021.log\n",
            "[19:46:17] Command: /root/prokka/bin/prokka --compliant --force /content/4_spades3libs/scaffolds.fasta --outdir /content/5_prokka_annotation\n",
            "[19:46:17] Appending to PATH: /root/prokka/bin/../binaries/linux\n",
            "[19:46:17] Appending to PATH: /root/prokka/bin/../binaries/linux/../common\n",
            "[19:46:17] Appending to PATH: /root/prokka/bin\n",
            "[19:46:17] Looking for 'aragorn' - found /root/prokka/bin/../binaries/linux/aragorn\n",
            "[19:46:18] Determined aragorn version is 001002 from 'ARAGORN v1.2.38 Dean Laslett'\n",
            "[19:46:18] Looking for 'blastp' - found /usr/bin/blastp\n",
            "[19:46:18] Determined blastp version is 002006 from 'blastp: 2.6.0+'\n",
            "[19:46:18] Looking for 'cmpress' - found /usr/bin/cmpress\n",
            "[19:46:18] Determined cmpress version is 001001 from '# INFERNAL 1.1.2 (July 2016)'\n",
            "[19:46:18] Looking for 'cmscan' - found /usr/bin/cmscan\n",
            "[19:46:18] Determined cmscan version is 001001 from '# INFERNAL 1.1.2 (July 2016)'\n",
            "[19:46:18] Looking for 'egrep' - found /bin/egrep\n",
            "[19:46:18] Looking for 'find' - found /usr/bin/find\n",
            "[19:46:18] Looking for 'grep' - found /bin/grep\n",
            "[19:46:18] Looking for 'hmmpress' - found /usr/bin/hmmpress\n",
            "[19:46:18] Determined hmmpress version is 003001 from '# HMMER 3.1b2 (February 2015); http://hmmer.org/'\n",
            "[19:46:18] Looking for 'hmmscan' - found /usr/bin/hmmscan\n",
            "[19:46:18] Determined hmmscan version is 003001 from '# HMMER 3.1b2 (February 2015); http://hmmer.org/'\n",
            "[19:46:18] Looking for 'java' - found /usr/bin/java\n",
            "[19:46:18] Looking for 'makeblastdb' - found /usr/bin/makeblastdb\n",
            "[19:46:18] Determined makeblastdb version is 002006 from 'makeblastdb: 2.6.0+'\n",
            "[19:46:18] Looking for 'minced' - found /root/prokka/bin/../binaries/linux/../common/minced\n",
            "[19:46:18] Determined minced version is 002000 from 'minced 0.2.0'\n",
            "[19:46:18] Looking for 'parallel' - found /usr/bin/parallel\n",
            "[19:46:18] Determined parallel version is 20161222 from 'GNU parallel 20161222'\n",
            "[19:46:18] Looking for 'prodigal' - found /root/prokka/bin/../binaries/linux/prodigal\n",
            "[19:46:18] Determined prodigal version is 002006 from 'Prodigal V2.6.3: February, 2016'\n",
            "[19:46:18] Looking for 'prokka-genbank_to_fasta_db' - found /root/prokka/bin/prokka-genbank_to_fasta_db\n",
            "[19:46:18] Looking for 'sed' - found /bin/sed\n",
            "[19:46:18] Looking for 'tbl2asn' - found /root/prokka/bin/../binaries/linux/tbl2asn\n",
            "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
            "[19:46:18] Determined tbl2asn version is 025008 from 'tbl2asn 25.8   arguments:'\n",
            "[19:46:18] Using genetic code table 11.\n",
            "[19:46:18] Loading and checking input file: /content/4_spades3libs/scaffolds.fasta\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_206_length_195_cov_53.0286_ID_565434\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_207_length_190_cov_182.089_ID_565474\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_208_length_189_cov_134.187_ID_565652\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_209_length_185_cov_73.9154_ID_565796\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_210_length_185_cov_100.585_ID_565476\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_211_length_184_cov_144.496_ID_565576\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_212_length_184_cov_285.101_ID_565694\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_213_length_181_cov_156.389_ID_565552\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_214_length_180_cov_76.872_ID_565192\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_215_length_180_cov_68.032_ID_565194\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_216_length_179_cov_37.2419_ID_565816\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_217_length_174_cov_39.3361_ID_565530\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_218_length_170_cov_69.2609_ID_565702\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_219_length_165_cov_76.2636_ID_565196\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_220_length_152_cov_87.9278_ID_565538\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_221_length_151_cov_104.438_ID_565512\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_222_length_151_cov_377.802_ID_565418\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_223_length_151_cov_285.802_ID_565544\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_224_length_147_cov_33.7717_ID_565790\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_225_length_147_cov_80.8587_ID_565198\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_226_length_146_cov_63.3956_ID_565818\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_227_length_144_cov_50.9326_ID_565200\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_228_length_143_cov_60.25_ID_565510\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_229_length_143_cov_54.7841_ID_565202\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_230_length_142_cov_222.414_ID_565726\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_231_length_141_cov_56.7791_ID_565662\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_232_length_138_cov_51.4578_ID_565786\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_233_length_134_cov_275.228_ID_565540\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_234_length_132_cov_212.649_ID_565422\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_235_length_130_cov_146.667_ID_565762\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_236_length_129_cov_359.216_ID_565204\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_237_length_128_cov_271.781_ID_565514\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_238_length_125_cov_217.214_ID_565470\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_239_length_123_cov_108.603_ID_565700\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_240_length_120_cov_58.1385_ID_565206\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_241_length_119_cov_123.797_ID_565208\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_242_length_118_cov_61.0794_ID_565438\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_243_length_117_cov_53.4839_ID_565648\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_244_length_116_cov_94.3934_ID_565210\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_245_length_115_cov_61.35_ID_565212\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_246_length_115_cov_128.7_ID_565704\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_247_length_114_cov_105.288_ID_565810\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_248_length_113_cov_93.4655_ID_565668\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_249_length_112_cov_87.6667_ID_565772\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_250_length_112_cov_107.596_ID_565432\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_251_length_111_cov_101.018_ID_565468\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_252_length_111_cov_67.1429_ID_565214\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_253_length_111_cov_167.143_ID_565696\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_254_length_110_cov_115.055_ID_565506\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_255_length_110_cov_76.3455_ID_565436\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_256_length_110_cov_388.309_ID_565684\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_257_length_110_cov_77.9091_ID_565216\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_258_length_110_cov_177.764_ID_565218\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_259_length_109_cov_99.7407_ID_565614\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_260_length_109_cov_136.241_ID_565424\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_261_length_109_cov_154.481_ID_565616\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_262_length_109_cov_88.6481_ID_565532\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_263_length_108_cov_46.8302_ID_565220\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_264_length_107_cov_92.75_ID_565496\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_265_length_106_cov_69.5098_ID_565708\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_266_length_106_cov_109.961_ID_565222\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_267_length_103_cov_64.7708_ID_565820\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_268_length_103_cov_161.333_ID_565373\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_269_length_102_cov_107.447_ID_565656\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_270_length_100_cov_140.267_ID_565375\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_271_length_98_cov_32.3023_ID_565377\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_272_length_98_cov_39.7442_ID_565836\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_273_length_97_cov_52.3571_ID_565502\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_274_length_95_cov_75.05_ID_565586\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_275_length_95_cov_66.5_ID_565379\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_276_length_95_cov_114.175_ID_565426\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_277_length_95_cov_116.625_ID_565381\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_278_length_95_cov_100_ID_565383\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_279_length_90_cov_118.657_ID_565678\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_280_length_88_cov_113.333_ID_565480\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_281_length_87_cov_84.3125_ID_565385\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_282_length_87_cov_32.8438_ID_565572\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_283_length_82_cov_335.111_ID_565536\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_284_length_82_cov_63.7778_ID_565584\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_285_length_80_cov_232.64_ID_565636\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_286_length_79_cov_160.417_ID_565450\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_287_length_78_cov_66.3913_ID_565387\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_288_length_75_cov_210.15_ID_565452\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_289_length_74_cov_252.737_ID_565389\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_290_length_72_cov_383.941_ID_565522\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_291_length_72_cov_160.412_ID_565391\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_292_length_71_cov_153.188_ID_565393\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_293_length_71_cov_198.875_ID_565592\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_294_length_70_cov_367.867_ID_565632\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_295_length_69_cov_298.5_ID_565736\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_296_length_68_cov_84.7692_ID_565395\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_297_length_68_cov_181.538_ID_565660\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_298_length_64_cov_238.556_ID_565658\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_299_length_63_cov_125.75_ID_565397\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_300_length_63_cov_261_ID_565516\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_301_length_61_cov_147.667_ID_565674\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_302_length_60_cov_302.4_ID_565446\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_303_length_60_cov_65.4_ID_565399\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_304_length_60_cov_407_ID_565548\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_305_length_59_cov_42_ID_565720\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_306_length_59_cov_176_ID_565706\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_307_length_57_cov_195.5_ID_565442\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_308_length_57_cov_77.5_ID_565401\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_309_length_57_cov_143_ID_565508\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_310_length_57_cov_93_ID_565403\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_311_length_57_cov_71.5_ID_565594\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_312_length_57_cov_135_ID_565405\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_313_length_57_cov_85.5_ID_565407\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_314_length_57_cov_66_ID_565409\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_315_length_57_cov_147.5_ID_565472\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_316_length_56_cov_137_ID_565566\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_317_length_56_cov_145_ID_565411\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_318_length_56_cov_322_ID_565546\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_319_length_56_cov_37_ID_565676\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_320_length_56_cov_227_ID_565486\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_321_length_56_cov_158_ID_565734\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_322_length_56_cov_259_ID_565742\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_323_length_56_cov_269_ID_565832\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_324_length_56_cov_245_ID_565828\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_325_length_56_cov_183_ID_565794\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_326_length_56_cov_162_ID_565550\n",
            "[19:46:18] Skipping short (<200 bp) contig: NODE_327_length_56_cov_148_ID_565838\n",
            "[19:46:18] Wrote 205 contigs totalling 5424432 bp.\n",
            "[19:46:18] Predicting tRNAs and tmRNAs\n",
            "[19:46:18] Running: aragorn -l -gc11  -w \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.fna\n",
            "[19:46:21] 1 tRNA-SeC [72129,72223] 35 (tca)\n",
            "[19:46:21] 2 tRNA-Ala c[122894,122969] 34 (ggc)\n",
            "[19:46:21] 3 tRNA-Ala c[123009,123084] 34 (ggc)\n",
            "[19:46:21] 4 tRNA-Val [125784,125859] 34 (tac)\n",
            "[19:46:21] 5 tRNA-Glu c[326197,326272] 35 (ttc)\n",
            "[19:46:21] 6 tmRNA [352225,352587] 90,125 ANDENYALAA**\n",
            "[19:46:21] 7 tRNA-Met c[369872,369947] 34 (cat)\n",
            "[19:46:21] 8 tRNA-Arg c[401974,402050] 35 (acg)\n",
            "[19:46:21] 9 tRNA-Arg c[402249,402325] 35 (acg)\n",
            "[19:46:21] 10 tRNA-Arg c[402524,402600] 35 (acg)\n",
            "[19:46:21] 11 tRNA-Ser c[402604,402696] 35 (gct)\n",
            "[19:46:21] 12 tRNA-Met c[494712,494788] 35 (cat)\n",
            "[19:46:21] 13 tRNA-Leu c[498569,498655] 35 (gag)\n",
            "[19:46:21] 14 tRNA-Thr c[592314,592389] 34 (ggt)\n",
            "[19:46:21] 15 tRNA-Pro c[892264,892340] 35 (cgg)\n",
            "[19:46:21] 16 tRNA-Thr [981718,981793] 34 (cgt)\n",
            "[19:46:21] 17 tRNA-Ser [1069278,1069358] 38 (gga)\n",
            "[19:46:21] 18 tRNA-Arg [1286866,1286942] 35 (tct)\n",
            "[19:46:21] 19 tRNA-Arg c[1516746,1516822] 35 (tct)\n",
            "[19:46:21] 20 tRNA-Tyr [1557019,1557103] 35 (gta)\n",
            "[19:46:21] 21 tRNA-Ser [1730723,1730810] 35 (gga)\n",
            "[19:46:21] 22 tRNA-Asn c[1764729,1764804] 34 (gtt)\n",
            "[19:46:21] 23 tRNA-Asn c[1767138,1767213] 34 (gtt)\n",
            "[19:46:21] 24 tRNA-Asn [1768962,1769037] 34 (gtt)\n",
            "[19:46:21] 25 tRNA-Asn c[1808853,1808928] 34 (gtt)\n",
            "[19:46:21] 26 tRNA-Ser [1809920,1810009] 35 (cga)\n",
            "[19:46:21] 27 tRNA-Gly [1864131,1864206] 34 (gcc)\n",
            "[19:46:21] 28 tRNA-Cys [1864261,1864334] 33 (gca)\n",
            "[19:46:21] 29 tRNA-Leu [1864347,1864433] 35 (taa)\n",
            "[19:46:21] 30 tRNA-Val c[2198146,2198222] 35 (gac)\n",
            "[19:46:21] 31 tRNA-Val c[2198227,2198303] 35 (gac)\n",
            "[19:46:21] 32 tRNA-Asp c[2492348,2492424] 35 (gtc)\n",
            "[19:46:21] 33 tRNA-Asp c[2500571,2500647] 35 (gtc)\n",
            "[19:46:21] 34 tRNA-Ala c[2504139,2504214] 34 (tgc)\n",
            "[19:46:21] 35 tRNA-Ile c[2504257,2504333] 35 (gat)\n",
            "[19:46:21] 36 tRNA-Leu [2764127,2764213] 35 (cag)\n",
            "[19:46:21] 37 tRNA-Leu [2764242,2764328] 35 (cag)\n",
            "[19:46:21] 38 tRNA-Leu [2764363,2764449] 35 (cag)\n",
            "[19:46:21] 1 tRNA-Met c[23543,23618] 34 (cat)\n",
            "[19:46:21] 2 tRNA-Phe c[225213,225288] 34 (gaa)\n",
            "[19:46:21] 3 tRNA-Gly [340653,340726] 33 (ccc)\n",
            "[19:46:21] 4 tRNA-Met c[399015,399091] 35 (cat)\n",
            "[19:46:21] 5 tRNA-Met c[399124,399200] 35 (cat)\n",
            "[19:46:21] 6 tRNA-Met c[399233,399309] 35 (cat)\n",
            "[19:46:21] 1 tRNA-Arg c[138625,138701] 35 (tct)\n",
            "[19:46:21] 2 tRNA-Arg c[138715,138791] 35 (tcg)\n",
            "[19:46:21] 3 tRNA-Met c[138801,138876] 34 (cat)\n",
            "[19:46:21] 4 tRNA-Ser [197579,197666] 35 (tga)\n",
            "[19:46:21] 5 tRNA-Ser [304837,304924] 35 (gga)\n",
            "[19:46:21] 6 tRNA-Lys c[496259,496334] 34 (ttt)\n",
            "[19:46:21] 7 tRNA-Val c[496339,496414] 34 (tac)\n",
            "[19:46:21] 8 tRNA-Lys c[496564,496639] 34 (ttt)\n",
            "[19:46:21] 9 tRNA-Val c[496644,496719] 34 (tac)\n",
            "[19:46:21] 10 tRNA-Lys c[496755,496830] 34 (ttt)\n",
            "[19:46:21] 1 tRNA-Pro [204937,205013] 35 (ggg)\n",
            "[19:46:21] 2 tRNA-Arg [369799,369873] 34 (cct)\n",
            "[19:46:21] 1 tRNA-Thr [52721,52796] 34 (tgt)\n",
            "[19:46:21] 2 tRNA-Tyr [52805,52889] 35 (gta)\n",
            "[19:46:21] 3 tRNA-Gly [53006,53080] 34 (tcc)\n",
            "[19:46:21] 4 tRNA-Thr [53087,53162] 34 (ggt)\n",
            "[19:46:21] 1 tRNA-Glu [113579,113654] 35 (ttc)\n",
            "[19:46:21] 2 tRNA-Asp [117017,117093] 35 (gtc)\n",
            "[19:46:21] 3 tRNA-Trp [117102,117177] 34 (cca)\n",
            "[19:46:21] 4 tRNA-Arg [152521,152597] 35 (ccg)\n",
            "[19:46:21] 5 tRNA-His [152656,152731] 34 (gtg)\n",
            "[19:46:21] 6 tRNA-Leu [152752,152838] 35 (cag)\n",
            "[19:46:21] 7 tRNA-Pro [152881,152957] 35 (tgg)\n",
            "[19:46:21] 1 tRNA-Gly c[36411,36486] 34 (gcc)\n",
            "[19:46:21] 2 tRNA-Gly c[36523,36598] 34 (gcc)\n",
            "[19:46:21] 3 tRNA-Gly c[36635,36710] 34 (gcc)\n",
            "[19:46:21] 4 tRNA-Phe [66438,66513] 34 (gaa)\n",
            "[19:46:21] 5 tRNA-Leu c[111903,111987] 35 (caa)\n",
            "[19:46:21] 1 tRNA-Gln c[106291,106365] 33 (ctg)\n",
            "[19:46:21] 2 tRNA-Gln c[106403,106477] 33 (ctg)\n",
            "[19:46:21] 3 tRNA-Met c[106526,106602] 35 (cat)\n",
            "[19:46:21] 4 tRNA-Gln c[106618,106692] 33 (ttg)\n",
            "[19:46:21] 5 tRNA-Gln c[106727,106801] 33 (ttg)\n",
            "[19:46:21] 6 tRNA-Leu c[106825,106909] 35 (tag)\n",
            "[19:46:21] 7 tRNA-Met c[106918,106994] 35 (cat)\n",
            "[19:46:21] 1 tRNA-Arg [62,138] 35 (tcg)\n",
            "[19:46:21] Found 80 tRNAs\n",
            "[19:46:21] Predicting Ribosomal RNAs\n",
            "[19:46:21] You need either Barrnap or RNAmmer installed to predict rRNAs!\n",
            "[19:46:21] Skipping ncRNA search, enable with --rfam if desired.\n",
            "[19:46:21] Total of 79 tRNA + rRNA features\n",
            "[19:46:21] Searching for CRISPR repeats\n",
            "[19:46:22] CRISPR1 gnl|Prokka|IAEMKGFD_2 442127 with 14 spacers\n",
            "[19:46:22] Found 1 CRISPRs\n",
            "[19:46:22] Predicting coding sequences\n",
            "[19:46:22] Contigs total 5424432 bp, so using single mode\n",
            "[19:46:22] Running: prodigal -i \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.fna -c -m -g 11 -p single -f sco -q\n",
            "[19:46:31] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_1:592237..592560 on + strand\n",
            "[19:46:31] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_1:1069348..1069593 on + strand\n",
            "[19:46:31] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_1:1557010..1557282 on - strand\n",
            "[19:46:33] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_2:225079..225300 on - strand\n",
            "[19:46:33] Found 5064 CDS\n",
            "[19:46:33] Connecting features back to sequences\n",
            "[19:46:33] Not using genus-specific database. Try --usegenus to enable it.\n",
            "[19:46:33] Annotating CDS, please be patient.\n",
            "[19:46:33] Will use 2 CPUs for similarity searching.\n",
            "[19:46:36] There are still 5064 unannotated CDS left (started with 5064)\n",
            "[19:46:36] Will use blast to search against /root/prokka/db/kingdom/Bacteria/IS with 2 CPUs\n",
            "[19:46:36] Running: cat \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.IS\\.tmp\\.35410\\.faa | parallel --gnu --plain -j 2 --block 395525 --recstart '>' --pipe blastp -query - -db /root/prokka/db/kingdom/Bacteria/IS -evalue 1e-30 -qcov_hsp_perc 90 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.IS\\.tmp\\.35410\\.blast 2> /dev/null\n",
            "[19:48:01] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.IS.tmp.35410.faa\n",
            "[19:48:01] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.IS.tmp.35410.blast\n",
            "[19:48:04] There are still 4989 unannotated CDS left (started with 5064)\n",
            "[19:48:04] Will use blast to search against /root/prokka/db/kingdom/Bacteria/AMR with 2 CPUs\n",
            "[19:48:04] Running: cat \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.AMR\\.tmp\\.35410\\.faa | parallel --gnu --plain -j 2 --block 390517 --recstart '>' --pipe blastp -query - -db /root/prokka/db/kingdom/Bacteria/AMR -evalue 9.99999999999999e-301 -qcov_hsp_perc 90 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.AMR\\.tmp\\.35410\\.blast 2> /dev/null\n",
            "[19:49:14] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.AMR.tmp.35410.faa\n",
            "[19:49:14] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.AMR.tmp.35410.blast\n",
            "[19:49:17] There are still 4989 unannotated CDS left (started with 5064)\n",
            "[19:49:17] Will use blast to search against /root/prokka/db/kingdom/Bacteria/sprot with 2 CPUs\n",
            "[19:49:17] Running: cat \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.sprot\\.tmp\\.35410\\.faa | parallel --gnu --plain -j 2 --block 390517 --recstart '>' --pipe blastp -query - -db /root/prokka/db/kingdom/Bacteria/sprot -evalue 1e-09 -qcov_hsp_perc 80 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.sprot\\.tmp\\.35410\\.blast 2> /dev/null\n",
            "[19:55:13] Modify product: 23S rRNA 5-hydroxycytidine C2501 synthase => 23S rRNA 5-hydroxycytidine synthase\n",
            "[19:55:13] Modify product: Uncharacterized protein Rv3292 => putative protein\n",
            "[19:55:13] Modify product: Uncharacterized protein YnbD => putative protein YnbD\n",
            "[19:55:13] Modify product: Uncharacterized protein YnbC => putative protein YnbC\n",
            "[19:55:13] Modify product: Probable DNA endonuclease SmrA => putative DNA endonuclease SmrA\n",
            "[19:55:13] Modify product: Uncharacterized protein YcjY => putative protein YcjY\n",
            "[19:55:13] Modify product: UPF0283 membrane protein YcjF => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized protein YcjX => putative protein YcjX\n",
            "[19:55:13] Modify product: Uncharacterized protein HI_1162 => putative protein\n",
            "[19:55:13] Modify product: UPF0509 protein YciZ => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized protein YciH => putative protein YciH\n",
            "[19:55:13] Modify product: Probable protease SohB => putative protease SohB\n",
            "[19:55:13] Modify product: Uncharacterized oxidoreductase YciK => putative oxidoreductase YciK\n",
            "[19:55:13] Modify product: Uncharacterized protein YciO => putative protein YciO\n",
            "[19:55:13] Modify product: Prophage tail fiber assembly protein homolog TfaE => Prophage tail fiber assembly protein TfaE\n",
            "[19:55:13] Modify product: UPF0259 membrane protein YciC => hypothetical protein\n",
            "[19:55:13] Modify product: Probable intracellular septation protein A => putative intracellular septation protein A\n",
            "[19:55:13] Modify product: Uncharacterized acyl-CoA thioester hydrolase HI_0827 => putative acyl-CoA thioester hydrolase\n",
            "[19:55:13] Modify product: UPF0263 protein YciU => hypothetical protein\n",
            "[19:55:13] Modify product: UPF0056 inner membrane protein MarC => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized NTE family protein Rv1063c => putative NTE family protein\n",
            "[19:55:13] Modify product: UPF0225 protein YchJ => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized protein YcgM => putative protein YcgM\n",
            "[19:55:13] Modify product: Probable autotransporter ROD_p1121 => putative autotransporter\n",
            "[19:55:13] Modify product: UPF0757 protein YmgG => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized protein YmgD => putative protein YmgD\n",
            "[19:55:13] Modify product: Probable cyclic di-GMP phosphodiesterase PdeG => putative cyclic di-GMP phosphodiesterase PdeG\n",
            "[19:55:13] Modify product: Uncharacterized protein YmgC => putative protein YmgC\n",
            "[19:55:13] Modify product: Probable two-component-system connector protein AriR => putative two-component-system connector protein AriR\n",
            "[19:55:13] Modify product: Probable two-component-system connector protein YcgZ => putative two-component-system connector protein YcgZ\n",
            "[19:55:13] Modify product: Uncharacterized protein YdfO => putative protein YdfO\n",
            "[19:55:13] Modify product: Probable L,D-transpeptidase YcfS => putative L,D-transpeptidase YcfS\n",
            "[19:55:13] Modify product: UPF0227 protein YcfP => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized metal-dependent hydrolase YcfH => putative metal-dependent hydrolase YcfH\n",
            "[19:55:13] Modify product: UPF0502 protein YceH => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized lipoprotein YceB => putative lipoprotein YceB\n",
            "[19:55:13] Modify product: Uncharacterized protein YceO => putative protein YceO\n",
            "[19:55:13] Modify product: Cytochrome b561 homolog 2 => Cytochrome b561 \n",
            "[19:55:13] Modify product: UPF0176 protein YceA => hypothetical protein\n",
            "[19:55:13] Modify product: Probable phosphatase YcdX => putative phosphatase YcdX\n",
            "[19:55:13] Modify product: Probable L,D-transpeptidase ErfK/SrfK => putative L,D-transpeptidase ErfK/SrfK\n",
            "[19:55:13] Modify product: Probable FMN/FAD exporter YeeO => putative FMN/FAD exporter YeeO\n",
            "[19:55:13] Modify product: Probable transcriptional regulatory protein YeeN => putative transcriptional regulatory protein YeeN\n",
            "[19:55:13] Modify product: Putative multidrug export ATP-binding/permease protein SAV1866 => Putative multidrug export ATP-binding/permease protein\n",
            "[19:55:13] Modify product: Uncharacterized protein YedJ => putative protein YedJ\n",
            "[19:55:13] Modify product: Uncharacterized inner membrane transporter YedA => putative inner membrane transporter YedA\n",
            "[19:55:13] Modify product: Probable diguanylate cyclase DgcQ => putative diguanylate cyclase DgcQ\n",
            "[19:55:13] Modify product: Probable diguanylate cyclase DgcQ => putative diguanylate cyclase DgcQ\n",
            "[19:55:13] Modify product: UPF0098 protein YbcL => hypothetical protein\n",
            "[19:55:13] Modify product: UPF0394 inner membrane protein YedE => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized HTH-type transcriptional regulator HI_1476 => putative HTH-type transcriptional regulator\n",
            "[19:55:13] Modify product: Probable transcriptional regulatory protein YebC => putative transcriptional regulatory protein YebC\n",
            "[19:55:13] Modify product: Uncharacterized protein YebV => putative protein YebV\n",
            "[19:55:13] Modify product: Probable manganese efflux pump MntP => putative manganese efflux pump MntP\n",
            "[19:55:13] Modify product: UPF0266 membrane protein YobD => hypothetical protein\n",
            "[19:55:13] Modify product: UPF0053 inner membrane protein YoaE => hypothetical protein\n",
            "[19:55:13] Modify product: Probable cyclic di-GMP phosphodiesterase PdeD => putative cyclic di-GMP phosphodiesterase PdeD\n",
            "[19:55:13] Modify product: Probable ATP-dependent DNA helicase YoaA => putative ATP-dependent DNA helicase YoaA\n",
            "[19:55:13] Modify product: Uncharacterized protein YeaR => putative protein YeaR\n",
            "[19:55:13] Modify product: Probable diguanylate cyclase DgcJ => putative diguanylate cyclase DgcJ\n",
            "[19:55:13] Modify product: Probable diguanylate cyclase CdgI => putative diguanylate cyclase CdgI\n",
            "[19:55:13] Modify product: Stress response UPF0229 protein YhbH => hypothetical protein\n",
            "[19:55:13] Modify product: Uncharacterized zinc-type alcohol dehydrogenase-like protein YdjJ => putative zinc-type alcohol dehydrogenase-like protein YdjJ\n",
            "[19:55:13] Modify product: Uncharacterized protein YdjI => putative protein YdjI\n",
            "[19:55:13] Modify product: Uncharacterized sugar kinase YdjH => putative sugar kinase YdjH\n",
            "[19:55:13] Modify product: Uncharacterized HTH-type transcriptional regulator YdjF => putative HTH-type transcriptional regulator YdjF\n",
            "[19:55:14] Modify product: Probable ketoamine kinase lp_1983 => putative ketoamine kinase\n",
            "[19:55:14] Modify product: Uncharacterized protein YniD => putative protein YniD\n",
            "[19:55:14] Modify product: Prophage tail fiber assembly protein homolog TfaE => Prophage tail fiber assembly protein TfaE\n",
            "[19:55:14] Modify product: Uncharacterized protein YdiE => putative protein YdiE\n",
            "[19:55:14] Modify product: Uncharacterized protein HI_1163 => putative protein\n",
            "[19:55:14] Modify product: Probable ATP-dependent transporter SufC => putative ATP-dependent transporter SufC\n",
            "[19:55:14] Modify product: Probable L,D-transpeptidase YnhG => putative L,D-transpeptidase YnhG\n",
            "[19:55:14] Modify product: Uncharacterized ferredoxin-like protein YdhY => putative ferredoxin-like protein YdhY\n",
            "[19:55:14] Modify product: Uncharacterized oxidoreductase YdhV => putative oxidoreductase YdhV\n",
            "[19:55:14] Modify product: Uncharacterized protein YdhW => putative protein YdhW\n",
            "[19:55:14] Modify product: Uncharacterized ferredoxin-like protein YdhX => putative ferredoxin-like protein YdhX\n",
            "[19:55:14] Modify product: Uncharacterized protein YdhT => putative protein YdhT\n",
            "[19:55:14] Modify product: Probable autotransporter ROD_p1121 => putative autotransporter\n",
            "[19:55:14] Modify product: Uncharacterized HTH-type transcriptional regulator YahB => putative HTH-type transcriptional regulator YahB\n",
            "[19:55:14] Modify product: Uncharacterized protein YnhF => putative protein YnhF\n",
            "[19:55:14] Modify product: Uncharacterized oxidoreductase YdgJ => putative oxidoreductase YdgJ\n",
            "[19:55:14] Modify product: Probable dimethyl sulfoxide reductase chain YnfF => putative dimethyl sulfoxide reductase chain YnfF\n",
            "[19:55:14] Modify product: Uncharacterized protein YnfD => putative protein YnfD\n",
            "[19:55:14] Modify product: UPF0482 protein YnfB => hypothetical protein\n",
            "[19:55:14] Modify product: UPF0060 membrane protein YnfA => hypothetical protein\n",
            "[19:55:14] Modify product: Probable amino-acid metabolite efflux pump => putative amino-acid metabolite efflux pump\n",
            "[19:55:14] Modify product: UPF0056 inner membrane protein MarC => hypothetical protein\n",
            "[19:55:14] Modify product: Probable glutamate/gamma-aminobutyrate antiporter => putative glutamate/gamma-aminobutyrate antiporter\n",
            "[19:55:14] Modify product: Probable D,D-dipeptide-binding periplasmic protein DdpA => putative D,D-dipeptide-binding periplasmic protein DdpA\n",
            "[19:55:14] Modify product: Probable D,D-dipeptide transport system permease protein DdpB => putative D,D-dipeptide transport system permease protein DdpB\n",
            "[19:55:14] Modify product: Probable D,D-dipeptide transport system permease protein DdpC => putative D,D-dipeptide transport system permease protein DdpC\n",
            "[19:55:14] Modify product: Probable D,D-dipeptide transport ATP-binding protein DdpD => putative D,D-dipeptide transport ATP-binding protein DdpD\n",
            "[19:55:14] Modify product: Probable D,D-dipeptide transport ATP-binding protein DdpF => putative D,D-dipeptide transport ATP-binding protein DdpF\n",
            "[19:55:14] Modify product: Uncharacterized HTH-type transcriptional regulator YddM => putative HTH-type transcriptional regulator YddM\n",
            "[19:55:14] Modify product: Probable nitrate reductase molybdenum cofactor assembly chaperone NarW => putative nitrate reductase molybdenum cofactor assembly chaperone NarW\n",
            "[19:55:14] Modify product: Uncharacterized isomerase YddE => putative isomerase YddE\n",
            "[19:55:14] Modify product: Uncharacterized methyltransferase YcgJ => putative methyltransferase YcgJ\n",
            "[19:55:14] Modify product: Uncharacterized protein YaeQ => putative protein YaeQ\n",
            "[19:55:14] Modify product: UPF0253 protein YaeP => hypothetical protein\n",
            "[19:55:14] Modify product: UPF0126 inner membrane protein YadS => hypothetical protein\n",
            "[19:55:14] Modify product: Uncharacterized fimbrial-like protein YadN => putative fimbrial-like protein YadN\n",
            "[19:55:14] Modify product: Probable fimbrial chaperone YadV => putative fimbrial chaperone YadV\n",
            "[19:55:14] Modify product: Uncharacterized fimbrial-like protein YadM => putative fimbrial-like protein YadM\n",
            "[19:55:14] Modify product: Uncharacterized fimbrial-like protein YadL => putative fimbrial-like protein YadL\n",
            "[19:55:14] Modify product: Uncharacterized fimbrial-like protein YadK => putative fimbrial-like protein YadK\n",
            "[19:55:14] Modify product: Uncharacterized fimbrial-like protein YadC => putative fimbrial-like protein YadC\n",
            "[19:55:14] Modify product: Uncharacterized ABC transporter ATP-binding protein YadG => putative ABC transporter ATP-binding protein YadG\n",
            "[19:55:14] Modify product: UPF0231 protein YacL => hypothetical protein\n",
            "[19:55:14] Modify product: Probable peptidoglycan glycosyltransferase FtsW => putative peptidoglycan glycosyltransferase FtsW\n",
            "[19:55:14] Modify product: Uncharacterized protein YibL => putative protein YibL\n",
            "[19:55:14] Modify product: Uncharacterized protein YibN => putative protein YibN\n",
            "[19:55:14] Modify product: Uncharacterized protein HI_0755 => putative protein\n",
            "[19:55:14] Modify product: UPF0758 protein SP_1088 => hypothetical protein\n",
            "[19:55:14] Modify product: UPF0701 protein YicC => hypothetical protein\n",
            "[19:55:14] Modify product: UPF0126 inner membrane protein YicG => hypothetical protein\n",
            "[19:55:14] Modify product: Probable multidrug resistance protein EmrY => putative multidrug resistance protein EmrY\n",
            "[19:55:14] Modify product: Probable multidrug resistance protein EmrK => putative multidrug resistance protein EmrK\n",
            "[19:55:14] Modify product: Probable cyclic di-GMP phosphodiesterase PdeA => putative cyclic di-GMP phosphodiesterase PdeA\n",
            "[19:55:14] Modify product: Uncharacterized protein YfeY => putative protein YfeY\n",
            "[19:55:14] Modify product: Uncharacterized protein Rv2575 => putative protein\n",
            "[19:55:14] Modify product: Probable formate transporter 2 => putative formate transporter 2\n",
            "[19:55:14] Modify product: Uncharacterized protein YfgD => putative protein YfgD\n",
            "[19:55:14] Modify product: Uncharacterized protein YfgJ => putative protein YfgJ\n",
            "[19:55:14] Modify product: UPF0070 protein HI_0370 => hypothetical protein\n",
            "[19:55:14] Modify product: Uncharacterized protein Rv2307c => putative protein\n",
            "[19:55:14] Modify product: Probable 3-phenylpropionic acid transporter => putative 3-phenylpropionic acid transporter\n",
            "[19:55:14] Modify product: Uncharacterized protein YphB => putative protein YphB\n",
            "[19:55:14] Modify product: Uncharacterized tRNA/rRNA methyltransferase YfiF => putative tRNA/rRNA methyltransferase YfiF\n",
            "[19:55:14] Modify product: UPF0053 inner membrane protein YfjD => hypothetical protein\n",
            "[19:55:14] Modify product: UPF0125 protein RatB => hypothetical protein\n",
            "[19:55:14] Modify product: Probable HTH-type transcriptional regulator YgaV => putative HTH-type transcriptional regulator YgaV\n",
            "[19:55:14] Modify product: Uncharacterized protein YgaM => putative protein YgaM\n",
            "[19:55:14] Modify product: Probable transporter SCO4007 => putative transporter\n",
            "[19:55:14] Modify product: Uncharacterized protein YqjD => putative protein YqjD\n",
            "[19:55:14] Modify product: UPF0597 protein YhaM => hypothetical protein\n",
            "[19:55:14] Modify product: Probable serine transporter => putative serine transporter\n",
            "[19:55:14] Modify product: Probable galactarate transporter => putative galactarate transporter\n",
            "[19:55:14] Modify product: Uncharacterized fimbrial-like protein YraH => putative fimbrial-like protein YraH\n",
            "[19:55:14] Modify product: Probable fimbrial chaperone YraI => putative fimbrial chaperone YraI\n",
            "[19:55:14] Modify product: Uncharacterized fimbrial-like protein YraK => putative fimbrial-like protein YraK\n",
            "[19:55:14] Modify product: UPF0213 protein BH0048 => hypothetical protein\n",
            "[19:55:15] Modify product: Uncharacterized protease YhbU => putative protease YhbU\n",
            "[19:55:15] Modify product: Uncharacterized inner membrane transporter YhbE => putative inner membrane transporter YhbE\n",
            "[19:55:15] Modify product: Uncharacterized protein YrbL => putative protein YrbL\n",
            "[19:55:15] Modify product: Uncharacterized protein YhcH => putative protein YhcH\n",
            "[19:55:15] Modify product: Uncharacterized protein YhcO => putative protein YhcO\n",
            "[19:55:15] Modify product: Probable acrylyl-CoA reductase AcuI => putative acrylyl-CoA reductase AcuI\n",
            "[19:55:15] Modify product: Uncharacterized membrane protein YhdT => putative membrane protein YhdT\n",
            "[19:55:15] Modify product: Protein SlyX homolog => Protein SlyX \n",
            "[19:55:15] Modify product: Uncharacterized ABC transporter ATP-binding protein YheS => putative ABC transporter ATP-binding protein YheS\n",
            "[19:55:15] Modify product: UPF0270 protein PA3463 => hypothetical protein\n",
            "[19:55:15] Modify product: Probable protein adenylyltransferase Fic => putative protein adenylyltransferase Fic\n",
            "[19:55:15] Modify product: Uncharacterized protein YhfG => putative protein YhfG\n",
            "[19:55:15] Modify product: Probable fructoselysine/psicoselysine transporter FrlA => putative fructoselysine/psicoselysine transporter FrlA\n",
            "[19:55:15] Modify product: Probable fructoselysine utilization operon transcriptional repressor => putative fructoselysine utilization operon transcriptional repressor\n",
            "[19:55:15] Modify product: Uncharacterized protein YhfS => putative protein YhfS\n",
            "[19:55:15] Modify product: Uncharacterized protein YhfX => putative protein YhfX\n",
            "[19:55:15] Modify product: Putative membrane protein IgaA homolog => Putative membrane protein IgaA \n",
            "[19:55:15] Modify product: Probable [Fe-S]-dependent transcriptional repressor FeoC => putative [Fe-S]-dependent transcriptional repressor FeoC\n",
            "[19:55:15] Modify product: UPF0056 inner membrane protein YhgN => hypothetical protein\n",
            "[19:55:15] Modify product: Uncharacterized oxidoreductase YhhX => putative oxidoreductase YhhX\n",
            "[19:55:15] Modify product: Uncharacterized protein YhhA => putative protein YhhA\n",
            "[19:55:15] Modify product: Uncharacterized MFS-type transporter YhhS => putative MFS-type transporter YhhS\n",
            "[19:55:15] Modify product: Probable copper-binding protein PcoE => putative copper-binding protein PcoE\n",
            "[19:55:15] Modify product: Uncharacterized protein HI_0933 => putative protein\n",
            "[19:55:15] Modify product: Probable cyclic di-GMP phosphodiesterase PdeK => putative cyclic di-GMP phosphodiesterase PdeK\n",
            "[19:55:15] Modify product: Probable fimbrial subunit LpfE => putative fimbrial subunit LpfE\n",
            "[19:55:15] Modify product: Probable minor fimbrial subunit LpfD => putative minor fimbrial subunit LpfD\n",
            "[19:55:15] Modify product: Probable fimbrial chaperone LpfB => putative fimbrial chaperone LpfB\n",
            "[19:55:15] Modify product: Probable major fimbrial subunit LpfA => putative major fimbrial subunit LpfA\n",
            "[19:55:15] Modify product: Uncharacterized MFS-type transporter YhjX => putative MFS-type transporter YhjX\n",
            "[19:55:15] Modify product: Probable lipoprotein YiaD => putative lipoprotein YiaD\n",
            "[19:55:15] Modify product: Uncharacterized protein YhcH => putative protein YhcH\n",
            "[19:55:15] Modify product: Uncharacterized GST-like protein YibF => putative GST-like protein YibF\n",
            "[19:55:15] Modify product: Uncharacterized N-acetyltransferase YafP => putative N-acetyltransferase YafP\n",
            "[19:55:15] Modify product: UPF0255 protein VV1_0328 => hypothetical protein\n",
            "[19:55:15] Modify product: Probable fimbrial chaperone EcpE => putative fimbrial chaperone EcpE\n",
            "[19:55:15] Modify product: Probable outer membrane usher protein EcpC => putative outer membrane usher protein EcpC\n",
            "[19:55:15] Modify product: Probable fimbrial chaperone EcpB => putative fimbrial chaperone EcpB\n",
            "[19:55:15] Modify product: Uncharacterized membrane protein YkgR => putative membrane protein YkgR\n",
            "[19:55:15] Modify product: Probable pyridine nucleotide-disulfide oxidoreductase RclA => putative pyridine nucleotide-disulfide oxidoreductase RclA\n",
            "[19:55:15] Modify product: Uncharacterized HTH-type transcriptional regulator YahB => putative HTH-type transcriptional regulator YahB\n",
            "[19:55:15] Modify product: Uncharacterized transporter YycB => putative transporter YycB\n",
            "[19:55:15] Modify product: Probable diguanylate cyclase DgcC => putative diguanylate cyclase DgcC\n",
            "[19:55:15] Modify product: UPF0178 protein YaiI => hypothetical protein\n",
            "[19:55:15] Modify product: Uncharacterized protein YaiA => putative protein YaiA\n",
            "[19:55:15] Modify product: Uncharacterized lipoprotein YajI => putative lipoprotein YajI\n",
            "[19:55:15] Modify product: Uncharacterized lipoprotein YajI => putative lipoprotein YajI\n",
            "[19:55:15] Modify product: UPF0234 protein YajQ => hypothetical protein\n",
            "[19:55:15] Modify product: Uncharacterized protein HI_1008 => putative protein\n",
            "[19:55:15] Modify product: Probable multidrug resistance ABC transporter ATP-binding/permease protein YheI => putative multidrug resistance ABC transporter ATP-binding/permease protein YheI\n",
            "[19:55:15] Modify product: Uncharacterized lipoprotein YbaY => putative lipoprotein YbaY\n",
            "[19:55:15] Modify product: Uncharacterized protein YbaA => putative protein YbaA\n",
            "[19:55:15] Modify product: Probable cyclic di-GMP phosphodiesterase PdeB => putative cyclic di-GMP phosphodiesterase PdeB\n",
            "[19:55:15] Modify product: Uncharacterized HTH-type transcriptional regulator YbaQ => putative HTH-type transcriptional regulator YbaQ\n",
            "[19:55:15] Modify product: Probable iron export ATP-binding protein FetA => putative iron export ATP-binding protein FetA\n",
            "[19:55:15] Modify product: Probable iron export permease protein FetB => putative iron export permease protein FetB\n",
            "[19:55:15] Modify product: Uncharacterized protein YbbC => putative protein YbbC\n",
            "[19:55:15] Modify product: Uncharacterized protein YbcJ => putative protein YbcJ\n",
            "[19:55:15] Modify product: Bifunctional protein FolD => Bifunctional protein FolD protein\n",
            "[19:55:15] Modify product: Uncharacterized fimbrial-like protein SfmA => putative fimbrial-like protein SfmA\n",
            "[19:55:15] Modify product: Probable fimbrial chaperone SfmC => putative fimbrial chaperone SfmC\n",
            "[19:55:15] Modify product: Uncharacterized fimbrial-like protein SfmH => putative fimbrial-like protein SfmH\n",
            "[19:55:15] Modify product: Uncharacterized fimbrial-like protein SfmF => putative fimbrial-like protein SfmF\n",
            "[19:55:15] Modify product: Uncharacterized protein YncE => putative protein YncE\n",
            "[19:55:15] Modify product: Uncharacterized protein YdcY => putative protein YdcY\n",
            "[19:55:15] Modify product: Probable inner membrane protein Smp => putative inner membrane protein Smp\n",
            "[19:55:15] Modify product: Uncharacterized protein YjjI => putative protein YjjI\n",
            "[19:55:15] Modify product: Uncharacterized metal-dependent hydrolase YjjV => putative metal-dependent hydrolase YjjV\n",
            "[19:55:15] Modify product: UPF0442 protein YjjB => hypothetical protein\n",
            "[19:55:15] Modify product: Probable HTH-type transcriptional regulator LgoR => putative HTH-type transcriptional regulator LgoR\n",
            "[19:55:15] Modify product: Probable L-galactonate transporter => putative L-galactonate transporter\n",
            "[19:55:15] Modify product: Probable type I restriction enzyme BthVORF4518P M protein => putative type I restriction enzymeP M protein\n",
            "[19:55:15] Modify product: Uncharacterized protein YjiK => putative protein YjiK\n",
            "[19:55:15] Modify product: Probable RNA 2'-phosphotransferase => putative RNA 2'-phosphotransferase\n",
            "[19:55:16] Modify product: Probable glycerol-3-phosphate acyltransferase => putative glycerol-3-phosphate acyltransferase\n",
            "[19:55:16] Modify product: Uncharacterized fimbrial-like protein YbgD => putative fimbrial-like protein YbgD\n",
            "[19:55:16] Modify product: Probable quinol monooxygenase YgiN => putative quinol monooxygenase YgiN\n",
            "[19:55:16] Modify product: Uncharacterized oxidoreductase YghA => putative oxidoreductase YghA\n",
            "[19:55:16] Modify product: Probable Ni/Fe-hydrogenase 2 b-type cytochrome subunit => putative Ni/Fe-hydrogenase 2 b-type cytochrome subunit\n",
            "[19:55:16] Modify product: Uncharacterized protein YtfJ => putative protein YtfJ\n",
            "[19:55:16] Modify product: Uncharacterized FAD-linked oxidoreductase Rv2280 => putative FAD-linked oxidoreductase\n",
            "[19:55:16] Modify product: UPF0401 protein YkfF => hypothetical protein\n",
            "[19:55:16] Modify product: Probable Fe(2+)-trafficking protein => putative Fe(2+)-trafficking protein\n",
            "[19:55:16] Modify product: Uncharacterized protein HI_0341 => putative protein\n",
            "[19:55:16] Modify product: UPF0235 protein YggU => hypothetical protein\n",
            "[19:55:16] Modify product: Uncharacterized protein YggR => putative protein YggR\n",
            "[19:55:16] Modify product: UPF0301 protein YqgE => hypothetical protein\n",
            "[19:55:16] Modify product: UPF0149 protein HI_0817 => hypothetical protein\n",
            "[19:55:16] Modify product: Uncharacterized oxidoreductase YgfF => putative oxidoreductase YgfF\n",
            "[19:55:16] Modify product: UPF0267 protein YqfB => hypothetical protein\n",
            "[19:55:16] Modify product: UPF0073 inner membrane protein YqfA => hypothetical protein\n",
            "[19:55:16] Modify product: Uncharacterized protein YqfG => putative protein YqfG\n",
            "[19:55:16] Modify product: Uncharacterized response regulatory protein YgeK => putative response regulatory protein YgeK\n",
            "[19:55:16] Modify product: Uncharacterized lipoprotein YgdR => putative lipoprotein YgdR\n",
            "[19:55:16] Modify product: UPF0053 inner membrane protein YgdQ => hypothetical protein\n",
            "[19:55:16] Modify product: Uncharacterized lipoprotein YgdR => putative lipoprotein YgdR\n",
            "[19:55:16] Modify product: UPF0382 inner membrane protein YgdD => hypothetical protein\n",
            "[19:55:16] Modify product: Uncharacterized HTH-type transcriptional regulator YdjF => putative HTH-type transcriptional regulator YdjF\n",
            "[19:55:16] Modify product: Uncharacterized protein YqcC => putative protein YqcC\n",
            "[19:55:16] Modify product: Uncharacterized protein PA3435 => putative protein\n",
            "[19:55:16] Modify product: Probable glucarate transporter => putative glucarate transporter\n",
            "[19:55:16] Modify product: Uncharacterized protein YqcG => putative protein YqcG\n",
            "[19:55:16] Modify product: Uncharacterized protein YgcP => putative protein YgcP\n",
            "[19:55:16] Modify product: Probable UbiX-like flavin prenyltransferase => putative UbiX-like flavin prenyltransferase\n",
            "[19:55:16] Modify product: Probable diguanylate cyclase DgcT => putative diguanylate cyclase DgcT\n",
            "[19:55:16] Modify product: Probable malonic semialdehyde reductase RutE => putative malonic semialdehyde reductase RutE\n",
            "[19:55:16] Modify product: Uncharacterized protein YccJ => putative protein YccJ\n",
            "[19:55:16] Modify product: Uncharacterized lipoprotein GfcB => putative lipoprotein GfcB\n",
            "[19:55:16] Modify product: Probable Ni/Fe-hydrogenase 1 B-type cytochrome subunit => putative Ni/Fe-hydrogenase 1 B-type cytochrome subunit\n",
            "[19:55:16] Modify product: Uncharacterized protein YccU => putative protein YccU\n",
            "[19:55:16] Modify product: UPF0319 protein HI_1681 => hypothetical protein\n",
            "[19:55:16] Modify product: Uncharacterized protein YcbX => putative protein YcbX\n",
            "[19:55:16] Modify product: Uncharacterized fimbrial chaperone YcbF => putative fimbrial chaperone YcbF\n",
            "[19:55:16] Modify product: Uncharacterized fimbrial-like protein YcbV => putative fimbrial-like protein YcbV\n",
            "[19:55:16] Modify product: Uncharacterized fimbrial-like protein YcbU => putative fimbrial-like protein YcbU\n",
            "[19:55:16] Modify product: Uncharacterized fimbrial-like protein ElfG => putative fimbrial-like protein ElfG\n",
            "[19:55:16] Modify product: Probable outer membrane usher protein ElfC => putative outer membrane usher protein ElfC\n",
            "[19:55:16] Modify product: Probable fimbrial chaperone protein ElfD => putative fimbrial chaperone protein ElfD\n",
            "[19:55:16] Modify product: Probable L,D-transpeptidase YcbB => putative L,D-transpeptidase YcbB\n",
            "[19:55:16] Modify product: UPF0434 protein YcaR => hypothetical protein\n",
            "[19:55:16] Modify product: Probable formate transporter 1 => putative formate transporter 1\n",
            "[19:55:16] Modify product: Uncharacterized MFS-type transporter YcaD => putative MFS-type transporter YcaD\n",
            "[19:55:16] Modify product: Probable hydrolase YcaC => putative hydrolase YcaC\n",
            "[19:55:16] Modify product: UPF0145 protein YbjQ => hypothetical protein\n",
            "[19:55:16] Modify product: Uncharacterized lipoprotein YbjP => putative lipoprotein YbjP\n",
            "[19:55:16] Modify product: Uncharacterized protein YbjN => putative protein YbjN\n",
            "[19:55:16] Modify product: Probable cyclic di-GMP phosphodiesterase PdeI => putative cyclic di-GMP phosphodiesterase PdeI\n",
            "[19:55:16] Modify product: Uncharacterized protein YbiU => putative protein YbiU\n",
            "[19:55:16] Modify product: Uncharacterized ABC transporter ATP-binding protein YbiT => putative ABC transporter ATP-binding protein YbiT\n",
            "[19:55:16] Modify product: Probable L,D-transpeptidase YbiS => putative L,D-transpeptidase YbiS\n",
            "[19:55:16] Modify product: Uncharacterized protein McbA => putative protein McbA\n",
            "[19:55:16] Modify product: PKHD-type hydroxylase Sbal_3634 => PKHD-type hydroxylase\n",
            "[19:55:16] Modify product: Uncharacterized protein YbiI => putative protein YbiI\n",
            "[19:55:16] Modify product: Uncharacterized protein YbiB => putative protein YbiB\n",
            "[19:55:16] Modify product: UPF0194 membrane protein YbhG => hypothetical protein\n",
            "[19:55:16] Modify product: Probable multidrug ABC transporter ATP-binding protein YbhF => putative multidrug ABC transporter ATP-binding protein YbhF\n",
            "[19:55:16] Modify product: Probable multidrug ABC transporter permease YbhS => putative multidrug ABC transporter permease YbhS\n",
            "[19:55:16] Modify product: Probable multidrug ABC transporter permease YbhR => putative multidrug ABC transporter permease YbhR\n",
            "[19:55:16] Modify product: UPF0098 protein YbhB => hypothetical protein\n",
            "[19:55:16] Modify product: Probable diguanylate cyclase CdgI => putative diguanylate cyclase CdgI\n",
            "[19:55:16] Modify product: Prophage tail fiber assembly protein homolog TfaE => Prophage tail fiber assembly protein TfaE\n",
            "[19:55:16] Modify product: Uncharacterized protein YdfO => putative protein YdfO\n",
            "[19:55:16] Modify product: Uncharacterized protein YlcG => putative protein YlcG\n",
            "[19:55:16] Modify product: Uncharacterized HTH-type transcriptional regulator HI_1476 => putative HTH-type transcriptional regulator\n",
            "[19:55:16] Modify product: Uncharacterized fimbrial-like protein YbgD => putative fimbrial-like protein YbgD\n",
            "[19:55:16] Modify product: UPF0394 inner membrane protein YeeE => hypothetical protein\n",
            "[19:55:16] Modify product: UPF0053 inner membrane protein YoaE => hypothetical protein\n",
            "[19:55:16] Modify product: Probable diguanylate cyclase DgcE => putative diguanylate cyclase DgcE\n",
            "[19:55:16] Modify product: UPF0339 protein YegP => hypothetical protein\n",
            "[19:55:17] Modify product: 23S rRNA 5-hydroxycytidine C2501 synthase => 23S rRNA 5-hydroxycytidine synthase\n",
            "[19:55:17] Modify product: Uncharacterized fimbrial-like protein YehA => putative fimbrial-like protein YehA\n",
            "[19:55:17] Modify product: Probable fimbrial chaperone YehC => putative fimbrial chaperone YehC\n",
            "[19:55:17] Modify product: Uncharacterized fimbrial-like protein YehD => putative fimbrial-like protein YehD\n",
            "[19:55:17] Modify product: Uncharacterized protein YehL => putative protein YehL\n",
            "[19:55:17] Modify product: Uncharacterized lipoprotein YehR => putative lipoprotein YehR\n",
            "[19:55:17] Modify product: UPF0387 membrane protein YohO => hypothetical protein\n",
            "[19:55:17] Modify product: Uncharacterized oxidoreductase YohF => putative oxidoreductase YohF\n",
            "[19:55:17] Modify product: UPF0299 membrane protein YohJ => hypothetical protein\n",
            "[19:55:17] Modify product: UPF0324 inner membrane protein YeiH => hypothetical protein\n",
            "[19:55:17] Modify product: UPF0153 protein YeiW => hypothetical protein\n",
            "[19:55:17] Modify product: Probable cyclic di-GMP phosphodiesterase PdeN => putative cyclic di-GMP phosphodiesterase PdeN\n",
            "[19:55:17] Modify product: Uncharacterized ABC transporter ATP-binding protein YejF => putative ABC transporter ATP-binding protein YejF\n",
            "[19:55:17] Modify product: Uncharacterized protein YejG => putative protein YejG\n",
            "[19:55:17] Modify product: UPF0352 protein YejL => hypothetical protein\n",
            "[19:55:17] Modify product: Probable autotransporter ROD_p1121 => putative autotransporter\n",
            "[19:55:17] Modify product: Probable autotransporter YfaL => putative autotransporter YfaL\n",
            "[19:55:17] Modify product: Probable 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD => putative 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD\n",
            "[19:55:17] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE\n",
            "[19:55:17] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF\n",
            "[19:55:17] Modify product: UPF0304 protein YfbU => hypothetical protein\n",
            "[19:55:17] Modify product: UPF0208 membrane protein YfbV => hypothetical protein\n",
            "[19:55:17] Modify product: Uncharacterized Nudix hydrolase YfcD => putative Nudix hydrolase YfcD\n",
            "[19:55:17] Modify product: Epimerase family protein SA0724 => Epimerase family protein\n",
            "[19:55:17] Modify product: Uncharacterized MFS-type transporter YfcJ => putative MFS-type transporter YfcJ\n",
            "[19:55:17] Modify product: Probable membrane transporter protein YfcA => putative membrane transporter protein YfcA\n",
            "[19:55:17] Modify product: Uncharacterized protein YfcO => putative protein YfcO\n",
            "[19:55:17] Modify product: Uncharacterized fimbrial-like protein YfcP => putative fimbrial-like protein YfcP\n",
            "[19:55:17] Modify product: Uncharacterized fimbrial-like protein YfcQ => putative fimbrial-like protein YfcQ\n",
            "[19:55:17] Modify product: Uncharacterized fimbrial-like protein YfcR => putative fimbrial-like protein YfcR\n",
            "[19:55:17] Modify product: Probable fimbrial chaperone YfcS => putative fimbrial chaperone YfcS\n",
            "[19:55:17] Modify product: Uncharacterized fimbrial-like protein YfcV => putative fimbrial-like protein YfcV\n",
            "[19:55:17] Modify product: UPF0381 protein YiiS => hypothetical protein\n",
            "[19:55:17] Modify product: Uncharacterized protein HI_1410 => putative protein\n",
            "[19:55:17] Modify product: Probable cystine transporter YijE => putative cystine transporter YijE\n",
            "[19:55:17] Modify product: Probable L-galactonate transporter => putative L-galactonate transporter\n",
            "[19:55:17] Modify product: Uncharacterized protein HI_0431 => putative protein\n",
            "[19:55:17] Modify product: Uncharacterized lipoprotein GfcB => putative lipoprotein GfcB\n",
            "[19:55:17] Modify product: UPF0337 protein YjbJ => hypothetical protein\n",
            "[19:55:17] Modify product: Uncharacterized protein YjbR => putative protein YjbR\n",
            "[19:55:17] Modify product: Probable cyclic di-GMP phosphodiesterase PdeC => putative cyclic di-GMP phosphodiesterase PdeC\n",
            "[19:55:17] Modify product: Uncharacterized ferredoxin-like protein YdhX => putative ferredoxin-like protein YdhX\n",
            "[19:55:17] Modify product: Probable transcriptional regulator PhnF => putative transcriptional regulator PhnF\n",
            "[19:55:17] Modify product: Uncharacterized protein YjdJ => putative protein YjdJ\n",
            "[19:55:17] Modify product: Probable cadaverine/lysine antiporter => putative cadaverine/lysine antiporter\n",
            "[19:55:17] Modify product: Uncharacterized inner membrane transporter YicL => putative inner membrane transporter YicL\n",
            "[19:55:17] Modify product: Uncharacterized symporter YidK => putative symporter YidK\n",
            "[19:55:17] Modify product: Probable 2-dehydro-3-deoxygalactonokinase DgoK1 => putative 2-dehydro-3-deoxygalactonokinase DgoK1\n",
            "[19:55:17] Modify product: UPF0167 protein CbrC => hypothetical protein\n",
            "[19:55:17] Modify product: Probable glucosamine-6-phosphate deaminase 2 => putative glucosamine-6-phosphate deaminase 2\n",
            "[19:55:17] Modify product: Uncharacterized fimbrial-like protein ElfG => putative fimbrial-like protein ElfG\n",
            "[19:55:17] Modify product: Probable outer membrane usher protein ElfC => putative outer membrane usher protein ElfC\n",
            "[19:55:17] Modify product: Probable transport protein HsrA => putative transport protein HsrA\n",
            "[19:55:17] Modify product: UPF0438 protein YifE => hypothetical protein\n",
            "[19:55:17] Modify product: Probable ECA polymerase => putative ECA polymerase\n",
            "[19:55:17] Modify product: Probable transport protein YifK => putative transport protein YifK\n",
            "[19:55:17] Modify product: Probable protein kinase UbiB => putative protein kinase UbiB\n",
            "[19:55:17] Modify product: Probable L-ascorbate-6-phosphate lactonase UlaG => putative L-ascorbate-6-phosphate lactonase UlaG\n",
            "[19:55:17] Modify product: UPF0719 inner membrane protein YjfL => hypothetical protein\n",
            "[19:55:17] Modify product: UPF0716 protein FxsA => hypothetical protein\n",
            "[19:55:17] Modify product: Uncharacterized HTH-type transcriptional regulator YbbH => putative HTH-type transcriptional regulator YbbH\n",
            "[19:55:17] Modify product: Uncharacterized protein YjgL => putative protein YjgL\n",
            "[19:55:17] Modify product: Uncharacterized protein YjgL => putative protein YjgL\n",
            "[19:55:17] Modify product: UPF0307 protein YjgA => hypothetical protein\n",
            "[19:55:17] Modify product: UPF0053 inner membrane protein YtfL => hypothetical protein\n",
            "[19:55:17] Modify product: Uncharacterized protein YtfJ => putative protein YtfJ\n",
            "[19:55:17] Modify product: Uncharacterized protein YbdM => putative protein YbdM\n",
            "[19:55:17] Modify product: Uncharacterized HTH-type transcriptional regulator YbdO => putative HTH-type transcriptional regulator YbdO\n",
            "[19:55:17] Modify product: Probable zinc-binding alcohol dehydrogenase Rv1895 => putative zinc-binding alcohol dehydrogenase\n",
            "[19:55:17] Modify product: Uncharacterized HTH-type transcriptional regulator YbdO => putative HTH-type transcriptional regulator YbdO\n",
            "[19:55:17] Modify product: UPF0250 protein YbeD => hypothetical protein\n",
            "[19:55:18] Modify product: Uncharacterized lipoprotein ChiQ => putative lipoprotein ChiQ\n",
            "[19:55:18] Modify product: Uncharacterized protein YbfE => putative protein YbfE\n",
            "[19:55:18] Modify product: UPF0381 protein YiiS => hypothetical protein\n",
            "[19:55:18] Modify product: UPF0761 membrane protein YihY => hypothetical protein\n",
            "[19:55:18] Modify product: Uncharacterized protein YphB => putative protein YphB\n",
            "[19:55:18] Modify product: Uncharacterized zinc-type alcohol dehydrogenase-like protein YjmD => putative zinc-type alcohol dehydrogenase-like protein YjmD\n",
            "[19:55:18] Modify product: Uncharacterized protein YshB => putative protein YshB\n",
            "[19:55:18] Modify product: Probable GTP-binding protein EngB => putative GTP-binding protein EngB\n",
            "[19:55:18] Modify product: Probable acyltransferase YihG => putative acyltransferase YihG\n",
            "[19:55:18] Modify product: Uncharacterized protein YggR => putative protein YggR\n",
            "[19:55:18] Modify product: UPF0401 protein YkfF => hypothetical protein\n",
            "[19:55:18] Modify product: Probable RNA polymerase sigma factor FecI => putative RNA polymerase sigma factor FecI\n",
            "[19:55:18] Modify product: Uncharacterized ABC transporter ATP-binding protein YknY => putative ABC transporter ATP-binding protein YknY\n",
            "[19:55:18] Modify product: GTP cyclohydrolase 1 type 2 homolog => GTP cyclohydrolase 1 type 2 \n",
            "[19:55:18] Modify product: UPF0401 protein YkfF => hypothetical protein\n",
            "[19:55:18] Modify product: UPF0265 protein YeeX => hypothetical protein\n",
            "[19:55:18] Cleaned 378 /product names\n",
            "[19:55:18] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.sprot.tmp.35410.faa\n",
            "[19:55:18] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.sprot.tmp.35410.blast\n",
            "[19:55:19] There are still 1418 unannotated CDS left (started with 5064)\n",
            "[19:55:19] Will use hmmer3 to search against /root/prokka/db/hmm/HAMAP.hmm with 2 CPUs\n",
            "[19:55:19] Running: cat \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.HAMAP\\.hmm\\.tmp\\.35410\\.faa | parallel --gnu --plain -j 2 --block 87975 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-09 --cpu 1 /root/prokka/db/hmm/HAMAP.hmm /dev/stdin > \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.HAMAP\\.hmm\\.tmp\\.35410\\.hmmer3 2> /dev/null\n",
            "[19:56:23] Modify product: Uncharacterized Nudix hydrolase NudL => putative Nudix hydrolase NudL\n",
            "[19:56:23] Cleaned 1 /product names\n",
            "[19:56:23] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.HAMAP.hmm.tmp.35410.faa\n",
            "[19:56:23] Deleting unwanted file: /content/5_prokka_annotation/PROKKA_11152021.HAMAP.hmm.tmp.35410.hmmer3\n",
            "[19:56:23] Labelling remaining 1329 proteins as 'hypothetical protein'\n",
            "[19:56:23] Possible /pseudo 'Small toxic polypeptide LdrD' at gnl|Prokka|IAEMKGFD_1 position 883952\n",
            "[19:56:23] Possible /pseudo 'Periplasmic alpha-amylase' at gnl|Prokka|IAEMKGFD_1 position 926371\n",
            "[19:56:23] Possible /pseudo 'ISAs1 family transposase ISEc1' at gnl|Prokka|IAEMKGFD_1 position 957869\n",
            "[19:56:23] Possible /pseudo 'putative lipoprotein YajI' at gnl|Prokka|IAEMKGFD_1 position 1153020\n",
            "[19:56:23] Possible /pseudo 'Small toxic polypeptide LdrA' at gnl|Prokka|IAEMKGFD_1 position 1574182\n",
            "[19:56:23] Possible /pseudo 'Hemolysin E, chromosomal' at gnl|Prokka|IAEMKGFD_1 position 1613271\n",
            "[19:56:23] Possible /pseudo 'Hemolysin E, chromosomal' at gnl|Prokka|IAEMKGFD_1 position 1613714\n",
            "[19:56:23] Possible /pseudo 'Outer membrane protein YedS' at gnl|Prokka|IAEMKGFD_1 position 1821247\n",
            "[19:56:23] Possible /pseudo 'putative diguanylate cyclase DgcQ' at gnl|Prokka|IAEMKGFD_1 position 1828610\n",
            "[19:56:23] Possible /pseudo 'Formate dehydrogenase, nitrate-inducible, major subunit' at gnl|Prokka|IAEMKGFD_1 position 2445595\n",
            "[19:56:23] Possible /pseudo 'Energy-coupling factor transporter ATP-binding protein EcfA1' at gnl|Prokka|IAEMKGFD_2 position 263200\n",
            "[19:56:23] Possible /pseudo 'Methylmalonyl-CoA mutase' at gnl|Prokka|IAEMKGFD_2 position 278090\n",
            "[19:56:23] Possible /pseudo 'tRNA pseudouridine synthase C' at gnl|Prokka|IAEMKGFD_2 position 423125\n",
            "[19:56:23] Possible /pseudo 'Glucarate dehydratase-related protein' at gnl|Prokka|IAEMKGFD_2 position 426142\n",
            "[19:56:23] Possible /pseudo 'General stress protein 16U' at gnl|Prokka|IAEMKGFD_3 position 3731\n",
            "[19:56:23] Possible /pseudo 'IS3 family transposase IS3' at gnl|Prokka|IAEMKGFD_3 position 74279\n",
            "[19:56:23] Possible /pseudo 'Cold shock-like protein CspG' at gnl|Prokka|IAEMKGFD_3 position 178893\n",
            "[19:56:23] Possible /pseudo 'D-tagatose-1,6-bisphosphate aldolase subunit GatZ' at gnl|Prokka|IAEMKGFD_4 position 93731\n",
            "[19:56:23] Possible /pseudo 'Protease ElaD' at gnl|Prokka|IAEMKGFD_4 position 289994\n",
            "[19:56:23] Possible /pseudo 'Formate dehydrogenase H' at gnl|Prokka|IAEMKGFD_5 position 177742\n",
            "[19:56:23] Possible /pseudo '6-phospho-alpha-glucosidase' at gnl|Prokka|IAEMKGFD_6 position 26360\n",
            "[19:56:23] Possible /pseudo 'Cryptic beta-glucoside bgl operon antiterminator' at gnl|Prokka|IAEMKGFD_6 position 71144\n",
            "[19:56:23] Possible /pseudo 'putative protein YjgL' at gnl|Prokka|IAEMKGFD_7 position 131264\n",
            "[19:56:23] Possible /pseudo 'PTS system mannose-specific EIIBCA component' at gnl|Prokka|IAEMKGFD_9 position 34878\n",
            "[19:56:23] Possible /pseudo 'Formate dehydrogenase-O major subunit' at gnl|Prokka|IAEMKGFD_9 position 42043\n",
            "[19:56:23] Possible /pseudo 'Multifunctional conjugation protein TraI' at gnl|Prokka|IAEMKGFD_16 position 5510\n",
            "[19:56:23] Possible /pseudo 'Antigen 43' at gnl|Prokka|IAEMKGFD_27 position 1368\n",
            "[19:56:23] Found 3206 unique /gene codes.\n",
            "[19:56:23] Fixed 3 duplicate /gene - hcaR_1 hcaR_2 hcaR_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - bglG_1 bglG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lgoT_1 lgoT_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fimA_1 fimA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - chuR_1 chuR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - hdfR_1 hdfR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yjiE_1 yjiE_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - acrB_1 acrB_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - cusC_1 cusC_2 cusC_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - fryA_1 fryA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yibA_1 yibA_2\n",
            "[19:56:23] Fixed 5 duplicate /gene - rhaS_1 rhaS_2 rhaS_3 rhaS_4 rhaS_5\n",
            "[19:56:23] Fixed 2 duplicate /gene - gspA_1 gspA_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - ygcS_1 ygcS_2 ygcS_3\n",
            "[19:56:23] Fixed 5 duplicate /gene - dmlR_1 dmlR_2 dmlR_3 dmlR_4 dmlR_5\n",
            "[19:56:23] Fixed 2 duplicate /gene - elfC_1 elfC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - epsM_1 epsM_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - copA_1 copA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - xerD_1 xerD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - chbA_1 chbA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - sapB_1 sapB_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - dmsB_1 dmsB_2 dmsB_3\n",
            "[19:56:23] Fixed 3 duplicate /gene - ndhB_1 ndhB_2 ndhB_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - yhaJ_1 yhaJ_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - idi_1 idi_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ydjZ_1 ydjZ_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rob_1 rob_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - intQ_1 intQ_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - mngR_1 mngR_2 mngR_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - yahB_1 yahB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - papC_1 papC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - cusF_1 cusF_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - xdhA_1 xdhA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - kduD_1 kduD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - glpG_1 glpG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ulaE_1 ulaE_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yhcH_1 yhcH_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ydgA_1 ydgA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - clpV1_1 clpV1_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - mtlA_1 mtlA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - focC_1 focC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - kdgK_1 kdgK_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - gudX_1 gudX_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - nanR_1 nanR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - bglF_1 bglF_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - bglA_1 bglA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - oxyR_1 oxyR_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - glnQ_1 glnQ_2 glnQ_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - ydjF_1 ydjF_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rfaH_1 rfaH_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - psuK_1 psuK_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - deoB_1 deoB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yiaW_1 yiaW_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - scpA_1 scpA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yjgL_1 yjgL_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ftsA_1 ftsA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - proA_1 proA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - flhA_1 flhA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fdoG_1 fdoG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - klcA_1 klcA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yggR_1 yggR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rhtC_1 rhtC_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - rbsC_1 rbsC_2 rbsC_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - alx_1 alx_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - nanE_1 nanE_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - epsF_1 epsF_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - potA_1 potA_2 potA_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - cusR_1 cusR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - pitA_1 pitA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - cspG_1 cspG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - hycG_1 hycG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - gstB_1 gstB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - cusB_1 cusB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rtcB_1 rtcB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - pduV_1 pduV_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - vapC_1 vapC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - tnpR_1 tnpR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fdhF_1 fdhF_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - mepS_1 mepS_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - topA_1 topA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - dnaC_1 dnaC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lsrD_1 lsrD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - spaS_1 spaS_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lptB_1 lptB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - hha_1 hha_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - bhsA_1 bhsA_2 bhsA_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - osmY_1 osmY_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - purR_1 purR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - artM_1 artM_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - pdhR_1 pdhR_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - intA_1 intA_2 intA_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - cusA_1 cusA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - deoR_1 deoR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - tdh_1 tdh_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - ais_1 ais_2 ais_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - acs_1 acs_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - gntR_1 gntR_2 gntR_3\n",
            "[19:56:23] Fixed 3 duplicate /gene - fixB_1 fixB_2 fixB_3\n",
            "[19:56:23] Fixed 3 duplicate /gene - mshA_1 mshA_2 mshA_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - gltC_1 gltC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ecfA1_1 ecfA1_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - aglB_1 aglB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - mipA_1 mipA_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - ydcV_1 ydcV_2 ydcV_3\n",
            "[19:56:23] Fixed 4 duplicate /gene - hyfA_1 hyfA_2 hyfA_3 hyfA_4\n",
            "[19:56:23] Fixed 2 duplicate /gene - hcpA_1 hcpA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - cusS_1 cusS_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rspA_1 rspA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yphB_1 yphB_2\n",
            "[19:56:23] Fixed 4 duplicate /gene - rusA_1 rusA_2 rusA_3 rusA_4\n",
            "[19:56:23] Fixed 2 duplicate /gene - mglA_1 mglA_2\n",
            "[19:56:23] Fixed 5 duplicate /gene - srlR_1 srlR_2 srlR_3 srlR_4 srlR_5\n",
            "[19:56:23] Fixed 2 duplicate /gene - fic_1 fic_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lacA_1 lacA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yehB_1 yehB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lutA_1 lutA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lsrK_1 lsrK_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - truC_1 truC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - trmH_1 trmH_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ldrD_1 ldrD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ssb_1 ssb_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - tal_1 tal_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - gadX_1 gadX_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - nagB_1 nagB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fdrA_1 fdrA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ndhI_1 ndhI_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fdnH_1 fdnH_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - murR_1 murR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - malS_1 malS_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - caiA_1 caiA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - sgrR_1 sgrR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - caiT_1 caiT_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - arcC1_1 arcC1_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - repA_1 repA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yedS_1 yedS_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - prfB_1 prfB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ydhX_1 ydhX_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - manZ_1 manZ_2 manZ_3\n",
            "[19:56:23] Fixed 3 duplicate /gene - emrE_1 emrE_2 emrE_3\n",
            "[19:56:23] Fixed 3 duplicate /gene - btuD_1 btuD_2 btuD_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - pspB_1 pspB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yhhT_1 yhhT_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ldrA_1 ldrA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ulaB_1 ulaB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ccmH_1 ccmH_2\n",
            "[19:56:23] Fixed 4 duplicate /gene - lexA_1 lexA_2 lexA_3 lexA_4\n",
            "[19:56:23] Fixed 2 duplicate /gene - parM_1 parM_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - cdgI_1 cdgI_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - elfG_1 elfG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - dmsC_1 dmsC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - gatZ_1 gatZ_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ccmF_1 ccmF_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ssuC_1 ssuC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ulaA_1 ulaA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - acpP_1 acpP_2\n",
            "[19:56:23] Fixed 5 duplicate /gene - intS_1 intS_2 intS_3 intS_4 intS_5\n",
            "[19:56:23] Fixed 2 duplicate /gene - ybcO_1 ybcO_2\n",
            "[19:56:23] Fixed 4 duplicate /gene - rhaR_1 rhaR_2 rhaR_3 rhaR_4\n",
            "[19:56:23] Fixed 2 duplicate /gene - ydjE_1 ydjE_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rhsD_1 rhsD_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - norR_1 norR_2 norR_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - elaD_1 elaD_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - tfaE_1 tfaE_2 tfaE_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - cutC_1 cutC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - agaC_1 agaC_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - manP_1 manP_2 manP_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - melR_1 melR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - cysH_1 cysH_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - pncC_1 pncC_2 pncC_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - leuC_1 leuC_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - vgrG1_1 vgrG1_2 vgrG1_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - dinI_1 dinI_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - fixX_1 fixX_2 fixX_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - tmk_1 tmk_2\n",
            "[19:56:23] Fixed 9 duplicate /gene - xerC_1 xerC_2 xerC_3 xerC_4 xerC_5 xerC_6 xerC_7 xerC_8 xerC_9\n",
            "[19:56:23] Fixed 2 duplicate /gene - rep_1 rep_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - pgrR_1 pgrR_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - umuD_1 umuD_2 umuD_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - acrR_1 acrR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - malP_1 malP_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fdnG_1 fdnG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ttdT_1 ttdT_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - php_1 php_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - prs_1 prs_2\n",
            "[19:56:23] Fixed 4 duplicate /gene - ompX_1 ompX_2 ompX_3 ompX_4\n",
            "[19:56:23] Fixed 2 duplicate /gene - yajI_1 yajI_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - gfcB_1 gfcB_2\n",
            "[19:56:23] Fixed 4 duplicate /gene - flu_1 flu_2 flu_3 flu_4\n",
            "[19:56:23] Fixed 2 duplicate /gene - slp_1 slp_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - coaA_1 coaA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - manX_1 manX_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - aspT_1 aspT_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - abaF_1 abaF_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - arnC_1 arnC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ytfJ_1 ytfJ_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - cynR_1 cynR_2 cynR_3\n",
            "[19:56:23] Fixed 5 duplicate /gene - yceD_1 yceD_2 yceD_3 yceD_4 yceD_5\n",
            "[19:56:23] Fixed 2 duplicate /gene - oleD_1 oleD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - yagU_1 yagU_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - narL_1 narL_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ftsH_1 ftsH_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - mdh_1 mdh_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rbsK_1 rbsK_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - rhsC_1 rhsC_2 rhsC_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - lolD_1 lolD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lapB_1 lapB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lsrB_1 lsrB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fixA_1 fixA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fryB_1 fryB_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - hokE_1 hokE_2 hokE_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - ubiJ_1 ubiJ_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - bglH_1 bglH_2 bglH_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - dcuA_1 dcuA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lon_1 lon_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - yhdJ_1 yhdJ_2 yhdJ_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - kefC_1 kefC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - aaeB_1 aaeB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - dnaK_1 dnaK_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - bioF_1 bioF_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - cysE_1 cysE_2 cysE_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - ompD_1 ompD_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - por_1 por_2 por_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - dcuD_1 dcuD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - higB-2_1 higB-2_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - argE_1 argE_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - nohA_1 nohA_2 nohA_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - fucP_1 fucP_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ygiW_1 ygiW_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - sbmC_1 sbmC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ftnA_1 ftnA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rlhA_1 rlhA_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - rrrD_1 rrrD_2 rrrD_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - yrdA_1 yrdA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - proQ_1 proQ_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - rapA_1 rapA_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - folP_1 folP_2 folP_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - bla_1 bla_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - aaeA_1 aaeA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ulaC_1 ulaC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - mepM_1 mepM_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - msbA_1 msbA_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - era_1 era_2 era_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - silE_1 silE_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - srmB_1 srmB_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - traI_1 traI_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - bioD1_1 bioD1_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ybdO_1 ybdO_2\n",
            "[19:56:23] Fixed 3 duplicate /gene - hlyE_1 hlyE_2 hlyE_3\n",
            "[19:56:23] Fixed 2 duplicate /gene - ybgD_1 ybgD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ampG_1 ampG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - azoR_1 azoR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - tsaR_1 tsaR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - nlpD_1 nlpD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - recD_1 recD_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - epsE_1 epsE_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - gntK_1 gntK_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - dgcQ_1 dgcQ_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - pal_1 pal_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ptsG_1 ptsG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fes_1 fes_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lldR_1 lldR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - lptG_1 lptG_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - hin_1 hin_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ygdR_1 ygdR_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - fruK_1 fruK_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - mlc_1 mlc_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - higA_1 higA_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - chbC_1 chbC_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - ydfO_1 ydfO_2\n",
            "[19:56:23] Fixed 2 duplicate /gene - cirA_1 cirA_2\n",
            "[19:56:23] Fixed 270 colliding /gene names.\n",
            "[19:56:23] Adding /locus_tag identifiers\n",
            "[19:56:23] Assigned 5144 locus_tags to CDS and RNA features.\n",
            "[19:56:23] Writing outputs to /content/5_prokka_annotation/\n",
            "[19:56:31] Generating annotation statistics file\n",
            "[19:56:31] Generating Genbank and Sequin files\n",
            "[19:56:31] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.14.6 from https://github.com/tseemann/prokka' -Z \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.err -i \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.fsa 2> /dev/null\n",
            "[19:56:46] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.14.6 from https://github.com/tseemann/prokka' -Z \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.err -i \\/content\\/5_prokka_annotation\\/PROKKA_11152021\\.fsa 2> /dev/null\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "U89I4UdFHruU",
        "outputId": "39cf423b-4bea-4bf2-9ea9-d8b6c2eff0f2"
      },
      "source": [
        "!head /content/4_spades3libs/scaffolds.fasta"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            ">NODE_1_length_2815616_cov_74.3819_ID_564387\n",
            "GGTGCAGATGCTCGACCGGCTGGAAAGTGAAATCCTGGCTGACCGGGTGAGTGAGGAAAG\n",
            "CCGCCGCTGGCTGGCATCGTGCGGCCTGACCGTGGAGCAGATGCAAAACCAGATGGACCC\n",
            "GGTGTACACGCCGGCGCGAAAAATCCACCTGTACCACTGCGACCATCGCGGCCTGCCGCT\n",
            "GGCGCTCATCAGCACGGAAGGGGCAACAGCGTGGTGCGCAGAATATGATGAATGGGGCAA\n",
            "CCTGCTGAATGAAGAGAACCCGCATCAGCTGCAGCAGCTTATCCGGCTGCCGGGGCAGCA\n",
            "GTATGATGAGGAGTCCGGCCTGTATTACAACCGCCACCGCTATTATGACCCGCTGCAGGG\n",
            "GCGGTATATCACTCAGGATCCGATTGGGCTGAAGGGGGGATGGAATTTTTATCAGTATCC\n",
            "GTTGAATCCGGTCATAAATGTAGATCCGCAAGGTTTGGTTGATATAAATTTATACCCCGA\n",
            "AAGTGATCTTATCCATTCTGTAGCTGATGAGATTAATATCCCAGGCGTTTTCACAATCGG\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bsX-tEZxM9nZ"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "nCWno5TVM-Pe"
      },
      "source": [
        "## **6. Finding the closest relative of E. coli X**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "4MZ_MR8JNPCk"
      },
      "source": [
        "Цель: найти известный геном, наиболее близкий к собранному нами геному *E.Coli* X.\n",
        "\n",
        "Поиск производим по базе RefSeq при помощи BLAST по какому-нибудь консервативному гену  (в нашем случае выбрали 16S ribosomal RNA). Для этого, для начала, нам надо найти ген 16S RNA в файле собранного генома (using **barrnap**)."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "zUZGZEKUS1JI"
      },
      "source": [
        "!mkdir 6_closest_relative"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8qhQ37JxSJ7D"
      },
      "source": [
        "Search and output found results in .fasta file:"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0vg1Uz81SInw",
        "outputId": "63080c2f-1c38-4385-af24-a0c42724bdc1"
      },
      "source": [
        "!barrnap /content/4_spades3libs/scaffolds.fasta > 6_closest_relative/rRNA_gene_prediction.gff"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[20:55:10] This is barrnap 0.8\n",
            "[20:55:10] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
            "[20:55:10] Obtained from https://github.com/tseemann/barrnap\n",
            "[20:55:10] Detected operating system: linux\n",
            "[20:55:10] Adding /usr/bin/../binaries/linux to end of PATH\n",
            "[20:55:10] Using HMMER binary: /usr/bin/nhmmer\n",
            "[20:55:10] Will use 8 threads\n",
            "[20:55:10] Setting evalue cutoff to 1e-06\n",
            "[20:55:10] Will tag genes < 0.8 of expected length.\n",
            "[20:55:10] Will reject genes < 0.5 of expected length.\n",
            "[20:55:10] Using database: /usr/share/barrnap/db/bac.hmm\n",
            "[20:55:10] Scanning /content/4_spades3libs/scaffolds.fasta for bac rRNA genes... please wait\n",
            "[20:55:10] Command: /usr/bin/nhmmer --cpu 8 -E 1e-06 --w_length 3878  -o /dev/null --tblout /dev/stdout \\/usr\\/share\\/barrnap\\/db\\/bac\\.hmm \\/content\\/4_spades3libs\\/scaffolds\\.fasta\n",
            "[20:55:12] Found: 16S_rRNA NODE_1_length_2815616_cov_74.3819_ID_564387 L=1538/1585 326359..327896 - 16S ribosomal RNA\n",
            "[20:55:12] Found: 16S_rRNA NODE_1_length_2815616_cov_74.3819_ID_564387 L=1538/1585 595966..597503 - 16S ribosomal RNA\n",
            "[20:55:12] Found: 16S_rRNA NODE_1_length_2815616_cov_74.3819_ID_564387 L=1538/1585 2504403..2505940 - 16S ribosomal RNA\n",
            "[20:55:12] Found: 16S_rRNA NODE_5_length_236041_cov_85.7779_ID_563492 L=1538/1585 43835..45372 + 16S ribosomal RNA\n",
            "[20:55:12] Found: 16S_rRNA NODE_5_length_236041_cov_85.7779_ID_563492 L=1538/1585 85462..86999 + 16S ribosomal RNA\n",
            "[20:55:12] Found: 16S_rRNA NODE_6_length_209194_cov_80.6603_ID_563563 L=1538/1585 111955..113492 + 16S ribosomal RNA\n",
            "[20:55:12] Rejecting short 285 nt predicted 16S_rRNA. Adjust via --reject option.\n",
            "[20:55:12] Rejecting short 406 nt predicted 16S_rRNA. Adjust via --reject option.\n",
            "[20:55:12] Rejecting short 180 nt predicted 16S_rRNA. Adjust via --reject option.\n",
            "[20:55:12] Rejecting short 204 nt predicted 16S_rRNA. Adjust via --reject option.\n",
            "[20:55:12] Found: 5S_rRNA NODE_184_length_223_cov_0.720238_ID_565088 L=111/119 95..205 - 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_1_length_2815616_cov_74.3819_ID_564387 L=111/119 322892..323002 - 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_1_length_2815616_cov_74.3819_ID_564387 L=111/119 592407..592517 - 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_1_length_2815616_cov_74.3819_ID_564387 L=111/119 2500844..2500954 - 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_5_length_236041_cov_85.7779_ID_563492 L=111/119 90356..90466 + 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_6_length_209194_cov_80.6603_ID_563563 L=111/119 116849..116959 + 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_9_length_87669_cov_86.9393_ID_563775 L=111/119 84700..84810 - 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_1_length_2815616_cov_74.3819_ID_564387 L=111/119 592162..592272 - 5S ribosomal RNA\n",
            "[20:55:12] Found: 5S_rRNA NODE_5_length_236041_cov_85.7779_ID_563492 L=102/119 48983..49084 + 5S ribosomal RNA\n",
            "[20:55:12] Found 15 ribosomal RNA features.\n",
            "[20:55:12] Sorting features and outputting GFF3...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8aemW7xxSTPA"
      },
      "source": [
        "<br>\n",
        "\n",
        "16s RNA is very important for bacteria, that's why it can have multiple copies of this gene (and that's why we got several matches)."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "sLDhIkDRZBHy"
      },
      "source": [
        "Now, we have to extract all found for 16s rRNA sequences.\n",
        "\n",
        "For that, we manually remove all extra lines from 'rRNA_gene_prediction.gff' file, and use **bedtools getfasta**."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MIuKiL-PWPMi",
        "outputId": "0ff04dd9-9a10-4f35-fc89-32a33bbf574a"
      },
      "source": [
        "!bedtools getfasta -fi /content/4_spades3libs/scaffolds.fasta -bed /content/6_closest_relative/rRNA_gene_prediction.gff"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "index file /content/4_spades3libs/scaffolds.fasta.fai not found, generating...\n",
            ">NODE_1_length_2815616_cov_74.3819_ID_564387:326358-327896\n",
            "AAGGAGGTGATCCAACCGCAGGTTCCCCTACGGTTACCTTGTTACGACTTCACCCCAGTCATGAATCACAAAGTGGTAAGCGCCCTCCCGAAGGTTAAGCTACCTACTTCTTTTGCAACCCACTCCCATGGTGTGACGGGCGGTGTGTACAAGGCCCGGGAACGTATTCACCGTGGCATTCTGATCCACGATTACTAGCGATTCCGACTTCATGGAGTCGAGTTGCAGACTCCAATCCGGACTACGACGCACTTTATGAGGTCCGCTTGCTCTCGCGAGGTCGCTTCTCTTTGTATGCGCCATTGTAGCACGTGTGTAGCCCTGGTCGTAAGGGCCATGATGACTTGACGTCATCCCCACCTTCCTCCAGTTTATCACTGGCAGTCTCCTTTGAGTTCCCGGCCGGACCGCTGGCAACAAAGGATAAGGGTTGCGCTCGTTGCGGGACTTAACCCAACATTTCACAACACGAGCTGACGACAGCCATGCAGCACCTGTCTCACGGTTCCCGAAGGCACATTCTCATCTCTGAAAACTTCCGTGGATGTCAAGACCAGGTAAGGTTCTTCGCGTTGCATCGAATTAAACCACATGCTCCACCGCTTGTGCGGGCCCCCGTCAATTCATTTGAGTTTTAACCTTGCGGCCGTACTCCCCAGGCGGTCGACTTAACGCGTTAGCTCCGGAAGCCACGCCTCAAGGGCACAACCTCCAAGTCGACATCGTTTACGGCGTGGACTACCAGGGTATCTAATCCTGTTTGCTCCCCACGCTTTCGCACCTGAGCGTCAGTCTTCGTCCAGGGGGCCGCCTTCGCCACCGGTATTCCTCCAGATCTCTACGCATTTCACCGCTACACCTGGAATTCTACCCCCCTCTACGAGACTCAAGCTTGCCAGTATCAGATGCAGTTCCCAGGTTGAGCCCGGGGATTTCACATCTGACTTAACAAACCGCCTGCGTGCGCTTTACGCCCAGTAATTCCGATTAACGCTTGCACCCTCCGTATTACCGCGGCTGCTGGCACGGAGTTAGCCGGTGCTTCTTCTGCGGGTAACGTCAATGAGCAAAGGTATTAACTTTACTCCCTTCCTCCCCGCTGAAAGTACTTTACAACCCGAAGGCCTTCTTCATACACGCGGCATGGCTGCATCAGGCTTGCGCCCATTGTGCAATATTCCCCACTGCTGCCTCCCGTAGGAGTCTGGACCGTGTCTCAGTTCCAGTGTGGCTGGTCATCCTCTCAGACCAGCTAGGGATCGTCGCCTTGGTGAGCCGTTACCCCACCAACAAGCTAATCCCATCTGGGCACATCCGATGGCAAGAGGCCCGAAGGTCCCCCTCTTTGGTCTTGCGACGTTATGCGGTATTAGCTACCGTTTCCAGTAGTTATCCCCCTCCATCAGGCAGTTTCCCAGACATTACTCACCCGTCCGCCACTCGTCAGCGAAACAGCAAGCTGTTTCCTGTTACCGTTCGACTTGCATGTGTTAGGCCTGCCGCCAGCGTTCAATCTGAGCCATGATCAAACTCTTCAA\n",
            ">NODE_1_length_2815616_cov_74.3819_ID_564387:595965-597503\n",
            "AAGGAGGTGATCCAACCGCAGGTTCCCCTACGGTTACCTTGTTACGACTTCACCCCAGTCATGAATCACAAAGTGGTAAGCGCCCTCCCGAAGGTTAAGCTACCTACTTCTTTTGCAACCCACTCCCATGGTGTGACGGGCGGTGTGTACAAGGCCCGGGAACGTATTCACCGTGGCATTCTGATCCACGATTACTAGCGATTCCGACTTCATGGAGTCGAGTTGCAGACTCCAATCCGGACTACGACGCACTTTATGAGGTCCGCTTGCTCTCGCGAGGTCGCTTCTCTTTGTATGCGCCATTGTAGCACGTGTGTAGCCCTGGTCGTAAGGGCCATGATGACTTGACGTCATCCCCACCTTCCTCCAGTTTATCACTGGCAGTCTCCTTTGAGTTCCCGGCCGGACCGCTGGCAACAAAGGATAAGGGTTGCGCTCGTTGCGGGACTTAACCCAACATTTCACAACACGAGCTGACGACAGCCATGCAGCACCTGTCTCACGGTTCCCGAAGGCACATTCTCATCTCTGAAAACTTCCGTGGATGTCAAGACCAGGTAAGGTTCTTCGCGTTGCATCGAATTAAACCACATGCTCCACCGCTTGTGCGGGCCCCCGTCAATTCATTTGAGTTTTAACCTTGCGGCCGTACTCCCCAGGCGGTCGACTTAACGCGTTAGCTCCGGAAGCCACGCCTCAAGGGCACAACCTCCAAGTCGACATCGTTTACGGCGTGGACTACCAGGGTATCTAATCCTGTTTGCTCCCCACGCTTTCGCACCTGAGCGTCAGTCTTCGTCCAGGGGGCCGCCTTCGCCACCGGTATTCCTCCAGATCTCTACGCATTTCACCGCTACACCTGGAATTCTACCCCCCTCTACGAGACTCAAGCTTGCCAGTATCAGATGCAGTTCCCAGGTTGAGCCCGGGGATTTCACATCTGACTTAACAAACCGCCTGCGTGCGCTTTACGCCCAGTAATTCCGATTAACGCTTGCACCCTCCGTATTACCGCGGCTGCTGGCACGGAGTTAGCCGGTGCTTCTTCTGCGGGTAACGTCAATGAGCAAAGGTATTAACTTTACTCCCTTCCTCCCCGCTGAAAGTACTTTACAACCCGAAGGCCTTCTTCATACACGCGGCATGGCTGCATCAGGCTTGCGCCCATTGTGCAATATTCCCCACTGCTGCCTCCCGTAGGAGTCTGGACCGTGTCTCAGTTCCAGTGTGGCTGGTCATCCTCTCAGACCAGCTAGGGATCGTCGCCTTGGTGAGCCGTTACCCCACCAACAAGCTAATCCCATCTGGGCACATCCGATGGCAAGAGGCCCGAAGGTCCCCCTCTTTGGTCTTGCGACGTTATGCGGTATTAGCTACCGTTTCCAGTAGTTATCCCCCTCCATCAGGCAGTTTCCCAGACATTACTCACCCGTCCGCCACTCGTCAGCGAAACAGCAAGCTGTTTCCTGTTACCGTTCGACTTGCATGTGTTAGGCCTGCCGCCAGCGTTCAATCTGAGCCATGATCAAACTCTTCAA\n",
            ">NODE_1_length_2815616_cov_74.3819_ID_564387:2504402-2505940\n",
            "AAGGAGGTGATCCAACCGCAGGTTCCCCTACGGTTACCTTGTTACGACTTCACCCCAGTCATGAATCACAAAGTGGTAAGCGCCCTCCCGAAGGTTAAGCTACCTACTTCTTTTGCAACCCACTCCCATGGTGTGACGGGCGGTGTGTACAAGGCCCGGGAACGTATTCACCGTGGCATTCTGATCCACGATTACTAGCGATTCCGACTTCATGGAGTCGAGTTGCAGACTCCAATCCGGACTACGACGCACTTTATGAGGTCCGCTTGCTCTCGCGAGGTCGCTTCTCTTTGTATGCGCCATTGTAGCACGTGTGTAGCCCTGGTCGTAAGGGCCATGATGACTTGACGTCATCCCCACCTTCCTCCAGTTTATCACTGGCAGTCTCCTTTGAGTTCCCGGCCGGACCGCTGGCAACAAAGGATAAGGGTTGCGCTCGTTGCGGGACTTAACCCAACATTTCACAACACGAGCTGACGACAGCCATGCAGCACCTGTCTCACGGTTCCCGAAGGCACATTCTCATCTCTGAAAACTTCCGTGGATGTCAAGACCAGGTAAGGTTCTTCGCGTTGCATCGAATTAAACCACATGCTCCACCGCTTGTGCGGGCCCCCGTCAATTCATTTGAGTTTTAACCTTGCGGCCGTACTCCCCAGGCGGTCGACTTAACGCGTTAGCTCCGGAAGCCACGCCTCAAGGGCACAACCTCCAAGTCGACATCGTTTACGGCGTGGACTACCAGGGTATCTAATCCTGTTTGCTCCCCACGCTTTCGCACCTGAGCGTCAGTCTTCGTCCAGGGGGCCGCCTTCGCCACCGGTATTCCTCCAGATCTCTACGCATTTCACCGCTACACCTGGAATTCTACCCCCCTCTACGAGACTCAAGCTTGCCAGTATCAGATGCAGTTCCCAGGTTGAGCCCGGGGATTTCACATCTGACTTAACAAACCGCCTGCGTGCGCTTTACGCCCAGTAATTCCGATTAACGCTTGCACCCTCCGTATTACCGCGGCTGCTGGCACGGAGTTAGCCGGTGCTTCTTCTGCGGGTAACGTCAATGAGCAAAGGTATTAACTTTACTCCCTTCCTCCCCGCTGAAAGTACTTTACAACCCGAAGGCCTTCTTCATACACGCGGCATGGCTGCATCAGGCTTGCGCCCATTGTGCAATATTCCCCACTGCTGCCTCCCGTAGGAGTCTGGACCGTGTCTCAGTTCCAGTGTGGCTGGTCATCCTCTCAGACCAGCTAGGGATCGTCGCCTTGGTGAGCCGTTACCCCACCAACAAGCTAATCCCATCTGGGCACATCCGATGGCAAGAGGCCCGAAGGTCCCCCTCTTTGGTCTTGCGACGTTATGCGGTATTAGCTACCGTTTCCAGTAGTTATCCCCCTCCATCAGGCAGTTTCCCAGACATTACTCACCCGTCCGCCACTCGTCAGCGAAACAGCAAGCTGTTTCCTGTTACCGTTCGACTTGCATGTGTTAGGCCTGCCGCCAGCGTTCAATCTGAGCCATGATCAAACTCTTCAA\n",
            ">NODE_5_length_236041_cov_85.7779_ID_563492:43834-45372\n",
            "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\n",
            ">NODE_5_length_236041_cov_85.7779_ID_563492:85461-86999\n",
            "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\n",
            ">NODE_6_length_209194_cov_80.6603_ID_563563:111954-113492\n",
            "TTGAAGAGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAACAGCTTGCTGTTTCGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTTGTTGGTGGGGTAACGGCTCACCAAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_InNXExLPCj-"
      },
      "source": [
        "<br>\n",
        "\n",
        "On the next step we use BLAST (http://blast.ncbi.nlm.nih.gov) to search for similar genome in the RefSeq db with the following **SEARCH PARAMETERS**:\n",
        "\n",
        "* Nucleotide blast\n",
        "* \"RefSeq Genome Database\" in \"Database\" field\n",
        "* \"Escherchia coli (taxid:562)\" in \"Organism\" field\n",
        "* Enter .fasta from the previous cell into the \"Enter Query Sequence\" field.\n",
        "* \"1900/01/01:2011/01/01[PDAT]\" in \"Entrez Query\" field to search only among genomes that were present in the GenBank database at the beginning of 2011"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "xl2jk-vrbiLn"
      },
      "source": [
        "<div>\n",
        "<img src=\"https://thumbs.gfycat.com/GrimyEthicalAsianpiedstarling-size_restricted.gif\" alt=\"searching\" width = \"100px\"/>\n",
        "</div>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bU9Mp4WqcL7s"
      },
      "source": [
        "$100\\%$ identity found for \"Escherichia coli 55989, complete sequence\" (Sequence ID: [NC_011748.1](https://www.ncbi.nlm.nih.gov/nucleotide/NC_011748.1?report=genbank&log$=nuclalign&blast_rank=1&RID=T5JMACUF013)).\n",
        "\n",
        "Save this sequence as \"*55989.fasta*\".\n",
        "\n",
        "If no time limits are specified, then \n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cDWb6iu0nmJE"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "K1FMu7DNnnHD"
      },
      "source": [
        "## **7. What is the genetic cause of HUS?**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "7nbyS-Sfno10"
      },
      "source": [
        "Our reference belongs to an enteroaggregative *E.Coli* strain, harboring the pAA plasmid, which contains aggregative adherence fimbria (AAF) genes allowing bacteria to stick to cells in the intestine."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Oz4iC-7Ts6L6"
      },
      "source": [
        "Fimbria on *E.Coli*:\n",
        "<img src='https://upload.wikimedia.org/wikipedia/commons/d/d4/E._coli_fimbriae.png' height = '150'>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cuK6OzQ4CyXr"
      },
      "source": [
        "$2$ insertions were found in *E.Coli* X genome using **Muave** tool:\n",
        "\n",
        " gene  |           Product               |       position      | length |\n",
        "-------|---------------------------------|---------------------|--------|\n",
        "*stxA* | Shiga toxin subunit A precursor | $3483886 - 3484845$ |  $959$ |\n",
        "*stxB* | Shiga toxin subunit B precursor | $3483605 - 3483874$ |  $269$ |"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "sv14-GNwcGf2"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "WRQQUprm5Z_n"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "G8oBKd5dcL8p"
      },
      "source": [
        "## **8. Tracing the source of toxin genes in E. coli X**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "c2-0jxdhFfRW"
      },
      "source": [
        "Делаем Zoom out и смотрим на аннотацию соседних генов. Видим, что там много мобильных элементов. Судя по всему, именно благодаря ним *E.Coli* X и преобрела additional DNA with virulence factors (shinga toxin precursors).\n",
        "\n",
        "\n",
        "Phage integrases are enzymes that mediate unidirectional site-specific recombination between two DNA recognition sequences, the phage attachment site, attP, and the bacterial attachment site, attB.\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0rtdJa51bpMa"
      },
      "source": [
        "<img src='https://psv4.userapi.com/c536132/u61105779/docs/d16/282f3ba9613d/Bez_Nazvania.png?extra=HxagZ1qWQrrbFDjz4r1NDxOXRoA6b8S-zlaIZS5l8TWTmdYYG6VKBZ1bovwgr90BXJzrT0UDsyM8v-KZGfHHH58XIMYAT6AWgJ2QwaH6aX2E5X3c10bTloMCfw8Rxp992b0zBj6YJdQ0kaxuDiHL4Q'>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "hSe_QxzucnPe"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CxJmm6akkjmq"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "v4zTus1okkrg"
      },
      "source": [
        "## **9. Antibiotic resistance detection**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "s1eFAarMklxz"
      },
      "source": [
        "Checking resistance of reference (55989.fasta) *E.Coli* sequence and for assembled (scaffolds.fasta) *E.Coli* X sequence using [*ResFinder*](https://cge.cbs.dtu.dk/services/ResFinder/).\n",
        "\n",
        "$\\underline{Reference E.Coli}$ seems to be resistant only to tetracycline, whereas $\\underline{E.Coli X}$ is resistant to a much wider range of antibiotics (cefepime, ampicillin, cefotaxime, ceftazidime from beta-lactam family, antibiotic tetracycline, and folate pathway antagonist antibiotics trimethoprim and sulfamethoxazole)."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "dJ1qR_as4z_B"
      },
      "source": [
        "$~$"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "SIqn2q-X41Eu"
      },
      "source": [
        "## **10. Antibiotic resistance mechanism**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "NbPJiEg2AG4l"
      },
      "source": [
        "Таким образом, при помощи горизонтального переноса генов (HGT) *E.Coli* X получил не только Shiga toxin genes, но и устойчивость к антибиотикам. "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Jn7rwqk0u55M"
      },
      "source": [
        "class A beta lactamase (EC3.5.2.6) => TEM family 5199263 - 5200123\n",
        "\n",
        "class A beta lactamase => CTX-M family  5195566 - 5196441\n",
        "\n",
        "Class C beta lactamase => BlaEC family 4758802 - 4759935"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5aQO-LX2BkOz"
      },
      "source": [
        "Нашли 3 фермента, теперь надо почитать статьи о них и понять как они дают антибиотикорезистентность *E.Coli* X."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "1LmoDmEW4LCG"
      },
      "source": [
        "# for (dirpath, dirnames, filenames) in os.walk('raw_data/'):\n",
        "#   for filename in filenames:\n",
        "#     if filename.endswith('.fastq'):\n",
        "#       pass"
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}